% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/getters_and_setters.R
\name{SetDatExpr}
\alias{SetDatExpr}
\title{SetDatExpr}
\usage{
SetDatExpr(
  seurat_obj,
  group_name,
  use_metacells = TRUE,
  group.by = NULL,
  multi.group.by = NULL,
  multi_group_name = NULL,
  return_seurat = TRUE,
  assay = NULL,
  slot = "data",
  layer = "data",
  mat = NULL,
  features = NULL,
  wgcna_name = NULL,
  ...
)
}
\arguments{
\item{seurat_obj}{A Seurat object}

\item{group_name}{A string containing a group present in the provided group.by column or in the Seurat Idents. A character vector can be provided to select multiple groups at a time.}

\item{use_metacells}{A logical determining if we use the metacells (TRUE) or the full expression matrix (FALSE)}

\item{group.by}{A string containing the name of a column in the Seurat object with cell groups (clusters, cell types, etc). If NULL (default), hdWGCNA uses the Seurat Idents as the group.}

\item{multi.group.by}{A string containing the name of a column in the Seurat object with groups for consensus WGCNA (dataset, sample, condition, etc)}

\item{multi_group_name}{A string containing the name of a group present in the multi.group.by column.}

\item{assay}{The name of the assay in the Seurat object}

\item{slot}{Slot to extract data for aggregation. Default = 'counts'. Slot is used with Seurat v4 instead of layer.}

\item{layer}{Layer to extract data for aggregation. Default = 'counts'. Layer is used with Seurat v5 instead of slot.}

\item{mat}{A Matrix containing gene expression data. Supplying a matrix using this parameter ignores other options. This is almost exclusively used for pseudobulk analysis.}

\item{features}{A list of features to use to override the features that have been previously set.}

\item{wgcna_name}{A string containing the name of the WGCNA slot in seurat_obj@misc. Default = NULL which retrieves the currently active WGCNA data}
}
\description{
This function specifies the gene expression matrix for co-expression network analysis.
}
\details{
SetDatExpr is a critical function of the hdWGCNA pipeline that determines the gene expession
matrix that will be used for network analysis. We typically use this function to select a
cell type or group of cell types for network analysis using the group.by parameter, but we provide
additional parameters for further customization.
}
\keyword{scRNA-seq}
