<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Changelog • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-news">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Changelog</h1>
      <small>Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.4.00" id="hdwgcna-0400-2024-10-09">hdWGCNA 0.4.00 (2024-10-09)<a class="anchor" aria-label="anchor" href="#hdwgcna-0400-2024-10-09"></a></h2>
<div class="section level3">
<h3 id="added-0-4-00">Added<a class="anchor" aria-label="anchor" href="#added-0-4-00"></a></h3>
<ul><li>Transcription Factor Regulatory Network Analysis functionality.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.3.03" id="hdwgcna-0303-2024-06-07">hdWGCNA 0.3.03 (2024-06-07)<a class="anchor" aria-label="anchor" href="#hdwgcna-0303-2024-06-07"></a></h2>
<div class="section level3">
<h3 id="added-0-3-03">Added<a class="anchor" aria-label="anchor" href="#added-0-3-03"></a></h3>
<ul><li>
<code>ModuleRadarPlot</code> function to visualize expression levels of each module.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-3-03">Changes<a class="anchor" aria-label="anchor" href="#changes-0-3-03"></a></h3>
<ul><li>Bugfix in <code>PlotModuleTraitCorrelation</code>
</li>
<li>Update to basics to tutorial to include <code>ModuleRadarPlot</code>
</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.3.02" id="hdwgcna-0302-2024-05-28">hdWGCNA 0.3.02 (2024-05-28)<a class="anchor" aria-label="anchor" href="#hdwgcna-0302-2024-05-28"></a></h2>
<div class="section level3">
<h3 id="added-0-3-02">Added<a class="anchor" aria-label="anchor" href="#added-0-3-02"></a></h3>
<p>New functionality and methods for module preservation analysis using the <a href="https://github.com/sritchie73/NetRep" class="external-link">NetRep method from Ritchie et al 2016</a>. - <code>ModuleTopologyHeatmap</code> function. - <code>ModuleTopologyBarplot</code> function. - <code>ModulePreservationNetrep</code> function. - <code>PlotModulePreservationLollipop</code> function.</p>
</div>
<div class="section level3">
<h3 id="changes-0-3-02">Changes<a class="anchor" aria-label="anchor" href="#changes-0-3-02"></a></h3>
<ul><li>Update to module preservation tutorial to demonstrate how to perform module preservation analysis using NetRep.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.3.01" id="hdwgcna-0301-2024-03-07">hdWGCNA 0.3.01 (2024-03-07)<a class="anchor" aria-label="anchor" href="#hdwgcna-0301-2024-03-07"></a></h2>
<div class="section level3">
<h3 id="added-0-3-01">Added<a class="anchor" aria-label="anchor" href="#added-0-3-01"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-3-01">Changes<a class="anchor" aria-label="anchor" href="#changes-0-3-01"></a></h3>
<ul><li>Bugfix in <code>ModulePreservation</code>
</li>
<li>Update to module preservation tutorial and project modules tutorial.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.3.00" id="hdwgcna-0300-2024-02-27">hdWGCNA 0.3.00 (2024-02-27)<a class="anchor" aria-label="anchor" href="#hdwgcna-0300-2024-02-27"></a></h2>
<div class="section level3">
<h3 id="added-0-3-00">Added<a class="anchor" aria-label="anchor" href="#added-0-3-00"></a></h3>
<ul><li>First version with support for Seurat v5.</li>
<li>Updated the network visualization tutorial with a tutorial for making custom networks.</li>
</ul></div>
<div class="section level3">
<h3 id="changes-0-3-00">Changes<a class="anchor" aria-label="anchor" href="#changes-0-3-00"></a></h3>
<ul><li>Changed <code>FindDMEs</code> and <code>FindAllDMEs</code> to perform differential testing with module expression scores.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.27" id="hdwgcna-0227-2024-01-29">hdWGCNA 0.2.27 (2024-01-29)<a class="anchor" aria-label="anchor" href="#hdwgcna-0227-2024-01-29"></a></h2>
<div class="section level3">
<h3 id="added-0-2-27">Added<a class="anchor" aria-label="anchor" href="#added-0-2-27"></a></h3>
<ul><li>New option to <code>MetacellsByGroups</code> to specify <code>dims</code>.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-27">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-27"></a></h3>
<ul><li>Fixed <code>HubGeneNetworkPlot</code> to allow selecting specific modules.</li>
<li>
<code>GetHubGenes</code> now returns genes in order from highest to lowest kME in each module.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.26" id="hdwgcna-0226-2023-12-05">hdWGCNA 0.2.26 (2023-12-05)<a class="anchor" aria-label="anchor" href="#hdwgcna-0226-2023-12-05"></a></h2>
<div class="section level3">
<h3 id="added-0-2-26">Added<a class="anchor" aria-label="anchor" href="#added-0-2-26"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-26">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-26"></a></h3>
<ul><li>New options for changing colors in <code>EnrichrBarPlot</code>.</li>
<li>
<code>ConstructNetwork</code> naming of temporary files updated.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.25" id="hdwgcna-0225-2023-11-28">hdWGCNA 0.2.25 (2023-11-28)<a class="anchor" aria-label="anchor" href="#hdwgcna-0225-2023-11-28"></a></h2>
<div class="section level3">
<h3 id="added-0-2-25">Added<a class="anchor" aria-label="anchor" href="#added-0-2-25"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-25">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-25"></a></h3>
<ul><li>Requires Seurat version 4. Will update to support v5 in the future.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.24" id="hdwgcna-0224-2023-09-28">hdWGCNA 0.2.24 (2023-09-28)<a class="anchor" aria-label="anchor" href="#hdwgcna-0224-2023-09-28"></a></h2>
<div class="section level3">
<h3 id="added-0-2-24">Added<a class="anchor" aria-label="anchor" href="#added-0-2-24"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-24">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-24"></a></h3>
<ul><li>Update call to Harmony in ModuleEigengenes function</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.23" id="hdwgcna-0223-2023-09-10">hdWGCNA 0.2.23 (2023-09-10)<a class="anchor" aria-label="anchor" href="#hdwgcna-0223-2023-09-10"></a></h2>
<div class="section level3">
<h3 id="added-0-2-23">Added<a class="anchor" aria-label="anchor" href="#added-0-2-23"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-23">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-23"></a></h3>
<ul><li>Fixed bug in SetDatExpr</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.22" id="hdwgcna-0222-2023-09-08">hdWGCNA 0.2.22 (2023-09-08)<a class="anchor" aria-label="anchor" href="#hdwgcna-0222-2023-09-08"></a></h2>
<div class="section level3">
<h3 id="added-0-2-22">Added<a class="anchor" aria-label="anchor" href="#added-0-2-22"></a></h3>
<ul><li>New tutorial for hdWGCNA with pseudobulk data, including some new functions like <code>ConstructPseudobulk</code>.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-22">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-22"></a></h3>
<ul><li>Updated <code>SetDatExpr</code> and <code>SetMultiExpr</code> to use a pseudobulk expression matrix.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.21" id="hdwgcna-0221-2023-08-31">hdWGCNA 0.2.21 (2023-08-31)<a class="anchor" aria-label="anchor" href="#hdwgcna-0221-2023-08-31"></a></h2>
<div class="section level3">
<h3 id="added-0-2-21">Added<a class="anchor" aria-label="anchor" href="#added-0-2-21"></a></h3>
<ul><li>Additional checks for wgcna_name in several functions.</li>
<li>New section to the DME tutorial to show how to run it in a loop for multiple clusters.</li>
</ul></div>
<div class="section level3">
<h3 id="changes-0-2-21">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-21"></a></h3>
<ul><li>None.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.20" id="hdwgcna-0220-2023-08-17">hdWGCNA 0.2.20 (2023-08-17)<a class="anchor" aria-label="anchor" href="#hdwgcna-0220-2023-08-17"></a></h2>
<div class="section level3">
<h3 id="added-0-2-20">Added<a class="anchor" aria-label="anchor" href="#added-0-2-20"></a></h3>
<ul><li>None.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-20">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-20"></a></h3>
<ul><li>Dependency for tester pacakge.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.19" id="hdwgcna-0219-2023-06-13">hdWGCNA 0.2.19 (2023-06-13)<a class="anchor" aria-label="anchor" href="#hdwgcna-0219-2023-06-13"></a></h2>
<div class="section level3">
<h3 id="added-0-2-19">Added<a class="anchor" aria-label="anchor" href="#added-0-2-19"></a></h3>
<ul><li>None.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-19">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-19"></a></h3>
<ul><li>Updated README to include publication, and fixed several igraph function calls.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.18" id="hdwgcna-0218-2023-04-14">hdWGCNA 0.2.18 (2023-04-14)<a class="anchor" aria-label="anchor" href="#hdwgcna-0218-2023-04-14"></a></h2>
<div class="section level3">
<h3 id="added-0-2-18">Added<a class="anchor" aria-label="anchor" href="#added-0-2-18"></a></h3>
<ul><li>None.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-18">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-18"></a></h3>
<ul><li>We noticed on rare occasion that EnrichR would give duplicated results for different modules, so we added a new option in <code>RunEnrichr</code> to wait in between sending requests to the EnrichR server (default is 5 seconds).</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.17" id="hdwgcna-0217-2023-03-27">hdWGCNA 0.2.17 (2023-03-27)<a class="anchor" aria-label="anchor" href="#hdwgcna-0217-2023-03-27"></a></h2>
<div class="section level3">
<h3 id="added-0-2-17">Added<a class="anchor" aria-label="anchor" href="#added-0-2-17"></a></h3>
<ul><li>None.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-17">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-17"></a></h3>
<ul><li>New error checks in <code>SetupForWGCNA</code> and <code>SelectNetworkGenes</code>
</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.16" id="hdwgcna-0216-2023-03-20">hdWGCNA 0.2.16 (2023-03-20)<a class="anchor" aria-label="anchor" href="#hdwgcna-0216-2023-03-20"></a></h2>
<div class="section level3">
<h3 id="added-0-2-16">Added<a class="anchor" aria-label="anchor" href="#added-0-2-16"></a></h3>
<ul><li>
<code>PlotDMEsLollipop</code> function to visualize differential module eigengenes.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-16">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-16"></a></h3>
<ul><li>New error checks in <code>MetacellsByGroups</code>
</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.15" id="hdwgcna-0215-2023-03-02">hdWGCNA 0.2.15 (2023-03-02)<a class="anchor" aria-label="anchor" href="#hdwgcna-0215-2023-03-02"></a></h2>
<div class="section level3">
<h3 id="added-0-2-15">Added<a class="anchor" aria-label="anchor" href="#added-0-2-15"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-15">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-15"></a></h3>
<ul><li>Bugfix to allow <code>ResetModuleNames</code> and <code>ResetModuleColors</code> to work when a grey module is not present.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.14" id="hdwgcna-0214-2023-02-14">hdWGCNA 0.2.14 (2023-02-14)<a class="anchor" aria-label="anchor" href="#hdwgcna-0214-2023-02-14"></a></h2>
<div class="section level3">
<h3 id="added-0-2-14">Added<a class="anchor" aria-label="anchor" href="#added-0-2-14"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-14">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-14"></a></h3>
<ul><li>Fixed a bug in <code>SetDatExpr</code> that would throw an error when group.by=NULL was selected.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.13" id="hdwgcna-0213-2023-02-13">hdWGCNA 0.2.13 (2023-02-13)<a class="anchor" aria-label="anchor" href="#hdwgcna-0213-2023-02-13"></a></h2>
<div class="section level3">
<h3 id="added-0-2-13">Added<a class="anchor" aria-label="anchor" href="#added-0-2-13"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-13">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-13"></a></h3>
<ul><li>Fixed a bug in <code>ReassignModules</code> that cause some modules assignments to be NA for some genes.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.12" id="hdwgcna-0212-2023-02-04">hdWGCNA 0.2.12 (2023-02-04)<a class="anchor" aria-label="anchor" href="#hdwgcna-0212-2023-02-04"></a></h2>
<div class="section level3">
<h3 id="added-0-2-12">Added<a class="anchor" aria-label="anchor" href="#added-0-2-12"></a></h3>
<ul><li>Module eigengene dynamics with pseudotime tutorial</li>
<li>
<code>PlotModuleTrajectory</code> function.</li>
</ul></div>
<div class="section level3">
<h3 id="changes-0-2-12">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-12"></a></h3>
<ul><li>None.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.11" id="hdwgcna-0211-2023-01-30">hdWGCNA 0.2.11 (2023-01-30)<a class="anchor" aria-label="anchor" href="#hdwgcna-0211-2023-01-30"></a></h2>
<div class="section level3">
<h3 id="added-0-2-11">Added<a class="anchor" aria-label="anchor" href="#added-0-2-11"></a></h3>
<ul><li>MAS-Seq tutorial (still under construction)</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-11">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-11"></a></h3>
<ul><li>Bugfix in <code>ModuleConnectivity</code> and <code>ReassignModules</code>.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.1" id="hdwgcna-021-2023-01-24">hdWGCNA 0.2.1 (2023-01-24)<a class="anchor" aria-label="anchor" href="#hdwgcna-021-2023-01-24"></a></h2>
<div class="section level3">
<h3 id="added-0-2-1">Added<a class="anchor" aria-label="anchor" href="#added-0-2-1"></a></h3>
<ul><li>
<code>ReassignModules</code> function.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-1">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-1"></a></h3>
<ul><li>New option in <code>ModuleConnectivity</code> to use <code>corSparse</code> to compute the correlation, which greatly reduces runtime and memory usage.</li>
<li>New option in <code>ModuleConnectivity</code> to automatically reassign features to different modules if kME is negative.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.03" id="hdwgcna-0203-2022-12-15">hdWGCNA 0.2.03 (2022-12-15)<a class="anchor" aria-label="anchor" href="#hdwgcna-0203-2022-12-15"></a></h2>
<div class="section level3">
<h3 id="added-0-2-03">Added<a class="anchor" aria-label="anchor" href="#added-0-2-03"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-03">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-03"></a></h3>
<ul><li>Fixed a bug in <code>ResetModuleNames</code>.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.03" id="hdwgcna-0203-2022-11-10">hdWGCNA 0.2.03 (2022-11-10)<a class="anchor" aria-label="anchor" href="#hdwgcna-0203-2022-11-10"></a></h2>
<div class="section level3">
<h3 id="added-0-2-03-1">Added<a class="anchor" aria-label="anchor" href="#added-0-2-03-1"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-03-1">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-03-1"></a></h3>
<ul><li>New error checking in <code>MetaspotsByGroups</code>
</li>
<li>
<code>MetacellsByGroups</code> now keeps track of which cells were merged.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.02" id="hdwgcna-0202-2022-11-01">hdWGCNA 0.2.02 (2022-11-01)<a class="anchor" aria-label="anchor" href="#hdwgcna-0202-2022-11-01"></a></h2>
<div class="section level3">
<h3 id="added-0-2-02">Added<a class="anchor" aria-label="anchor" href="#added-0-2-02"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-02">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-02"></a></h3>
<ul><li>New warning in <code>SetupForWGCNA</code> if the user selects a very small number of genes.</li>
<li>
<code>MetaspotsByGroups</code> uses sparse matrix format internally.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.01" id="hdwgcna-0201-2022-10-06">hdWGCNA 0.2.01 (2022-10-06)<a class="anchor" aria-label="anchor" href="#hdwgcna-0201-2022-10-06"></a></h2>
<div class="section level3">
<h3 id="added-0-2-01">Added<a class="anchor" aria-label="anchor" href="#added-0-2-01"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-2-01">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-01"></a></h3>
<ul><li>Bugfix in <code>MetaspotsByGroups</code>.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2.00" id="hdwgcna-0200-2022-09-23">hdWGCNA 0.2.00 (2022-09-23)<a class="anchor" aria-label="anchor" href="#hdwgcna-0200-2022-09-23"></a></h2>
<div class="section level3">
<h3 id="added-0-2-00">Added<a class="anchor" aria-label="anchor" href="#added-0-2-00"></a></h3>
<ul><li>
<code>MetaspotsByGroups</code> to aggregate neighboring ST spots prior to network analysis.</li>
<li>Tutorial for applying hdWGCNA to spatial transcriptomics datasets.</li>
</ul></div>
<div class="section level3">
<h3 id="changes-0-2-00">Changes<a class="anchor" aria-label="anchor" href="#changes-0-2-00"></a></h3>
<ul><li>None</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.2.0001" id="hdwgcna-0120001-2022-09-19">hdWGCNA 0.1.2.0001 (2022-09-19)<a class="anchor" aria-label="anchor" href="#hdwgcna-0120001-2022-09-19"></a></h2>
<div class="section level3">
<h3 id="added-0-1-2-0001">Added<a class="anchor" aria-label="anchor" href="#added-0-1-2-0001"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-2-0001">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-2-0001"></a></h3>
<ul><li>networkType option in <code>TestSoftPowers</code>.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.2.0000" id="hdwgcna-0120000-2022-09-08">hdWGCNA 0.1.2.0000 (2022-09-08)<a class="anchor" aria-label="anchor" href="#hdwgcna-0120000-2022-09-08"></a></h2>
<div class="section level3">
<h3 id="added-0-1-2-0000">Added<a class="anchor" aria-label="anchor" href="#added-0-1-2-0000"></a></h3>
<ul><li>Differential Module Eigengene (DME) tutorial</li>
<li>FindDMEs function</li>
<li>FindAllDMEs function</li>
<li>PlotDMEsVolcano function</li>
</ul></div>
<div class="section level3">
<h3 id="changes-0-1-2-0000">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-2-0000"></a></h3>
<ul><li>None</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9015" id="hdwgcna-0119015-2022-09-06">hdWGCNA 0.1.1.9015 (2022-09-06)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119015-2022-09-06"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9015">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9015"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9015">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9015"></a></h3>
<ul><li>New data format check in <code>MetacellsByGroups</code> to ensure that the selected slot is present in the selected assay.</li>
<li>
<code>SetDatExpr</code> now backs up to the full dataset if the metacell dataset isn’t found.</li>
<li>Changed some text to clarify some points in the Module Preservation tutorial.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9014" id="hdwgcna-0119014-2022-08-26">hdWGCNA 0.1.1.9014 (2022-08-26)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119014-2022-08-26"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9014">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9014"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9014">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9014"></a></h3>
<ul><li>Bugfix in <code>ModuleConnectivity</code> that caused kMEs to be out of order.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9013" id="hdwgcna-0119013-2022-08-25">hdWGCNA 0.1.1.9013 (2022-08-25)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119013-2022-08-25"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9013">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9013"></a></h3>
<ul><li>Tutorial for using SCTransform normalized data.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9013">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9013"></a></h3>
<ul><li>None</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9012" id="hdwgcna-0119012-2022-08-24">hdWGCNA 0.1.1.9012 (2022-08-24)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119012-2022-08-24"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9012">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9012"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9012">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9012"></a></h3>
<ul><li>Now includes options for other types of correlations in <code>ModuleConnectivity</code>.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9011" id="hdwgcna-0119011-2022-08-17">hdWGCNA 0.1.1.9011 (2022-08-17)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119011-2022-08-17"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9011">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9011"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9011">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9011"></a></h3>
<ul><li>Reverted the <code>exclude_grey</code> flag back to doing nothing in the <code>ModuleEigengenes</code> function because it messed up some downstream tasks, will resolve in a future update.</li>
<li>
<code>ProjectModules</code> now excludes modules with too many missing genes in the query dataset, tunable by the <code>overlap_proportion</code> parameter.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9010" id="hdwgcna-0119010-2022-07-30">hdWGCNA 0.1.1.9010 (2022-07-30)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119010-2022-07-30"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9010">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9010"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9010">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9010"></a></h3>
<ul><li>By default, <code>ModuleEigengenes</code> does not compute MEs for the grey module. User can change behavior with the <code>exclude_grey</code> flag.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9009" id="hdwgcna-0119009-2022-07-23">hdWGCNA 0.1.1.9009 (2022-07-23)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119009-2022-07-23"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9009">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9009"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9009">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9009"></a></h3>
<ul><li>Bugfix in <code>ProjectModules</code>.</li>
<li>Bugfix in <code>MetacellsByGroups</code>
</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9008" id="hdwgcna-0119008-2022-07-21">hdWGCNA 0.1.1.9008 (2022-07-21)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119008-2022-07-21"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9008">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9008"></a></h3>
<ul><li>New tutorial for consensus co-expression network analysis.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9008">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9008"></a></h3>
<ul><li>
<code>ModuleNetworkPlot</code> and <code>RunModuleUMAP</code> now checks if <code>ModuleConnectivity</code> has been computed in order to throw a more informative error.</li>
<li>
<code>GetTOM</code> checks if the TOM file exists in order to throw a more informative error.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9007" id="hdwgcna-0119007-2022-07-20">hdWGCNA 0.1.1.9007 (2022-07-20)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119007-2022-07-20"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9007">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9007"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9007">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9007"></a></h3>
<ul><li>New warning message in <code>MetacellsByGroups</code> if there are any groups that are excluded by <code>min_cells</code>.</li>
<li>Assay in Metacell seurat object is now the same as the assay supplied to <code>MetacellsByGroups</code>, instead of the default “RNA”.</li>
<li>
<code>ModuleEigengenes</code> takes <code>assay</code> as an argument, clears up some issues with <code>RunHarmony</code>.</li>
<li>
<code>ModuleEigengenes</code> doesn’t require a “counts” slot to be present in the given assay, but now it throws an error if the normalized data slot is missing.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9006" id="hdwgcna-0119006-2022-07-14">hdWGCNA 0.1.1.9006 (2022-07-14)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119006-2022-07-14"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9006">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9006"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9006">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9006"></a></h3>
<ul><li>
<code>ConstructNetwork</code> now checks if the TOM file already exists, and whether the user wants to overwrite the existing TOM.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9005" id="hdwgcna-0119005-2022-06-17">hdWGCNA 0.1.1.9005 (2022-06-17)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119005-2022-06-17"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9005">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9005"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9005">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9005"></a></h3>
<ul><li>Added new arguments to <code>MetacellsByGroups</code> and <code>ConstructMetacells</code> to exclude very small groups (<code>min_cells</code>), to reach a target number of metacells (<code>target_metacells</code>), and to exclude metacells with too much overlap (<code>max_shared</code>).</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9004" id="hdwgcna-0119004-2022-6-13">hdWGCNA 0.1.1.9004 (2022-6-13)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119004-2022-6-13"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9004">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9004"></a></h3>
<ul><li>None</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9004">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9004"></a></h3>
<ul><li>Bug fix so <code>ModuleTraitCorrelation</code> can be run with a single trait.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9003" id="hdwgcna-0119003-2022-6-11">hdWGCNA 0.1.1.9003 (2022-6-11)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119003-2022-6-11"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9003">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9003"></a></h3>
<ul><li>
<code>GetHubGenes</code> function to extract the top hub genes from the module assignment table.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9003">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9003"></a></h3>
<ul><li>
<code>ConstructMetacells</code> stores run statistics as a table, and has a new option to exclude metacells that overlap with each other.</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9002" id="hdwgcna-0119002-2022-5-24">hdWGCNA 0.1.1.9002 (2022-5-24)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119002-2022-5-24"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9002">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9002"></a></h3>
<ul><li>
<code>ModuleEigengenes</code> checks to make sure the data has been scaled.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9002">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9002"></a></h3>
<ul><li>None</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9001" id="hdwgcna-0119001-2022-05-09">hdWGCNA 0.1.1.9001 (2022-05-09)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119001-2022-05-09"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9001">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9001"></a></h3>
<ul><li>Wrote docstring for <code>PlotKMEs</code> so it would actually be included in the package.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9001">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9001"></a></h3>
<ul><li>None</li></ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1.1.9000" id="hdwgcna-0119000-2022-05-05">hdWGCNA 0.1.1.9000 (2022-05-05)<a class="anchor" aria-label="anchor" href="#hdwgcna-0119000-2022-05-05"></a></h2>
<div class="section level3">
<h3 id="added-0-1-1-9000">Added<a class="anchor" aria-label="anchor" href="#added-0-1-1-9000"></a></h3>
<ul><li>
<code>PlotKMEs</code> function to visualize the genes in a module ranked by kME.</li></ul></div>
<div class="section level3">
<h3 id="changes-0-1-1-9000">Changes<a class="anchor" aria-label="anchor" href="#changes-0-1-1-9000"></a></h3>
<ul><li>Changed the name of the package from scWGCNA to hdWGCNA, since we plan to accommodate data types beyond single cell alone.</li>
<li>Updated tutorial to recommend computing kMEs in a specific cell population.</li>
</ul></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

