<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>hdWGCNA â€¢ hdWGCNA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="hdWGCNA">
<meta name="description" content="hdWGCNA is an R package for performing weighted gene co-expression network analysis in high dimensional -omics such as single-cell RNA-seq or spatial transcriptomics.">
<meta property="og:description" content="hdWGCNA is an R package for performing weighted gene co-expression network analysis in high dimensional -omics such as single-cell RNA-seq or spatial transcriptomics.">
<meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="articles/index.html">All vignettes</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="high-dimensional-wgcna-">high dimensional WGCNA <a class="anchor" aria-label="anchor" href="#high-dimensional-wgcna-"></a>
</h1>
</div>
<p><a href="https://github.com/smorabit/hdWGCNA/tree/dev" class="external-link"><img src="https://img.shields.io/github/r-package/v/smorabit/hdWGCNA" alt="R"></a> <a href="https://github.com/smorabit/hdWGCNA/issues" class="external-link"><img src="https://img.shields.io/github/issues/smorabit/hdWGCNA" alt="ISSUES"></a> <a href="https://www.cell.com/cell-reports-methods/fulltext/S2667-2375(23)00127-3" class="external-link"><img src="https://img.shields.io/badge/publication-Cell%20Rep%20Meth-%2300A1D7" alt="Publication"></a> <a href="https://github.com/smorabit/hdWGCNA" class="external-link"><img src="https://img.shields.io/badge/Lifecycle-Maturing-007EC6" alt="Lifecycle:Maturing"></a> <a href="https://github.com/smorabit/hdWGCNA/" class="external-link"><img src="https://img.shields.io/github/stars/smorabit/hdWGCNA?style=social" alt="Stars"></a></p>
<p>hdWGCNA is an R package for performing weighted gene co-expression network analysis <a href="https://doi.org/10.1186/1471-2105-9-559" class="external-link">(WGCNA)</a> in high dimensional transcriptomics data such as single-cell RNA-seq or spatial transcriptomics. hdWGCNA is highly modular and can construct context-specific co-expression networks across cellular and spatial hierarchies. hdWGNCA identifies modules of highly co-expressed genes and provides context for these modules via statistical testing and biological knowledge sources. hdWGCNA uses datasets formatted as <a href="https://satijalab.org/seurat/index.html" class="external-link">Seurat</a> objects. Check out the <a href="https://smorabit.github.io/hdWGCNA/articles/basic_tutorial.html">hdWGCNA in single-cell data tutorial</a> or the <a href="https://smorabit.github.io/hdWGCNA/articles/ST_basics.html">hdWGCNA in spatial transcriptomics data tutorial</a> to get started.</p>
<p><strong>New functionality</strong> hdWGCNA is now able to perform <a href="https://smorabit.github.io/hdWGCNA/articles/tf_network.html">Transcription Factor Regulatory Network Analysis</a>. This functionality was introduced in our publication <a href="https://www.sciencedirect.com/science/article/pii/S2211124724002845" class="external-link">Childs &amp; Morabito et al., Cell Reports (2024)</a>.</p>
<p>If you use hdWGCNA in your research, please cite the following papers in addition to the <a href="https://doi.org/10.1186/1471-2105-9-559" class="external-link">original WGCNA publication</a>:</p>
<ul>
<li><a href="https://www.cell.com/cell-reports-methods/fulltext/S2667-2375(23)00127-3" class="external-link">Morabito et al., Cell Reports Methods (2023)</a></li>
<li><a href="https://doi.org/10.1038/s41588-021-00894-z" class="external-link">Morabito &amp; Miyoshi et al., Nature Genetics (2021)</a></li>
</ul>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>We recommend creating an R <a href="https://docs.conda.io/en/latest/" class="external-link">conda environment</a> environment for hdWGCNA.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create new conda environment for R</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> hdWGCNA <span class="at">-c</span> conda-forge r-base r-essentials</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># activate conda environment</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate hdWGCNA</span></code></pre></div>
<p>Next open R and install the required dependencies:</p>
<ul>
<li>
<a href="https://www.bioconductor.org/" class="external-link">Bioconductor</a>, an R-based software ecosystem for bioinformatics and biostatistics.</li>
<li>
<a href="https://devtools.r-lib.org/" class="external-link">devtools</a>, a package for package development in R.</li>
<li>
<a href="https://satijalab.org/seurat/index.html" class="external-link">Seurat</a>, a general-purpose toolkit for single-cell data science.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install BiocManager</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># install Bioconductor core packages</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># install devtools</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># install latest version of Seurat from CRAN</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"Seurat"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># alternatively, install Seurat v4</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"Seurat"</span>, repos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"https://satijalab.r-universe.dev', 'https://cloud.r-project.org"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now you can install the hdWGCNA package using <code>devtools</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'smorabit/hdWGCNA'</span>, ref<span class="op">=</span><span class="st">'dev'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="suggested-reading">Suggested Reading<a class="anchor" aria-label="anchor" href="#suggested-reading"></a>
</h2>
<p>Check out the paper describing hdWGCNA, our paper introducing transcription factor regulatory network analysis with hdWGCNA, and our original description of applying WGCNA to single-nucleus RNA-seq data. For additional reading, we suggest the original WGCNA publication and papers describing relevant algorithms for co-expression network analysis.</p>
<ul>
<li><a href="https://www.cell.com/cell-reports-methods/fulltext/S2667-2375(23)00127-3" class="external-link">hdWGCNA identifies co-expression networks in high-dimensional transcriptomics data</a></li>
<li><a href="https://www.sciencedirect.com/science/article/pii/S2211124724002845" class="external-link">Relapse to cocaine seeking is regulated by medial habenula NR4A2/NURR1 in mice</a></li>
<li><a href="https://doi.org/10.1038/s41588-021-00894-z" class="external-link">Single-nucleus chromatin accessibility and transcriptomic characterization of Alzheimerâ€™s disease</a></li>
</ul>
<p>WGCNA and related algorithms:</p>
<ul>
<li><a href="https://doi.org/10.1186/1471-2105-9-559" class="external-link">WGCNA: an R package for weighted correlation network analysis</a></li>
<li><a href="https://doi.org/10.1093/bioinformatics/btm563" class="external-link">Defining clusters from a hierarchical cluster tree: the Dynamic Tree Cut package for R</a></li>
<li><a href="https://doi.org/10.1186/1752-0509-1-54" class="external-link">Eigengene networks for studying the relationships between co-expression modules</a></li>
<li><a href="https://doi.org/10.1371/journal.pcbi.1000117" class="external-link">Geometric Interpretation of Gene Coexpression Network Analysis</a></li>
<li><a href="https://doi.org/10.1371/journal.pcbi.1001057" class="external-link">Is My Network Module Preserved and Reproducible?</a></li>
</ul>
<p><strong>Note about package development:</strong> hdWGCNA is under active development, so you may run into errors and small typos. We welcome users to write <a href="https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue" class="external-link">GitHub issues</a> to report bugs, ask for help, and to request potential enhancements.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GNU GPLv3</small></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing hdWGCNA</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://smorabit.github.io" class="external-link">Sam Morabito</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7768-4856" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a> <br><small class="roles"> Funder </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
