<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential module eigengene (DME) analysis • hdWGCNA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Differential module eigengene (DME) analysis">
<meta property="og:description" content="Tutorial for performing differential module eigengene analysis.
">
<meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hdWGCNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.02</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/hdWGCNA.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Core functionality</li>
    <li>
      <a href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a>
    </li>
    <li>
      <a href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a>
    </li>
    <li>
      <a href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a>
    </li>
    <li>
      <a href="../articles/network_visualizations.html">Network visualization</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Biological context for co-expression modules</li>
    <li>
      <a href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a>
    </li>
    <li>
      <a href="../articles/module_trait_correlation.html">Module trait correlation</a>
    </li>
    <li>
      <a href="../articles/enrichment_analysis.html">Enrichment analysis</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Exploring modules in external datasets</li>
    <li>
      <a href="../articles/projecting_modules.html">Projecting modules to new datasets</a>
    </li>
    <li>
      <a href="../articles/module_preservation.html">Module preservation and reproducibility</a>
    </li>
    <li>
      <a href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Advanced topics</li>
    <li>
      <a href="../articles/consensus_wgcna.html">Consensus network analysis</a>
    </li>
    <li>
      <a href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a>
    </li>
    <li>
      <a href="../articles/motif_analysis.html">Motif analysis</a>
    </li>
    <li>
      <a href="../articles/other_metacells.html">Alternative metacell algorithms</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Other</li>
    <li>
      <a href="../articles/customization.html">Module customization</a>
    </li>
    <li>
      <a href="../articles/sctransform.html">Using SCTransform normalized data</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential module eigengene (DME) analysis</h1>
            
      
      
      <div class="hidden name"><code>differential_MEs.Rmd</code></div>

    </div>

    
    
<p>Compiled: 27-02-2024</p>
<p>Source: <code>vignettes/differential_MEs.Rmd</code></p>
<p>In this tutorial, we demonstrate how to perform differential expression analysis for our co-expression network modules. We refer to this analysis as “differential module eigengene” (DME) analysis, which reveals modules that are up- or down-regulated in specified groups of cells. For this analysis we take advantage of Seurat’s <code>FindMarkers</code> function to perform differential testing, so we have access to all of the same options.</p>
<p>First we load the single-cell dataset processed through the basics tutorial and the required R libraries for this tutorial.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># single-cell analysis package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>

<span class="co"># plotting and data science packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel" class="external-link">ggrepel</a></span><span class="op">)</span>

<span class="co"># co-expression network analysis packages:</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/" class="external-link">WGCNA</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://smorabit.github.io/hdWGCNA/">hdWGCNA</a></span><span class="op">)</span>

<span class="co"># using the cowplot theme for ggplot</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

<span class="co"># set random seed for reproducibility</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span>

<span class="co"># re-load the Zhou et al snRNA-seq dataset processed with hdWGCNA</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'Zhou_2020_hdWGCNA.rds'</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="dme-analysis-comparing-two-groups">DME analysis comparing two groups<a class="anchor" aria-label="anchor" href="#dme-analysis-comparing-two-groups"></a>
</h2>
<p>Here we discuss how to perform DME testing between two different groups. We use the hdWGCNA function <code>FindDMEs</code>, the syntax of which is similar to the Seurat function <code>FindMarkers</code>. Here we use the Mann-Whitney U test, also known as the Wilcoxon test, to compare two groups, but other tests can be specified with the <code>test.use</code> parameter.</p>
<p>Since the tutorial dataset only contains control brain samples, we will use sex to define our two groups. <code>FindDMEs</code> requires a list of barcodes for group1 and for group2. Further, we are only going to compare cells from the INH cluster since that is the group that we performed network analysis on.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">group1</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cell_type</span> <span class="op">==</span> <span class="st">'INH'</span> <span class="op">&amp;</span> <span class="va">msex</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">rownames</span>
<span class="va">group2</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cell_type</span> <span class="op">==</span> <span class="st">'INH'</span> <span class="op">&amp;</span> <span class="va">msex</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">rownames</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">group1</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "TCTTCGGCAAGACGTG-11" "ATTATCCGTTGATTCG-11" "CCACCTAGTCCAAGTT-11"
[4] "AGCGTATGTTAAGAAC-11" "GTAACTGTCCAGATCA-11" "CGATTGATCTTTACAC-11"</code></pre>
<p>Next, we run the <code>FindDMEs</code> function.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">DMEs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindDMEs.html">FindDMEs</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  barcodes1 <span class="op">=</span> <span class="va">group1</span>,
  barcodes2 <span class="op">=</span> <span class="va">group2</span>,
  test.use<span class="op">=</span><span class="st">'wilcox'</span>,
  wgcna_name<span class="op">=</span><span class="st">'INH'</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">DMEs</span><span class="op">)</span></code></pre></div>
<pre><code>p_val  avg_log2FC pct.1 pct.2    p_val_adj  module
INH-M18 4.714924e-23  0.40637874 0.894 0.729 8.486863e-22 INH-M18
INH-M16 5.257311e-08 -0.18257946 0.850 0.935 9.463160e-07 INH-M16
INH-M2  7.565615e-05 -0.18746938 0.661 0.738 1.361811e-03  INH-M2
INH-M15 1.496899e-03 -0.06603535 0.969 0.977 2.694417e-02 INH-M15
INH-M10 1.513458e-02 -0.07774661 0.975 0.980 2.724224e-01 INH-M10
INH-M17 2.034644e-02  0.24035946 0.589 0.557 3.662360e-01 INH-M17</code></pre>
<p>We can visualize the results using the hdWGNCA functions <code>PlotDMEsLollipop</code> or <code>PlotDMEsVolcano</code>. First we make a lollipop plot to visualize the DME results.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PlotDMEsLollipop.html">PlotDMEsLollipop</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>, 
  <span class="va">DMEs</span>, 
  wgcna_name<span class="op">=</span><span class="st">'INH'</span>, 
  pvalue <span class="op">=</span> <span class="st">"p_val_adj"</span>
<span class="op">)</span></code></pre></div>
<p><img src="figures/DMEs/test_DME_lollipop.png" width="800" height="900"></p>
<p>This plot shows the fold-change for each of the modules, and the size of each dot corresponds to the number of genes in that module. An “X” is placed over each point that does not reach statistical significance.</p>
<p>For <code>PlotDMEsLollipop</code>, if we have other self-defined columns in the DMEs data frame, we can use <code>group.by</code> parameter to supply the column names and comparison arguments for one comparison group or a list of comparison groups for plotting. For example:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PlotDMEsLollipop.html">PlotDMEsLollipop</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>, 
  <span class="va">DMEs</span>, 
  wgcna_name<span class="op">=</span><span class="st">'INH'</span>, 
  group.by <span class="op">=</span> <span class="st">"Comparisons"</span>, 
  comparison <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"group1_vs_control"</span>, <span class="st">"group2_vs_control"</span><span class="op">)</span>,  
  pvalue <span class="op">=</span> <span class="st">"p_val_adj"</span>
<span class="op">)</span> </code></pre></div>
<p>Alternatively, we can use <code>PlotDMEsVolcano</code> to make a volcano plot to show the effect size and the significance level together.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PlotDMEsVolcano.html">PlotDMEsVolcano</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  <span class="va">DMEs</span>,
  wgcna_name <span class="op">=</span> <span class="st">'INH'</span>
<span class="op">)</span></code></pre></div>
<p><img src="figures/DMEs/test_DME_volcano.png" width="800" height="900"></p>
<div class="section level3">
<h3 id="looping-through-multiple-clusters">Looping through multiple clusters<a class="anchor" aria-label="anchor" href="#looping-through-multiple-clusters"></a>
</h3>
<p>We can run DME analysis in a loop to perform the comparison in multiple groups. Here we will loop through the 5 INH subclusters and run <code>FindDMEs</code> in each cluster.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># list of clusters to loop through</span>
<span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"INH1 VIP+"</span>, <span class="st">"INH2 PVALB+"</span>, <span class="st">"INH3 SST+"</span>, <span class="st">'INH4 LAMP5+'</span>, <span class="st">"INH5 PVALB+"</span><span class="op">)</span>

<span class="co"># set up an empty dataframe for the DMEs</span>
<span class="va">DMEs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># loop through the clusters</span>
<span class="kw">for</span><span class="op">(</span><span class="va">cur_cluster</span> <span class="kw">in</span> <span class="va">clusters</span><span class="op">)</span><span class="op">{</span>

  <span class="co"># identify barcodes for group1 and group2 in eadh cluster</span>
  <span class="va">group1</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">annotation</span> <span class="op">==</span> <span class="va">cur_cluster</span> <span class="op">&amp;</span> <span class="va">msex</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">rownames</span>
  <span class="va">group2</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">annotation</span> <span class="op">==</span> <span class="va">cur_cluster</span> <span class="op">&amp;</span> <span class="va">msex</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">rownames</span>

  <span class="co"># run the DME test</span>
  <span class="va">cur_DMEs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindDMEs.html">FindDMEs</a></span><span class="op">(</span>
    <span class="va">seurat_obj</span>,
    barcodes1 <span class="op">=</span> <span class="va">group1</span>,
    barcodes2 <span class="op">=</span> <span class="va">group2</span>,
    test.use<span class="op">=</span><span class="st">'wilcox'</span>,
    pseudocount.use<span class="op">=</span><span class="fl">0.01</span>, <span class="co"># we can also change the pseudocount with this param</span>
    wgcna_name <span class="op">=</span> <span class="st">'INH'</span>
  <span class="op">)</span>

  <span class="co"># add the cluster info to the table</span>
  <span class="va">cur_DMEs</span><span class="op">$</span><span class="va">cluster</span> <span class="op">&lt;-</span> <span class="va">cur_cluster</span>

  <span class="co"># append the table</span>
  <span class="va">DMEs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">DMEs</span>, <span class="va">cur_DMEs</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>Here we will use ggplot2 to make a heatmap showing the DME effect sizes. Since we are just using ggplot2 here you can easily customize this heatmap to suit your needs.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get the modules table:</span>
<span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="va">mods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">modules</span><span class="op">$</span><span class="va">module</span><span class="op">)</span>; <span class="va">mods</span> <span class="op">&lt;-</span> <span class="va">mods</span><span class="op">[</span><span class="va">mods</span> <span class="op">!=</span> <span class="st">'grey'</span><span class="op">]</span>

<span class="co"># make a copy of the DME table for plotting</span>
<span class="va">plot_df</span> <span class="op">&lt;-</span> <span class="va">DMEs</span>

<span class="co"># set the factor level for the modules so they plot in the right order:</span>
<span class="va">plot_df</span><span class="op">$</span><span class="va">module</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">plot_df</span><span class="op">$</span><span class="va">module</span><span class="op">)</span>, levels<span class="op">=</span><span class="va">mods</span><span class="op">)</span>

<span class="co"># set a min/max threshold for plotting</span>
<span class="va">maxval</span> <span class="op">&lt;-</span> <span class="fl">0.5</span>; <span class="va">minval</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fl">0.5</span>
<span class="va">plot_df</span><span class="op">$</span><span class="va">avg_log2FC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">plot_df</span><span class="op">$</span><span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="va">maxval</span>, <span class="va">maxval</span>, <span class="va">plot_df</span><span class="op">$</span><span class="va">avg_log2FC</span><span class="op">)</span>
<span class="va">plot_df</span><span class="op">$</span><span class="va">avg_log2FC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">plot_df</span><span class="op">$</span><span class="va">avg_log2FC</span> <span class="op">&lt;</span> <span class="va">minval</span>, <span class="va">minval</span>, <span class="va">plot_df</span><span class="op">$</span><span class="va">avg_log2FC</span><span class="op">)</span>

<span class="co"># add significance levels</span>
<span class="va">plot_df</span><span class="op">$</span><span class="va">Significance</span> <span class="op">&lt;-</span> <span class="fu">gtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gtools/man/stars.pval.html" class="external-link">stars.pval</a></span><span class="op">(</span><span class="va">plot_df</span><span class="op">$</span><span class="va">p_val_adj</span><span class="op">)</span>

<span class="co"># change the text color to make it easier to see </span>
<span class="va">plot_df</span><span class="op">$</span><span class="va">textcolor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">plot_df</span><span class="op">$</span><span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">0.2</span>, <span class="st">'black'</span>, <span class="st">'white'</span><span class="op">)</span>

<span class="co"># make the heatmap with geom_tile</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="va">plot_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">cluster</span>, x<span class="op">=</span><span class="va">module</span>, fill<span class="op">=</span><span class="va">avg_log2FC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span><span class="op">)</span> 

<span class="co"># add the significance levels</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">plot_df</span><span class="op">$</span><span class="va">Significance</span>, color<span class="op">=</span><span class="va">plot_df</span><span class="op">$</span><span class="va">textcolor</span><span class="op">)</span> 

<span class="co"># customize the color and theme of the plot</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient2</a></span><span class="op">(</span>low<span class="op">=</span><span class="st">'purple'</span>, mid<span class="op">=</span><span class="st">'black'</span>, high<span class="op">=</span><span class="st">'yellow'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">RotatedAxis</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
    panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>fill<span class="op">=</span><span class="cn">NA</span>, color<span class="op">=</span><span class="st">'black'</span>, size<span class="op">=</span><span class="fl">1</span><span class="op">)</span>,
    axis.line.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
    axis.line.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
    plot.margin<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_equal</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">p</span></code></pre></div>
<p><img src="figures/DMEs/test_DME_heatmap.png" width="800" height="400"></p>
</div>
</div>
<div class="section level2">
<h2 id="one-versus-all-dme-analysis">One-versus-all DME analysis<a class="anchor" aria-label="anchor" href="#one-versus-all-dme-analysis"></a>
</h2>
<p>Similar to the Seurat function <code>FindAllMarkers</code>, we can perform a one-versus-all DME test using the function <code>FindAllDMEs</code> when specifying a column to group cells. Here we will <code>group.by</code> each cell type for the one-versus-all test.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">group.by</span> <span class="op">=</span> <span class="st">'cell_type'</span>

<span class="va">DMEs_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindAllDMEs.html">FindAllDMEs</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by <span class="op">=</span> <span class="st">'cell_type'</span>,
  wgcna_name <span class="op">=</span> <span class="st">'INH'</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">DMEs_all</span><span class="op">)</span></code></pre></div>
<p>The output looks similar to <code>FindDMEs</code>, but there is an extra column called <code>group</code> containing the information for each cell grouping.</p>
<pre><code>p_val avg_log2FC pct.1 pct.2 p_val_adj module group
EX.INH-M1     0 -10.313763 0.003 0.621         0 INH-M1    EX
EX.INH-M2     0   1.007115 0.670 0.445         0 INH-M2    EX
EX.INH-M3     0   2.641423 0.729 0.205         0 INH-M3    EX
EX.INH-M4     0   1.978553 0.910 0.246         0 INH-M4    EX
EX.INH-M5     0   2.447565 0.865 0.185         0 INH-M5    EX
EX.INH-M6     0   2.448315 0.877 0.209         0 INH-M6    EX</code></pre>
<p>Now we can plot the results with <code>PlotDMEsVolcano</code></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotDMEsVolcano.html">PlotDMEsVolcano</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  <span class="va">DMEs_all</span>,
  wgcna_name <span class="op">=</span> <span class="st">'INH'</span>,
  plot_labels<span class="op">=</span><span class="cn">FALSE</span>,
  show_cutoff<span class="op">=</span><span class="cn">FALSE</span>
<span class="op">)</span>

<span class="co"># facet wrap by each cell type</span>
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">group</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span></code></pre></div>
<p><img src="figures/DMEs/test_DME_volcano2.png" width="800" height="900"></p>
</div>
<div class="section level2">
<h2 id="using-modulescores-instead-of-mes">Using ModuleScores instead of MEs<a class="anchor" aria-label="anchor" href="#using-modulescores-instead-of-mes"></a>
</h2>
<p>We can also perform differential analysis using module expression scores instead of module eigengenes. These must first be calculated using the function <code>ModuleExprScore</code>, as shown below.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleExprScore.html">ModuleExprScore</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  n_genes <span class="op">=</span> <span class="fl">25</span>,
  method<span class="op">=</span><span class="st">'UCell'</span>
<span class="op">)</span></code></pre></div>
<p>Next we can perform differential analysis using these module scores by specifying <code>features = 'ModuleScores'</code> in the <code>FindDMEs</code> function.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">group1</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cell_type</span> <span class="op">==</span> <span class="st">'INH'</span> <span class="op">&amp;</span> <span class="va">msex</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">rownames</span>
<span class="va">group2</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cell_type</span> <span class="op">==</span> <span class="st">'INH'</span> <span class="op">&amp;</span> <span class="va">msex</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">rownames</span>

<span class="va">DMEs_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindDMEs.html">FindDMEs</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  features <span class="op">=</span> <span class="st">'ModuleScores'</span>,
  barcodes1 <span class="op">=</span> <span class="va">group1</span>,
  barcodes2 <span class="op">=</span> <span class="va">group2</span>,
  test.use<span class="op">=</span><span class="st">'wilcox'</span>,
  wgcna_name<span class="op">=</span><span class="st">'tutorial'</span>
<span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotDMEsLollipop.html">PlotDMEsLollipop</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>, 
  <span class="va">DMEs</span>, 
  wgcna_name <span class="op">=</span> <span class="st">'tutorial'</span>,
  pvalue <span class="op">=</span> <span class="st">"p_val_adj"</span>
<span class="op">)</span>

<span class="va">p</span></code></pre></div>
<p><img src="figures/DMEs/test_DMEscore_lollipop.png" width="800" height="900"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
