<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Enrichment analysis • hdWGCNA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Enrichment analysis">
<meta name="description" content="Tutorial for performing enrichment analysis with hdWGCNA.
">
<meta property="og:description" content="Tutorial for performing enrichment analysis with hdWGCNA.
">
<meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Enrichment analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/vignettes/enrichment_analysis.Rmd" class="external-link"><code>vignettes/enrichment_analysis.Rmd</code></a></small>
      <div class="d-none name"><code>enrichment_analysis.Rmd</code></div>
    </div>

    
    
<p>In this tutorial, we will perform enrichment tests on our hdWGCNA modules. We leverage the R pacakge <a href="https://maayanlab.cloud/Enrichr/" class="external-link"><code>enrichR</code></a> to perform enrichment tests on a wide range of curated gene lists. This analysis should point us towards biological processes that our hdWGCNA modules are involved in. Additionally, we perform a gene set overlap analysis to compare the genes in hdWGCNA modules with the marker genes identified using Seurat’s <a href="https://satijalab.org/seurat/reference/findallmarkers" class="external-link">FindAllMarkers</a> function.</p>
<p>First, we first need to load the data and the required libraries.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># single-cell analysis package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plotting and data science packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># co-expression network analysis packages:</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/" class="external-link">WGCNA</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://smorabit.github.io/hdWGCNA/">hdWGCNA</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># gene enrichment packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">enrichR</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://shenlab-sinai.github.io/shenlab-sinai/" class="external-link">GeneOverlap</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># using the cowplot theme for ggplot</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># set random seed for reproducibility</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load the Zhou et al snRNA-seq dataset</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'data/Zhou_control.rds'</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="enrichr">EnrichR<a class="anchor" aria-label="anchor" href="#enrichr"></a>
</h2>
<p>In this section we discuss how to perform Enrichr enrichment tests and how to visualize the results using hdWGCNA. hdWGCNA includes the function <code>RunEnrichr</code> to compare the set of genes in each module with any of the gene lists hosted by Enrichr. You can take a look at the list of different Enrichr gene lists <a href="https://maayanlab.cloud/Enrichr/#libraries" class="external-link">here</a>. We store the results of the enrichment tests in the hdWGCNA experiment, so can be easily retrieved for downstream analysis or exporting to external applicaitons like Excel. In the following example, we perform the enrichment test with three Gene Ontology datbases:</p>
<ul>
<li>GO_Biological_Process_2021</li>
<li>GO_Cellular_Component_2021</li>
<li>GO_Molecular_Function_2021</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># enrichr databases to test</span></span>
<span><span class="va">dbs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'GO_Biological_Process_2021'</span>,<span class="st">'GO_Cellular_Component_2021'</span>,<span class="st">'GO_Molecular_Function_2021'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># perform enrichment tests</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunEnrichr.html">RunEnrichr</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  dbs<span class="op">=</span><span class="va">dbs</span>, <span class="co"># character vector of enrichr databases to test</span></span>
<span>  max_genes <span class="op">=</span> <span class="fl">100</span> <span class="co"># number of genes per module to test. use max_genes = Inf to choose all genes!</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># retrieve the output table</span></span>
<span><span class="va">enrich_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetEnrichrTable.html">GetEnrichrTable</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span></span></code></pre></div>
<p>The outputs of the Enrichr test are explained in detail here, but we offer a short explanation below for the different columns in <code>enrich_df</code>:</p>
<ul>
<li>
<code>Term</code>: The name of the term (ie biological process, etc).</li>
<li>
<code>Overlap</code>: The fraction of genes overlapping between the module and the gene list.</li>
<li>
<code>P.value</code>: Fisher’s exact test p-value.</li>
<li>
<code>Adjusted.P.value</code>: Benjamini-Hochberg multiple testing correction for the Fisher’s exact test p-values.</li>
<li>
<code>Odds.Ratio</code>: statistic to quantify the association between the gene list in the current module and the gene list for the current Term.</li>
<li>
<code>Combined.Score</code>: natural log of the p-value multiplied by the z-score, where the z-score is the deviation from the expected rank.</li>
<li>
<code>Genes</code>: semicolon delimited list of gene symbols for the overlapping genes.</li>
<li>
<code>db</code>: the name of the Enrichr gene list.</li>
<li>
<code>module</code>: the name of the hdWGCNA module.</li>
</ul>
<div class="section level3">
<h3 id="visualize-enrichments">Visualize enrichments<a class="anchor" aria-label="anchor" href="#visualize-enrichments"></a>
</h3>
<p>Now that we have done the enrichment tests, there are several ways we can go about visualizing the results.</p>
<div class="section level4">
<h4 id="enrichrbarplot">
<code>EnrichrBarPlot</code><a class="anchor" aria-label="anchor" href="#enrichrbarplot"></a>
</h4>
<p>hdWGCNA includes the function <code>EnrichrBarPlot</code> to summarize the results of every Enrichr database and every module. This function outputs a .pdf figure for each module, containing a barplot showing the top <em>N</em> enriched terms. The following example will plot the top 10 terms in each module and will output the results to a folder called <code>enrichr_plots</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># make GO term plots:</span></span>
<span><span class="fu"><a href="../reference/EnrichrBarPlot.html">EnrichrBarPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  outdir <span class="op">=</span> <span class="st">"enrichr_plots"</span>, <span class="co"># name of output directory</span></span>
<span>  n_terms <span class="op">=</span> <span class="fl">10</span>, <span class="co"># number of enriched terms to show (sometimes more show if there are ties!!!)</span></span>
<span>  plot_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">7</span><span class="op">)</span>, <span class="co"># width, height of the output .pdfs</span></span>
<span>  logscale<span class="op">=</span><span class="cn">TRUE</span> <span class="co"># do you want to show the enrichment as a log scale?</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The following bar plot is a single example of the <code>EnrichrBarPlot</code> output:</p>
<p><img src="figures/enrichment/inh-m7_example.png" width="500" height="500"></p>
<p><strong><em>Interpreting Enrichr results</em></strong></p>
<p>Each of the enrichment bar plots are colored by the module’s unique color, and each term is sorted by the enrichment (combined score). We encourage users to carefully inspect the results of the enrichment tests, and use prior biological knowledge before jumping to conclusions. In this example, we see some terms that make sense for inhibitory neurons, such as “inhibitory synapse assembly” and “synaptic transmission, GABAergic”. On the other hand, we see several cardiac related terms that are realistically not at all related to our system in this example (human brain). Many genes take part in distinct biological processes in different tissues within the same organism, which leads to enrichment results like this.</p>
</div>
<div class="section level4">
<h4 id="enrichrdotplot">
<code>EnrichrDotPlot</code><a class="anchor" aria-label="anchor" href="#enrichrdotplot"></a>
</h4>
<p>hdWGCNA includes an additional visualization function for enrichment results, <code>EnrichrDotPlot</code>, which shows the top results for one Enrichr database in each module. In the following example, we plot the top term in the <code>GO_Biological_Process_2021</code> database.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># enrichr dotplot</span></span>
<span><span class="fu"><a href="../reference/EnrichrDotPlot.html">EnrichrDotPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  mods <span class="op">=</span> <span class="st">"all"</span>, <span class="co"># use all modules (this is the default behavior)</span></span>
<span>  database <span class="op">=</span> <span class="st">"GO_Biological_Process_2021"</span>, <span class="co"># this has to be one of the lists we used above!!!</span></span>
<span>  n_terms<span class="op">=</span><span class="fl">1</span> <span class="co"># number of terms for each module</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/enrichment/GO_dotplot.png" width="500" height="500"></p>
<p>In this plot, each dot is colored by the module’s unique color, and the size of each dot is scaled by the enrichment of the term.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="marker-gene-overlap-analysis">Marker gene overlap analysis<a class="anchor" aria-label="anchor" href="#marker-gene-overlap-analysis"></a>
</h2>
<p>In this section we discuss how to compare hdWGCNA modules with cluster or cell-type marker genes. First we use the Seurat function <code>FindAllMarkers</code> to identify the marker genes in each cell type, and then we use the hdWGCNA function <code>OverlapModulesDEGs</code> to overlap the modules and the DEGs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># compute cell-type marker genes with Seurat:</span></span>
<span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">$</span><span class="va">cell_type</span></span>
<span><span class="va">markers</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  only.pos <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  logfc.threshold<span class="op">=</span><span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># compute marker gene overlaps</span></span>
<span><span class="va">overlap_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OverlapModulesDEGs.html">OverlapModulesDEGs</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  deg_df <span class="op">=</span> <span class="va">markers</span>,</span>
<span>  fc_cutoff <span class="op">=</span> <span class="fl">1</span> <span class="co"># log fold change cutoff for overlap analysis</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Notably, the results from <code>OverlapModulesDEGs</code> are currently not stored in the hdWGCNA experiment. This is partially because this test is extremely fast to run, and partially because there are many different sets of DEGs that one could compare their modules to. The <code>overlap_df</code> contains the following columns:</p>
<ul>
<li>
<code>module</code>: the name of the hdWGCNA module.</li>
<li>
<code>group</code>: the name of the RNA-seq cluster / cell-type / etc.</li>
<li>
<code>color</code>: the module’s unique color.</li>
<li>
<code>odds_ratio</code>: statistic to quantify the association between the gene list in the current module and the gene list for the marker genes.</li>
<li>
<code>pval</code>: Fisher’s exact test p-value.</li>
<li>
<code>fdr</code>: False Discovery Rate (FDR) multiple testing correction for the Fisher’s p-values.</li>
<li>
<code>Significance</code>: Character vector indicating the significance level of the overlap.</li>
<li>
<code>Jaccard</code>: Jaccard index for the two gene lists.</li>
<li>
<code>size_intersection</code>: the number of genes that overlap between the two gene lists.</li>
</ul>
<div class="section level3">
<h3 id="visualize-overlaps">Visualize overlaps<a class="anchor" aria-label="anchor" href="#visualize-overlaps"></a>
</h3>
<p>hdWGCNA includes two functions to visualize the results from <code>OverlapModulesDEGs</code>, <code>OverlapDotPlot</code> and <code>OverlapBarPlot</code>.</p>
<p>Here we demonstrate using <code>OverlapBarPlot</code> to visualize the results of <code>OverlapModulesDEGs</code>. This function creates a bar plot for each cell type showing one of the overlap statistics for each module.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># overlap barplot, produces a plot for each cell type</span></span>
<span><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/OverlapBarPlot.html">OverlapBarPlot</a></span><span class="op">(</span><span class="va">overlap_df</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># stitch plots with patchwork</span></span>
<span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">plot_list</span>, ncol<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/enrichment/overlap_barplot.png" width="500" height="500"></p>
<p>Next we use <code>OverlapDotPlot</code> to visualize the overlap results in a single plot.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot odds ratio of the overlap as a dot plot</span></span>
<span><span class="fu"><a href="../reference/OverlapDotPlot.html">OverlapDotPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">overlap_df</span>,</span>
<span>  plot_var <span class="op">=</span> <span class="st">'odds_ratio'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">'Overlap of modules &amp; cell-type markers'</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/enrichment/overlap_dotplot.png" width="600" height="600"></p>
<p>Each dot is colored by the hdWGCNA module’s unique color, and the size of the dot is scaled by the overlap statistic. We show FDR significance levels as stars on top of the dots.</p>
<ul>
<li>’***’: 0 - 0.001</li>
<li>’**’: 0.001 - 0.01</li>
<li>’*’: 0.01 - 0.05</li>
<li>‘+’: 0.05 - 0.1</li>
<li>(No symbol): 0.1 - 1.0</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
