<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Vignettes overview â€¢ hdWGCNA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Vignettes overview">
<meta name="description" content="Document showing an overview of the different analysis vignettes.
">
<meta property="og:description" content="Document showing an overview of the different analysis vignettes.
">
<meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Vignettes overview</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/vignettes/hdWGCNA.Rmd" class="external-link"><code>vignettes/hdWGCNA.Rmd</code></a></small>
      <div class="d-none name"><code>hdWGCNA.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="co-expression-network-analysis">Co-expression network analysis<a class="anchor" aria-label="anchor" href="#co-expression-network-analysis"></a>
</h2>
<p>These tutorials cover the essentials of performing co-expression network analysis in single-cell transcriptomics data, and visualizing the key results.</p>
<div class="section level3">
<h3 id="hdwgcna-in-single-cell-data">
<a href="basic_tutorial.html">hdWGCNA in single-cell data</a><a class="anchor" aria-label="anchor" href="#hdwgcna-in-single-cell-data"></a>
</h3>
<p>This tutorial covers the essential functions to construct a co-expression network in single-cell transcriptomics data with hdWGCNA.</p>
<p><img src="figures/basic_tutorial/Zhou_featureplot_hMEs_selected_wide.png" width="500" height="500"></p>
</div>
<div class="section level3">
<h3 id="hdwgcna-in-spatial-transcriptomics-data">
<a href="ST_basics.html">hdWGCNA in spatial transcriptomics data</a><a class="anchor" aria-label="anchor" href="#hdwgcna-in-spatial-transcriptomics-data"></a>
</h3>
<p>This tutorial covers the essential functions to construct a co-expression network in spatial transcriptomics data with hdWGCNA.</p>
<p><img src="figures/ST_basics/spatial_clusters.png" width="500" height="500"></p>
</div>
<div class="section level3">
<h3 id="network-visualization">
<a href="network_visualizations.html">Network visualization</a><a class="anchor" aria-label="anchor" href="#network-visualization"></a>
</h3>
<p>This tutorial highlights several approaches for visualizing the hdWGCNA co-expression networks.</p>
<p><img src="figures/network_vis/hubgene_umap_igraph.png" width="400" height="400"></p>
</div>
</div>
<div class="section level2">
<h2 id="biological-context-for-co-expression-modules">Biological context for co-expression modules<a class="anchor" aria-label="anchor" href="#biological-context-for-co-expression-modules"></a>
</h2>
<p>These tutorials will provide further biological context for our co-expression modules, potentially revealing what experimental conditions and biological processes that these modules are involved in.</p>
<div class="section level3">
<h3 id="differential-module-eigengene-dme-analysis">
<a href="differential_MEs.html">Differential module eigengene (DME) analysis</a><a class="anchor" aria-label="anchor" href="#differential-module-eigengene-dme-analysis"></a>
</h3>
<p>This tutorial covers how to compare module eigengenes between experimental groups.</p>
<p><img src="figures/DMEs/test_DME_volcano.png" width="400" height="400" href="differential_MEs.html"></p>
</div>
<div class="section level3">
<h3 id="module-trait-correlation">
<a href="module_trait_correlation.html">Module trait correlation</a><a class="anchor" aria-label="anchor" href="#module-trait-correlation"></a>
</h3>
<p>This tutorial covers how to correlate continuous and categorical variables with module eigengenes or module expression scores, revealing which modules are related to different experimental conditions or covariates.</p>
<p><img src="figures/mt_correlation/ME_Trait_correlation_fdr.png" width="400" height="400" href="module_trait_correlation.html"></p>
</div>
<div class="section level3">
<h3 id="enrichment-analysis">
<a href="enrichment_analysis.html">Enrichment analysis</a><a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h3>
<p>This tutorial shows how to use Enrichr to compare the gene members of each co-expression module to curated gene lists, thereby pointing towards the biological functions of the co-expression modules.</p>
<p><img src="figures/enrichment/GO_dotplot.png" width="400" height="400"></p>
</div>
</div>
<div class="section level2">
<h2 id="exploring-modules-in-external-datasets">Exploring modules in external datasets<a class="anchor" aria-label="anchor" href="#exploring-modules-in-external-datasets"></a>
</h2>
<div class="section level3">
<h3 id="projecting-modules-to-new-datasets">
<a href="projecting_modules.html">Projecting modules to new datasets</a><a class="anchor" aria-label="anchor" href="#projecting-modules-to-new-datasets"></a>
</h3>
<p>This tutorial covers how to project co-expression modules from a reference to a query dataset.</p>
<p><img src="figures/projection/compare_umaps.png" width="500" height="500"></p>
</div>
<div class="section level3">
<h3 id="module-preservation-and-reproducibility">
<a href="module_preservation.html">Module preservation and reproducibility</a><a class="anchor" aria-label="anchor" href="#module-preservation-and-reproducibility"></a>
</h3>
<p>This tutorial covers statistical methods for assessing the preservation and reproducibility of co-expression networks using external datasets.</p>
<p><img src="figures/projection/module_preservation_summary.png" width="500" height="500"></p>
</div>
<div class="section level3">
<h3 id="cross-species-and-cross-modality-analysis">
<a href="projecting_modules_cross.html">Cross-species and cross-modality analysis</a><a class="anchor" aria-label="anchor" href="#cross-species-and-cross-modality-analysis"></a>
</h3>
<p>This tutorial covers how to project co-expression modules from a reference to a query dataset for special cases where the data modality or the species do not match between the reference and the query.</p>
<p><img src="figures/projection/atac_umap_covplot.png" width="700" height="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="advanced-topics">Advanced topics<a class="anchor" aria-label="anchor" href="#advanced-topics"></a>
</h2>
<div class="section level3">
<h3 id="consensus-network-analysis">
<a href="consensus_wgcna.html">Consensus network analysis</a><a class="anchor" aria-label="anchor" href="#consensus-network-analysis"></a>
</h3>
</div>
<div class="section level3">
<h3 id="motif-analysis">
<a href="motif_analysis.html">Motif analysis</a><a class="anchor" aria-label="anchor" href="#motif-analysis"></a>
</h3>
</div>
</div>
<div class="section level2">
<h2 id="other">Other<a class="anchor" aria-label="anchor" href="#other"></a>
</h2>
<div class="section level3">
<h3 id="module-customization">
<a href="customization.html">Module customization</a><a class="anchor" aria-label="anchor" href="#module-customization"></a>
</h3>
<p>This tutorial covers how to change the default names and colors for hdWGCNA modules.</p>
<p><img src="figures/customization/featureplot.png" width="400" height="400"></p>
</div>
<div class="section level3">
<h3 id="using-sctransform-normalized-data">
<a href="sctransform.html">Using SCTransform normalized data</a><a class="anchor" aria-label="anchor" href="#using-sctransform-normalized-data"></a>
</h3>
<p>This tutorial covers how to use SCTransform normalized data in hdWGCNA.</p>
<p><img src="figures/sctransform/dendro_SCT_compare.png" width="400" height="400" href="sctransform.html"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
