<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Articles â€¢ hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Articles"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-article-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Articles</h1>
    </div>

    <div class="section ">
      <h3>All vignettes</h3>
      <div class="section-desc"></div>

      <dl><dt><a href="basic_tutorial.html">hdWGCNA in single-cell data</a></dt>
        <dd><p>Tutorial for applying the core functions of hdWGCNA in single-cell data.</p>
        </dd><dt><a href="consensus_wgcna.html">Consensus network analysis</a></dt>
        <dd><p>Tutorial for performing consensus co-expression network analysis.</p>
        </dd><dt><a href="customization.html">Module customization</a></dt>
        <dd><p>Tutorial for customizing module colors and module names in hdWGCNA.</p>
        </dd><dt><a href="differential_MEs.html">Differential module eigengene (DME) analysis</a></dt>
        <dd><p>Tutorial for performing differential module eigengene analysis.</p>
        </dd><dt><a href="enrichment_analysis.html">Enrichment analysis</a></dt>
        <dd><p>Tutorial for performing enrichment analysis with hdWGCNA.</p>
        </dd><dt><a href="hdWGCNA.html">Vignettes overview</a></dt>
        <dd><p>Document showing an overview of the different analysis vignettes.</p>
        </dd><dt><a href="isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></dt>
        <dd><p>Full Seurat clustering and hdWGCNA analysis tutorial of PacBio MAS-Seq PBMC dataset.</p>
        </dd><dt><a href="module_preservation.html">Module preservation and reproducibility</a></dt>
        <dd><p>Tutorial for performing module preservation tests in hdWGCNA.</p>
        </dd><dt><a href="module_trait_correlation.html">Module Trait Correlation</a></dt>
        <dd><p>Tutorial for identifying relationships between hdWGCNA modules with biological and technical variables.</p>
        </dd><dt><a href="motif_analysis.html">Motif Analysis</a></dt>
        <dd><p>Tutorial for identifying transcription factor binding motifs that are related to hdWGCNA modules.</p>
        </dd><dt><a href="network_visualizations.html">Network Visualization</a></dt>
        <dd><p>Tutorial for visualizing the hdWGCNA networks using igraph.</p>
        </dd><dt><a href="other_metacells.html">Alternative metacell algorithms</a></dt>
        <dd><p>Tutorial for performing co-expression network analysis using alternative metacell aggregation algorithms.</p>
        </dd><dt><a href="projecting_modules_cross.html">Cross-species and cross-modality analysis</a></dt>
        <dd><p>Projecting hdWGCNA co-expression modules across species and -omics modalities.</p>
        </dd><dt><a href="projecting_modules.html">Projecting modules to new datasets</a></dt>
        <dd><p>Tutorial for applying the core functions of hdWGCNA.</p>
        </dd><dt><a href="pseudobulk.html">hdWGCNA with pseudobulk data</a></dt>
        <dd><p>Tutorial for running hdWGCNA with pseudobulk data rather than metacells/metaspots .</p>
        </dd><dt><a href="pseudotime.html">Co-expression module dynamics with pseudotime</a></dt>
        <dd><p>Full Seurat clustering and hdWGCNA analysis tutorial of PacBio MAS-Seq PBMC dataset.</p>
        </dd><dt><a href="sctransform.html">Using SCTransform normalized data</a></dt>
        <dd><p>Tutorial demonstrating hdWGCNA with SCTransform normalized data.</p>
        </dd><dt><a href="ST_basics.html">hdWGCNA in spatial transcriptomics</a></dt>
        <dd><p>Tutorial for applying the core functions of hdWGCNA in spatial transcriptomics data.</p>
        </dd><dt><a href="tf_network.html">Transcription factor regulatory network analysis</a></dt>
        <dd><p>Tutorial for using hdWGCNA for transcription factor regulatory network analysis.</p>
      </dd></dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

