<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Isoform co-expression network analysis with PacBio MAS-Seq • hdWGCNA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Isoform co-expression network analysis with PacBio MAS-Seq">
<meta property="og:description" content="Full Seurat clustering and hdWGCNA analysis tutorial of PacBio MAS-Seq PBMC dataset.
">
<meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hdWGCNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.03</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/hdWGCNA.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Core functionality</li>
    <li>
      <a href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a>
    </li>
    <li>
      <a href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a>
    </li>
    <li>
      <a href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a>
    </li>
    <li>
      <a href="../articles/network_visualizations.html">Network visualization</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Biological context for co-expression modules</li>
    <li>
      <a href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a>
    </li>
    <li>
      <a href="../articles/module_trait_correlation.html">Module trait correlation</a>
    </li>
    <li>
      <a href="../articles/enrichment_analysis.html">Enrichment analysis</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Exploring modules in external datasets</li>
    <li>
      <a href="../articles/projecting_modules.html">Projecting modules to new datasets</a>
    </li>
    <li>
      <a href="../articles/module_preservation.html">Module preservation and reproducibility</a>
    </li>
    <li>
      <a href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Advanced topics</li>
    <li>
      <a href="../articles/consensus_wgcna.html">Consensus network analysis</a>
    </li>
    <li>
      <a href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a>
    </li>
    <li>
      <a href="../articles/motif_analysis.html">Motif analysis</a>
    </li>
    <li>
      <a href="../articles/other_metacells.html">Alternative metacell algorithms</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Other</li>
    <li>
      <a href="../articles/customization.html">Module customization</a>
    </li>
    <li>
      <a href="../articles/sctransform.html">Using SCTransform normalized data</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Isoform co-expression network analysis with PacBio MAS-Seq</h1>
            
      
      
      <div class="hidden name"><code>isoform_pbmcs.Rmd</code></div>

    </div>

    
    
<p>This tutorial covers the basics of using hdWGCNA to perform <strong>isoform</strong> co-expression network analysis using <a href="https://www.pacb.com/wp-content/uploads/Application-note-MAS-Seq-for-single-cell-isoform-sequencing.pdf" class="external-link">PacBio MAS-Seq</a> data. Briefly, MAS-Seq is a single-cell RNA-seq protocol that uses long-read sequencing to profile full-length RNA molecules, potentially providing a much greater depth of information about individual RNA splicing isoforms compared to traditional short-read scRNA-seq. This tutorial starts from the typical genes by cells counts matrix, as well as an isoforms by cells counts matrix unique to long-read single-cell approaches like MAS-seq, and we cover how to perform clustering analysis in long-read single-cell data using Seurat. Finally, we demonstrate several different ways to use hdWGCNA for isoform co-expression network analysis. Here we use a MAS-Seq dataset of PBMCs from two donors, sequenced using the PacBio Revio platform. Before working through this tutorial, we recommend becoming familar with the basics of hdWGCNA using our single-cell tutorial.</p>
<div class="section level2">
<h2 id="download-the-tutorial-data">Download the tutorial data<a class="anchor" aria-label="anchor" href="#download-the-tutorial-data"></a>
</h2>
<p>Here we download the genes and isoforms counts matrices from PacBio for the two replicates using wget. These counts matrices were generated using the PacBio SMRT Link software, more details can be found at the <a href="https://isoseq.how/umi/" class="external-link">isoseq documentation website</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># download replicate 1</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> --recursive --no-parent https://downloads.pacbcloud.com/public/dataset/MAS-Seq/DATA-Revio-PBMC-1/4-SeuratMatrix/</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># download replicate 2</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> --recursive --no-parent https://downloads.pacbcloud.com/public/dataset/MAS-Seq/DATA-Revio-PBMC-2/4-SeuratMatrix/</span></code></pre></div>
<p>You can also <a href="https://drive.google.com/file/d/1pIINwgMlJX-3db4sLApt5Gi7GkWQElGf/view?usp=share_link" class="external-link">download the unprocessed Seurat object containing both replicates</a> instead of running the “Create a multi-assay Seurat object” section.</p>
</div>
<div class="section level2">
<h2 id="seurat-clustering-analysis">Seurat clustering analysis<a class="anchor" aria-label="anchor" href="#seurat-clustering-analysis"></a>
</h2>
<p>In this section, we will perform clustering analysis on MAS-Seq dataset using Seurat. In principal, this analysis could also be performed using alternative pipelines such as Scanpy before performing network analysis. We perform the standard clustering steps including quality control (QC) filtering, normalization, feature selection, linear dimensionality reduction (PCA), batch correction (harmony), non-linear dimensionality reduction (UMAP), and louvain clustering. Here we perform the clustering analysis using the gene counts matrix and the isoform counts matrix separately to compare the differences.</p>
<div class="section level3">
<h3 id="create-a-multi-assay-seurat-object">Create a multi-assay Seurat object<a class="anchor" aria-label="anchor" href="#create-a-multi-assay-seurat-object"></a>
</h3>
<p>The MAS-Seq dataset includes both gene and isoform level quantifications, and in this section we will load these matrices for both replicates to create a multi-assay Seurat object. The number of isoforms detected differs for each replicate, and here we have chosen to only include isoforms that were detected in both replicates. If you don’t want to run this section, you can <a href="https://drive.google.com/file/d/1pIINwgMlJX-3db4sLApt5Gi7GkWQElGf/view?usp=share_link" class="external-link">download the unprocessed Seurat object here</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">harmony</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org/" class="external-link">Matrix</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span>

<span class="co"># directories with the counts matrices for each replicate</span>
<span class="va">sample_dirs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
  <span class="st">'DATA-Revio-PBMC-1/4-SeuratMatrix/NoNovelGenesIsoforms/'</span>,
  <span class="st">'DATA-Revio-PBMC-2/4-SeuratMatrix/NoNovelGenesIsoforms/'</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sample_dirs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'PBMC-1'</span>, <span class="st">'PBMC-2'</span><span class="op">)</span>

<span class="co"># load isoform counts matrices for each sample:</span>
<span class="va">iso_X_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">sample_dirs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html" class="external-link">Read10X</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">'isoforms_seurat/'</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>

<span class="co"># load gene counts matrices for each sample:</span>
<span class="va">gene_X_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">sample_dirs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html" class="external-link">Read10X</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">'genes_seurat/'</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>

<span class="co"># only keep isoforms that are common in all samples:</span>
<span class="va">common_iso</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">intersect</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">iso_X_list</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span>
<span class="va">iso_X_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">iso_X_list</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">[</span><span class="va">common_iso</span>,<span class="op">]</span><span class="op">}</span><span class="op">)</span>

<span class="co"># create individual Seurat objects for each sample</span>
<span class="va">seurat_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sample_dirs</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span><span class="op">{</span>
  <span class="va">cur</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span><span class="va">gene_X_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, min.cells<span class="op">=</span><span class="fl">5</span><span class="op">)</span>;

  <span class="co"># add a column for the original barcode</span>
  <span class="va">cur</span><span class="op">$</span><span class="va">barcode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">cur</span><span class="op">)</span>

  <span class="co"># add a column indicating the sample name</span>
  <span class="va">cur</span><span class="op">$</span><span class="va">Sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sample_dirs</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>

  <span class="co"># add the iso assay to the seurat object to contain the isoform counts matrix</span>
  <span class="va">cur</span><span class="op">[[</span><span class="st">"iso"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">iso_X_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, min.cells<span class="op">=</span><span class="fl">5</span><span class="op">)</span>
  <span class="va">cur</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># merge replicates into one Seurat object</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">merge</span>, <span class="va">seurat_list</span><span class="op">)</span>

<span class="co"># save the unprocessed Seurat object</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">seurat_obj</span>, file <span class="op">=</span> <span class="st">'PBMC_masseq_unprocessed_seurat.rds'</span><span class="op">)</span>

<span class="co"># check basic info about this seurat object</span>
<span class="va">seurat_obj</span></code></pre></div>
<pre><code>An object of class Seurat 
225080 features across 17637 samples within 2 assays 
Active assay: RNA (19877 features, 0 variable features)
 1 other assay present: iso</code></pre>
<p>We have successfully created a Seurat object with two assays, <code>iso</code> containing the isoform counts counts matrix, and <code>RNA</code> containing the gene counts matrix. The <code>iso</code> assay contains 225080 different isoforms, and the <code>RNA</code> assay contains 19877 genes. The Seurat object has 17637 cell barcodes before QC filtering.</p>
</div>
<div class="section level3">
<h3 id="quality-control-qc-filtering">Quality control (QC) filtering<a class="anchor" aria-label="anchor" href="#quality-control-qc-filtering"></a>
</h3>
<p>Here we will insepct the distribution of certain QC metrics in order to determine reasonable cutoffs for outlier removal. We can inspect distributions of gene UMI counts, the number of genes detected, isoform UMI counts, and the number of isoforms detected.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load unprocessed seurat object:</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">'PBMC_masseq_unprocessed_seurat.rds'</span><span class="op">)</span>

<span class="co"># violin plots for different QC stats</span>
<span class="va">features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'nCount_RNA'</span>, <span class="st">'nFeature_RNA'</span>, <span class="st">'nCount_iso'</span>, <span class="st">'nFeature_iso'</span><span class="op">)</span>

<span class="co"># make a violin plot for each QC metric</span>
<span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">features</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html" class="external-link">VlnPlot</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  features <span class="op">=</span> <span class="va">x</span>,
  group.by <span class="op">=</span> <span class="st">'Sample'</span>,
  pt.size<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">RotatedAxis</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>notch<span class="op">=</span><span class="cn">TRUE</span>, fill<span class="op">=</span><span class="cn">NA</span>, outlier.shape<span class="op">=</span><span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span>, hjust<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># assemble plots with patchwork</span>
<span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">plot_list</span>, ncol<span class="op">=</span><span class="fl">4</span><span class="op">)</span></code></pre></div>
<p><img src="figures/isoseq/vln_qc.png" width="600" height="600"></p>
<p>Based on these distributions, we remove barcodes with fewer than 500 gene UMI counts and greater than 5000 gene UMI counts.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># keep cells with greater than 500 UMIs and fewer than 5000 UMIs</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">seurat_obj</span>, <span class="va">nCount_RNA</span> <span class="op">&gt;=</span> <span class="fl">500</span> <span class="op">&amp;</span> <span class="va">nCount_RNA</span> <span class="op">&lt;=</span> <span class="fl">5000</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="clustering-on-gene-counts-matrix">Clustering on gene counts matrix<a class="anchor" aria-label="anchor" href="#clustering-on-gene-counts-matrix"></a>
</h3>
<p>In this section, we perform clustering analysis on the gene counts matrix. We use <a href="https://github.com/immunogenomics/harmony" class="external-link">harmony</a> to integrate the two replicates prior to clustering, but we do also include the results without integration. In general this section follows the generic Seurat clustering workflow.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># change the assay to the gene counts matrix</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'RNA'</span>

<span class="co"># run normalization, feature selection, scaling, and linear dimensional reduction</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span>reduction.name <span class="op">=</span> <span class="st">'pca_RNA'</span><span class="op">)</span>

<span class="co"># run harmony to integrate the two replicates</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/harmony/man/RunHarmony.html" class="external-link">RunHarmony</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>, 
  group.by.vars <span class="op">=</span> <span class="st">'Sample'</span>,
  reduction <span class="op">=</span> <span class="st">'pca_RNA'</span>,
  reduction.save <span class="op">=</span> <span class="st">'harmony_RNA'</span>,
  assay.use <span class="op">=</span> <span class="st">'RNA'</span>
<span class="op">)</span>

<span class="co"># run UMAP</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  dims<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,
  min.dist<span class="op">=</span><span class="fl">0.3</span>,
  reduction<span class="op">=</span><span class="st">'harmony_RNA'</span>,
  reduction.name <span class="op">=</span> <span class="st">'umap_RNA'</span>
<span class="op">)</span>

<span class="co"># run clustering</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">seurat_obj</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, reduction<span class="op">=</span><span class="st">'harmony_RNA'</span><span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">seurat_obj</span>, resolution <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters_RNA</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters</span>

<span class="co"># plot clusters and samples on the UMAP</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters_RNA'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_RNA'</span><span class="op">)</span>  <span class="op">+</span> 
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'Sample'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_RNA'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, features<span class="op">=</span><span class="st">'nCount_RNA'</span>, reduction<span class="op">=</span><span class="st">'umap_RNA'</span>, order<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> 

<span class="co"># show the plots</span>
<span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span> <span class="op">|</span> <span class="va">p3</span></code></pre></div>
<p><img src="figures/isoseq/umap_RNA.png" width="900" height="600"></p>
<details><summary>
See clustering analysis without Harmony
</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># run UMAP</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  dims<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,
  min.dist<span class="op">=</span><span class="fl">0.3</span>,
  reduction<span class="op">=</span><span class="st">'pca_RNA'</span>,
  reduction.name <span class="op">=</span> <span class="st">'umap_pca'</span>
<span class="op">)</span>

<span class="co"># run clustering</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">seurat_obj</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, reduction<span class="op">=</span><span class="st">'pca_RNA'</span>, graph.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pca_nn'</span>, <span class="st">'pca_snn'</span><span class="op">)</span><span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">seurat_obj</span>, resolution <span class="op">=</span> <span class="fl">0.5</span>, graph.name<span class="op">=</span><span class="st">'pca_snn'</span><span class="op">)</span>
<span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters_pca</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters</span>

<span class="co"># plot clusters and samples on the UMAP</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'Sample'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_pca'</span><span class="op">)</span>  <span class="op">+</span> 
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters_pca'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_pca'</span><span class="op">)</span>  <span class="op">+</span> 
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">'PCA'</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'Sample'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_RNA'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters_RNA'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_RNA'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">'Harmony'</span><span class="op">)</span>

<span class="co"># show the plots</span>
<span class="op">(</span><span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="va">p3</span> <span class="op">|</span> <span class="va">p4</span><span class="op">)</span></code></pre></div>
<p>We can see that the PCA-based clustering separates each replicate, whereas the harmony-based clustering results in clusters originating from both replicates.</p>
<p><img src="figures/isoseq/umap_RNA_pca.png" width="900" height="600"></p>
</details>
</div>
<div class="section level3">
<h3 id="clustering-on-isoform-counts-matrix">Clustering on isoform counts matrix<a class="anchor" aria-label="anchor" href="#clustering-on-isoform-counts-matrix"></a>
</h3>
<p>In this section, we perform Seurat clustering analysis on the isoform counts matrix. We use nearly the same steps as above with the gene counts matrix, however we tackle the feature selection step in a slightly different way. In this particular dataset, there are many isoforms that are expressed much higher in one replicate or the other, and this variance in expression is picked up by the <code>FindVariableFeatures</code> function. Therefore, running <code>FindVariableFeatures</code> followed by the rest of the clustering steps will result in clusters that separate by replicate, even after running Harmony (we did not show this analysis here, but you could run it yourself as a fun exercise). Instead, we run <code>FindVariableFeatures</code> separately on each replicate and take the intersection of the variable isoforms for further analysis, giving us a set of variable isoforms for the whole dataset. The rest of the steps are identical to the gene-level analysis.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># switch to the isoform assay </span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'iso'</span>

<span class="co"># log normalize data</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>

<span class="co"># Get top 20k variable isoforms in each sample and take the intersection</span>
<span class="va">selected_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span><span class="va">seurat_obj</span>, split.by <span class="op">=</span> <span class="st">'Sample'</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">x</span>, nfeatures<span class="op">=</span><span class="fl">20000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">VariableFeatures</span>
<span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">intersect</span>, <span class="va">selected_features</span><span class="op">)</span>

<span class="co"># scale data and run PCA</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span>reduction.name<span class="op">=</span><span class="st">'pca_iso'</span><span class="op">)</span>

<span class="co"># run harmony</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/harmony/man/RunHarmony.html" class="external-link">RunHarmony</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by.vars <span class="op">=</span> <span class="st">'Sample'</span>,
  reduction<span class="op">=</span><span class="st">'pca_iso'</span>,
  assay.use <span class="op">=</span> <span class="st">'iso'</span>,
  reduction.save <span class="op">=</span> <span class="st">'harmony_iso'</span>
<span class="op">)</span>

<span class="co"># run umap</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  dims<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,
  min.dist<span class="op">=</span><span class="fl">0.3</span>,
  reduction<span class="op">=</span><span class="st">'harmony_iso'</span>,
  reduction.name<span class="op">=</span><span class="st">'umap_iso'</span>
<span class="op">)</span>

<span class="co"># clustering</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,
  reduction<span class="op">=</span><span class="st">'harmony_iso'</span>, graph.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'iso_nn'</span>, <span class="st">'iso_snn'</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">seurat_obj</span>, resolution <span class="op">=</span> <span class="fl">0.5</span>, graph.name<span class="op">=</span><span class="st">'iso_snn'</span><span class="op">)</span>
<span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters_iso</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters</span>

<span class="co"># plot clusters and samples on the UMAP</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters_iso'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_iso'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters_RNA'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_iso'</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'Sample'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_iso'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, features<span class="op">=</span><span class="st">'nCount_iso'</span>, reduction<span class="op">=</span><span class="st">'umap_iso'</span><span class="op">)</span>

<span class="co"># assemble with patchwork</span>
<span class="op">(</span><span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="va">p3</span> <span class="op">|</span> <span class="va">p4</span><span class="op">)</span></code></pre></div>
<p><img src="figures/isoseq/umap_iso.png" width="900" height="600"></p>
<p>We visualize the isoform-level clustering and the gene-level clustering on the isoform UMAP, and we can see that qualitatively there appears to be some overlap in the cluster assignments. Without further analysis, it is not immediately obvious which clustering is better than the other as both seem to separate the major cell populations in the PBMC dataset.</p>
<div class="section level4">
<h4 id="multimodal-clustering-with-genes-and-isoforms">Multimodal clustering with genes and isoforms<a class="anchor" aria-label="anchor" href="#multimodal-clustering-with-genes-and-isoforms"></a>
</h4>
<p>Seurat includes functionality for multimodal clustering analysis using an algorithm called <strong>Weighted Nearest Neighbors (WNN)</strong> to create a cell graph by accounting for information from two assays simultaneously. This has been used by the Seurat authors with CITE-Seq data and RNA + ATAC multiome data with some success, but it is not clear if this would help us better characterize our long-read single-cell data. Here we follow the major steps from the <a href="https://satijalab.org/seurat/articles/weighted_nearest_neighbor_analysis.html" class="external-link">Seurat WNN tutorial</a> to perform a multimodal clustering analysis using both the gene and isoform assays. Note that this analysis first requires you to have already processed both of your assays of interest.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># switch back to genes assay</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'RNA'</span>

<span class="co"># run the WNN</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMultiModalNeighbors.html" class="external-link">FindMultiModalNeighbors</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>, reduction.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"harmony_RNA"</span>, <span class="st">"harmony_iso"</span><span class="op">)</span>,
  dims.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span>, modality.weight.name <span class="op">=</span> <span class="st">"RNA.weight"</span>
<span class="op">)</span>

<span class="co"># UMAP and clustering, make sure to use the WNN output!</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  nn.name <span class="op">=</span> <span class="st">"weighted.nn"</span>,
  reduction.name <span class="op">=</span> <span class="st">"wnn.umap"</span>,
  reduction.key <span class="op">=</span> <span class="st">"wnnUMAP_"</span>
<span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  graph.name <span class="op">=</span> <span class="st">"wsnn"</span>,
  algorithm <span class="op">=</span> <span class="fl">3</span>,
  resolution <span class="op">=</span> <span class="fl">0.5</span>,
  verbose <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>
<span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters_wnn</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters</span>

<span class="co"># plot clusters from each analysis on the respective UMAPs</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters_RNA'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_RNA'</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">'Gene-level clusters'</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters_iso'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_iso'</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">'Isoform-level clusters'</span><span class="op">)</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'seurat_clusters_wnn'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'wnn.umap'</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">'Gene + Isoform WNN'</span><span class="op">)</span>


<span class="co"># assemble with patchwork</span>
<span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span> <span class="op">|</span> <span class="va">p3</span></code></pre></div>
<p><img src="figures/isoseq/umap_wnn.png" width="900" height="600"></p>
<p>Qualitatively, a side-by-side comparison of the three clustering analysis shows similar results, but we note that your milage may vary depending on your particular tissue or system of interest. WNN outputs cell-specific modality weights that tell us about the relative information content of each cell for that modality. This is a weight between 0 and 1 where close to 0 means most of the<br>
information would come from the isoform assay, close to 1 means most information would come from the gene assay, and close to 0.5 means that both assays provide similar levels of information. Here we visualize the distributions of these weights in each of the WNN clusters.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html" class="external-link">VlnPlot</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  features <span class="op">=</span> <span class="st">"RNA.weight"</span>,
  group.by <span class="op">=</span> <span class="st">'seurat_clusters_wnn'</span>,
  sort <span class="op">=</span> <span class="cn">TRUE</span>,
  pt.size<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>notch<span class="op">=</span><span class="cn">TRUE</span>, fill<span class="op">=</span><span class="cn">NA</span>, outlier.shape<span class="op">=</span><span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept<span class="op">=</span><span class="fl">0.5</span>, linetype<span class="op">=</span><span class="st">'dashed'</span>, color<span class="op">=</span><span class="st">'black'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">'WNN clusters'</span><span class="op">)</span>

<span class="va">p</span></code></pre></div>
<p><img src="figures/isoseq/vln_wnn_weight.png" width="900" height="600"></p>
<p>It appears that for the genes + isoforms WNN analysis, there does not seem to be a huge advantage of using the information from both assays simultaneously, as we can see that the median for most of these groups is near 0.5. Cluster 8 has the highest median <code>RNA.weight = 0.574</code>, so this cluster may have been occluded in the isoform-level clustering analysis. We can use a Sankey diagram to compare the clustering assignments from the three separate analyses. Note that this step is not necessary and can be skipped.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install ggsankey if necessary</span>
<span class="co"># install.packages("remotes")</span>
<span class="co"># remotes::install_github("davidsjoberg/ggsankey")</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ggsankey</span><span class="op">)</span>

<span class="va">df</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://rdrr.io/pkg/harmony/man/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">ggsankey</span><span class="fu">::</span><span class="fu">make_long</span><span class="op">(</span><span class="va">seurat_clusters_RNA</span>, <span class="va">seurat_clusters_wnn</span>, <span class="va">seurat_clusters_iso</span><span class="op">)</span>

<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, 
               next_x <span class="op">=</span> <span class="va">next_x</span>, 
               node <span class="op">=</span> <span class="va">node</span>, 
               next_node <span class="op">=</span> <span class="va">next_node</span>,
               fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">node</span><span class="op">)</span>,
               label <span class="op">=</span> <span class="va">node</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_sankey</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_sankey_label</span><span class="op">(</span>size<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_sankey</span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_x_discrete</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
    <span class="st">'seurat_clusters_RNA'</span> <span class="op">=</span> <span class="st">'Genes'</span>,
    <span class="st">'seurat_clusters_wnn'</span> <span class="op">=</span> <span class="st">'WNN'</span>,
    <span class="st">'seurat_clusters_iso'</span> <span class="op">=</span> <span class="st">'Isoforms'</span>
    <span class="op">)</span><span class="op">)</span> 

<span class="va">p</span></code></pre></div>
<p><img src="figures/isoseq/cluster_sankey.png" width="900" height="600"></p>
<p>It appears that WNN cluster 8 was grouped together with another cluster in the isoform-level analysis, but it was separated out in the RNA and WNN analyses.</p>
<p>Thus far we have performed three different clustering analyses to demonstrate different ways of analyzing long-read single-cell data, but we want to note that this is not a thorough or quantitative comparison of the different approaches and we ultimately do not have a clear method that is superior. For the remainder of this tutorial, we are going to use the UMAP and clustering from the gene-level analysis, but this could have just as easily been done with the isoform-level clusters or the WNN clusters with likely very similar results.</p>
</div>
</div>
<div class="section level3">
<h3 id="cell-type-annotation">Cell type annotation<a class="anchor" aria-label="anchor" href="#cell-type-annotation"></a>
</h3>
<p>In this section we manually assign cell-type labels to each cluster. We inspected the expression levels of many different known PBMC marker genes which we obtained from <a href="https://azimuth.hubmapconsortium.org/references/" class="external-link">Azimuth</a> to assign each cluster with a cell type label. Note that this can also be done automatically using transfer learning algorithms and a reasonable reference dataset.</p>
<details><summary>
See cell type marker genes
</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># list of selected cell type markers for different PBMC populations</span>
<span class="co"># note that this is not a comprehensive list!</span>
<span class="va">markers</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
<span class="st">'ASDC_mDC'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'AXL'</span>, <span class="st">'LILRA4'</span>, <span class="st">'SCN9A'</span>, <span class="st">'CLEC4C'</span><span class="op">)</span>,
<span class="st">'ASDC_pDC'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'SCT'</span>, <span class="st">'PROC'</span>, <span class="st">'LTK'</span><span class="op">)</span>,
<span class="st">'cDC'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'CD14'</span>, <span class="st">'FCER1A'</span>, <span class="st">'CLEC10A'</span>, <span class="st">'ENHO'</span>, <span class="st">'CD1C'</span><span class="op">)</span>,
<span class="st">'Eryth'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'HBM'</span>, <span class="st">'HBD'</span>, <span class="st">'SNCA'</span>, <span class="st">'ALAS2'</span><span class="op">)</span>,
<span class="st">'Mature B'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'MS4A1'</span>, <span class="st">'IGKC'</span>, <span class="st">'IGHM'</span>, <span class="st">'CD24'</span>, <span class="st">'CD22'</span><span class="op">)</span>,
<span class="st">'Memory B'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'BANK1'</span>, <span class="st">'CD79A'</span>, <span class="st">'SSPN'</span><span class="op">)</span>,
<span class="st">'Naive B'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'TCL1A'</span>, <span class="st">'IGHD'</span>, <span class="st">'IL4R'</span>, <span class="st">'CD79B'</span><span class="op">)</span>,
<span class="st">'CD14 Mono'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'LYZ'</span>, <span class="st">'CTSD'</span>, <span class="st">'VCAN'</span>, <span class="st">'CTSS'</span><span class="op">)</span>,
<span class="st">'CD16 Mono'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'LST1'</span>, <span class="st">'YBX1'</span>, <span class="st">'AIF1'</span>, <span class="st">'MS4A7'</span><span class="op">)</span>,
<span class="st">'CD4 CTL'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'GZMH'</span>, <span class="st">'CD4'</span>, <span class="st">'GNLY'</span>, <span class="st">'IL7R'</span>, <span class="st">'CCL5'</span><span class="op">)</span>,
<span class="st">'CD4+ Naive T'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'TCF7'</span>, <span class="st">'LEF1'</span>, <span class="st">'NUCB2'</span>, <span class="st">'LDHB'</span><span class="op">)</span>,
<span class="st">'Proliferating T'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'PCLAF'</span>, <span class="st">'CD8B'</span>, <span class="st">'CD3D'</span>, <span class="st">'TRAC'</span>, <span class="st">'CLSPN'</span><span class="op">)</span>,
<span class="st">'CD8 TCM'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'CD8A'</span>, <span class="st">'SELL'</span>, <span class="st">'ITGB1'</span>, <span class="st">'CD8'</span><span class="op">)</span>,
<span class="st">'gDT'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'TRDC'</span>, <span class="st">'TRGV9'</span>, <span class="st">'TRDV2'</span>, <span class="st">'KLRD1'</span><span class="op">)</span>,
<span class="st">'MAIT'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'KLRB1'</span>, <span class="st">'NKG7'</span>, <span class="st">'GZMK'</span>, <span class="st">'SLC4A10'</span>, <span class="st">'NCR3'</span>, <span class="st">'CTSW'</span>, <span class="st">'KLRG1'</span>, <span class="st">'CEBPD'</span>, <span class="st">'DUSP2'</span><span class="op">)</span>,
<span class="st">'NK'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'STMN1'</span>, <span class="st">'KLRC2'</span>, <span class="st">'S100A4'</span>, <span class="st">'CD3E'</span>, <span class="st">'CST7'</span><span class="op">)</span>,
<span class="st">'Plasmablast'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'TYMS'</span>, <span class="st">'TK1'</span>, <span class="st">'ASPM'</span>, <span class="st">'SHCBP1'</span>, <span class="st">'TPX2'</span><span class="op">)</span>,
<span class="st">'Platelet'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'GNG11'</span>, <span class="st">'PPBP'</span>, <span class="st">'NRGN'</span>, <span class="st">'PF4'</span>, <span class="st">'TUBB1'</span>, <span class="st">'CLU'</span><span class="op">)</span>,
<span class="st">'Treg'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'B2M'</span>, <span class="st">'FOXP3'</span>, <span class="st">'RTKN2'</span>, <span class="st">'TIGIT'</span>, <span class="st">'CTLA4'</span><span class="op">)</span>,
<span class="st">'Basophil'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'KIT'</span>, <span class="st">'CPA3'</span>, <span class="st">'TPSAB1'</span>, <span class="st">'CD44'</span>, <span class="st">'GATA2'</span>, <span class="st">'GRAP2'</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># make a dotplot for each set of markers</span>
<span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">markers</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span>
    <span class="va">seurat_obj</span>,
    features <span class="op">=</span> <span class="va">markers</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span>,
    group.by <span class="op">=</span> <span class="st">'seurat_clusters_RNA'</span>
  <span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># plot the results and save to a .pdf to inspect the results.</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">fig_dir</span>, <span class="st">'dotplot_markers.pdf'</span><span class="op">)</span>, width<span class="op">=</span><span class="fl">6</span>, height<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="kw">for</span><span class="op">(</span><span class="va">p</span> <span class="kw">in</span> <span class="va">plot_list</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</details><p>We generated a file called pbmc_annotations.txt containing the following table based on the marker gene expression profiles.</p>
<table class="table">
<thead><tr class="header">
<th>seurat_cluster</th>
<th>cell_type</th>
<th>annotation</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>T</td>
<td>CD4+ Naive T</td>
</tr>
<tr class="even">
<td>1</td>
<td>T</td>
<td>Proliferating T</td>
</tr>
<tr class="odd">
<td>2</td>
<td>Monocyte</td>
<td>CD14+ Monocyte</td>
</tr>
<tr class="even">
<td>3</td>
<td>T</td>
<td>Regulatory T</td>
</tr>
<tr class="odd">
<td>4</td>
<td>B</td>
<td>Naive B</td>
</tr>
<tr class="even">
<td>5</td>
<td>T</td>
<td>MAIT</td>
</tr>
<tr class="odd">
<td>6</td>
<td>T</td>
<td>CD8+ TEM</td>
</tr>
<tr class="even">
<td>7</td>
<td>T</td>
<td>NK</td>
</tr>
<tr class="odd">
<td>8</td>
<td>B</td>
<td>Memory B</td>
</tr>
<tr class="even">
<td>9</td>
<td>B</td>
<td>Naive B</td>
</tr>
<tr class="odd">
<td>10</td>
<td>Platelet</td>
<td>Platelet</td>
</tr>
<tr class="even">
<td>11</td>
<td>DC</td>
<td>DC</td>
</tr>
<tr class="odd">
<td>12</td>
<td>Monocyte</td>
<td>CD16+ Monocyte</td>
</tr>
</tbody>
</table>
<p>Next we add this information to the Seurat object and plot the labels on the UMAP.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'data/pbmc_annotations.txt'</span><span class="op">)</span>, header<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">'\t'</span><span class="op">)</span>
<span class="va">ix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">seurat_clusters_RNA</span>, <span class="va">annotations</span><span class="op">$</span><span class="va">seurat_cluster</span><span class="op">)</span>
<span class="va">seurat_obj</span><span class="op">$</span><span class="va">annotation</span> <span class="op">&lt;-</span> <span class="va">annotations</span><span class="op">$</span><span class="va">annotation</span><span class="op">[</span><span class="va">ix</span><span class="op">]</span>
<span class="va">seurat_obj</span><span class="op">$</span><span class="va">cell_type</span> <span class="op">&lt;-</span> <span class="va">annotations</span><span class="op">$</span><span class="va">cell_type</span><span class="op">[</span><span class="va">ix</span><span class="op">]</span>

<span class="co"># plot clusters and samples on the UMAP</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'cell_type'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_RNA'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by <span class="op">=</span> <span class="st">'annotation'</span>, label<span class="op">=</span><span class="cn">TRUE</span>, repel<span class="op">=</span><span class="cn">TRUE</span>, reduction<span class="op">=</span><span class="st">'umap_RNA'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># assemble with patchwork</span>
<span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span></code></pre></div>
<p><img src="figures/isoseq/umap_RNA_annotated.png" width="900" height="600"></p>
<p>We have now finished the Seurat clustering part of this tutorial, so it is a good idea to save your Seurat object at this point before moving on to network analysis.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">seurat_obj</span>, file <span class="op">=</span> <span class="st">'PBMC_masseq_processed_seurat.rds'</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="isoform-co-expression-network-analysis">Isoform co-expression network analysis<a class="anchor" aria-label="anchor" href="#isoform-co-expression-network-analysis"></a>
</h2>
<p>In this section, we perform isofom co-expression network analysis with hdWGCNA. We cover several different approaches in this tutorial. In general, isoform network analysis is more complicated than gene co-expression network analysis since we are working with a larger data matrix to start with. For this dataset, the isoform matrix has roughly ten times the number of features as the gene matrix. Since networks are represented as features-by-features square matrices, it is extremely computationally challenging to compute and represent a full isoform co-expression network. Additionally, many isoforms are expressed at low levels, and like ordinary scRNA-seq we have technical data sparsity as well. For these reasons, we have to carefully consider what isoforms and what cell populations to include in our co-expression network analysis.</p>
<ol style="list-style-type: decimal">
<li>Variable features in all PBMC clusters</li>
<li>Major isoforms of cluster marker genes in all PBMC clusters</li>
<li>All major isoforms in NK cells</li>
</ol>
<p>Before proceeding, we need to remove the very smallest cell populations which could cause trouble in our metacell grouping step. We also need to load the relevant libraries for running hdWGCNA, and we enable multithreading (optional) to speed up our processing time.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/" class="external-link">WGCNA</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://smorabit.github.io/hdWGCNA/">hdWGCNA</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/allowWGCNAThreads.html" class="external-link">enableWGCNAThreads</a></span><span class="op">(</span>nThreads <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>

<span class="co"># re-load processed Seurat object:</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'PBMC_masseq_processed_seurat.rds'</span><span class="op">)</span>

<span class="co"># subset to remove the smallest cell types</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">[</span>,<span class="op">!</span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">annotation</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'DC'</span>, <span class="st">'Platelet'</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<div class="section level3">
<h3 id="example-1-variable-features-in-all-pbmcs">Example 1: Variable Features in all PBMCs<a class="anchor" aria-label="anchor" href="#example-1-variable-features-in-all-pbmcs"></a>
</h3>
<p>First, we will demonstrate isoform co-expression network analysis using variable features as our input set of isoforms, and we will use all of the PBMCs. Similar to our clustering analysis, we will identify variable isoforms separately for each sample and take the intersection as our feature set. However, we would like to use more features for network analysis, so we increase the number of variable features to 50000.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># switch to isoform assay:</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'iso'</span>

<span class="co"># Get top 50k variable isoforms in each sample and take the intersection</span>
<span class="va">selected_features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span><span class="va">seurat_obj</span>, split.by <span class="op">=</span> <span class="st">'Sample'</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">x</span>, nfeatures<span class="op">=</span><span class="fl">50000</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/igraph/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">VariableFeatures</span>
<span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="va">intersect</span>, <span class="va">selected_features</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 9265</code></pre>
<p>This approach gives us 9265 isoforms to perform network analysis on. Next we run the <code>SetupForWGCNA</code> function and indicate that we want to use variable features as our set of input features for network analysis. We also perform metacell aggregation with the <code>MetacellsByGroups</code> function, specifying <code>assay="iso"</code> in order to obtain a metacells by isoforms matrix.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># setup this hdWGCNA experiment</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetupForWGCNA.html">SetupForWGCNA</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  gene_select <span class="op">=</span> <span class="st">"variable"</span>,
  wgcna_name <span class="op">=</span> <span class="st">"variable"</span>,
<span class="op">)</span>

<span class="co"># construct metacells:</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MetacellsByGroups.html">MetacellsByGroups</a></span><span class="op">(</span>
  seurat_obj <span class="op">=</span> <span class="va">seurat_obj</span>,
  group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"annotation"</span>, <span class="st">"cell_type"</span>, <span class="st">"Sample"</span><span class="op">)</span>,
  k <span class="op">=</span> <span class="fl">50</span>,
  max_shared<span class="op">=</span><span class="fl">25</span>,
  min_cells<span class="op">=</span><span class="fl">75</span>,
  reduction <span class="op">=</span> <span class="st">'harmony_RNA'</span>,
  ident.group <span class="op">=</span> <span class="st">'annotation'</span>,
  assay <span class="op">=</span> <span class="st">'iso'</span>,
  mode <span class="op">=</span> <span class="st">'sum'</span>,
  target_metacells<span class="op">=</span><span class="fl">150</span>
<span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NormalizeMetacells.html">NormalizeMetacells</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span></code></pre></div>
<p>Next we specify the cell groups that will be used for network analysis and we set up the expression matrix using <code>SetDatExpr</code>. Note that we are setting <code>group_name = unique(seurat_obj$cell_type)</code> in order to use all of the cell types for a single network analysis. Once again we ensure to specify <code>assay='iso'</code> for the isoform-level analysis. We then run the <code>TestSoftPowers</code> function to find suitable values for the <code>soft_power</code> parameter.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get list of cell populations</span>
<span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="../reference/GetMetacellObject.html">GetMetacellObject</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span><span class="op">$</span><span class="va">annotation</span><span class="op">)</span>

<span class="co"># set up gene expression matrix</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetDatExpr.html">SetDatExpr</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by<span class="op">=</span><span class="st">'cell_type'</span>,
  group_name <span class="op">=</span> <span class="va">groups</span>,
  use_metacells<span class="op">=</span><span class="cn">TRUE</span>,
  slot <span class="op">=</span> <span class="st">'data'</span>,
  assay <span class="op">=</span> <span class="st">'iso'</span>
<span class="op">)</span>

<span class="co"># test soft powers</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TestSoftPowers.html">TestSoftPowers</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotSoftPowers.html">PlotSoftPowers</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>

<span class="co"># assemble with patchwork</span>
<span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">plot_list</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="figures/isoseq/variable_test_softpower.png" width="900" height="600"></p>
<p>Now we are ready to construct the network, compute module eigen<strong>isoforms</strong> (MEiso), and compute eigenisoform-based connectivities (kMEiso).</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># construct wgcna network:</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ConstructNetwork.html">ConstructNetwork</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  soft_power<span class="op">=</span><span class="fl">5</span>,
  setDatExpr<span class="op">=</span><span class="cn">FALSE</span>,
  detectCutHeight<span class="op">=</span><span class="fl">0.995</span>,
  mergeCutHeight<span class="op">=</span><span class="fl">0.05</span>,
  TOM_name <span class="op">=</span> <span class="st">'variable'</span>,
  minModuleSize<span class="op">=</span><span class="fl">30</span>,
  overwrite_tom <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="co"># plot the dendrogram</span>
<span class="fu"><a href="../reference/PlotDendrogram.html">PlotDendrogram</a></span><span class="op">(</span><span class="va">seurat_obj</span>, main<span class="op">=</span><span class="st">'hdWGCNA Dendrogram'</span><span class="op">)</span>
     
<span class="co"># note: we have to run ScaleData or else harmony won't work in the ModuleEigengenes function</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seurat_obj</span>, features<span class="op">=</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span><span class="op">)</span>

<span class="co"># harmony correction by Sample</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleEigengenes.html">ModuleEigengenes</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by.vars <span class="op">=</span> <span class="st">'Sample'</span>,
  verbose<span class="op">=</span><span class="cn">TRUE</span>
<span class="op">)</span>       

<span class="co"># compute module connectivity:</span>
<span class="va">modules_orig</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleConnectivity.html">ModuleConnectivity</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span></code></pre></div>
<p><img src="figures/isoseq/variable_dendro.png" width="900" height="600"></p>
<p>We have now constructed the isoform co-expression network, so we can now perform downstream analysis to better characterize and understand the network and these isoform modules. Before we move on to another way of doing isoform co-expression network analysis, we plot the MEiso values for each module in each cluster with <code>DotPlot</code>, and we use <code>RunModuleUMAP</code> to visualize the isoform co-expression network in two dimensions.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get the MEisos from the seurat object and add it to the metadata</span>
<span class="va">MEiso</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMEs.html">GetMEs</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span>
<span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">meta</span>, <span class="va">MEiso</span><span class="op">)</span>

<span class="co"># get a list of features to plot</span>
<span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="va">mods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">modules</span><span class="op">$</span><span class="va">module</span><span class="op">)</span>
<span class="va">mods</span> <span class="op">&lt;-</span> <span class="va">mods</span><span class="op">[</span><span class="va">mods</span><span class="op">!=</span><span class="st">'grey'</span><span class="op">]</span>

<span class="co"># make dotplot</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by<span class="op">=</span><span class="st">'annotation'</span>,
  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">mods</span><span class="op">)</span>
<span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">RotatedAxis</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradient2</a></span><span class="op">(</span>high<span class="op">=</span><span class="st">'red'</span>, mid<span class="op">=</span><span class="st">'grey95'</span>, low<span class="op">=</span><span class="st">'blue'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,
    axis.line.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
    axis.line.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
    panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span>, fill<span class="op">=</span><span class="cn">NA</span>, size<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
  <span class="op">)</span>

<span class="co"># show plot</span>
<span class="va">p</span>

<span class="co"># restore the original metadata</span>
<span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> <span class="va">meta</span></code></pre></div>
<p><img src="figures/isoseq/variable_dotplot_MEs.png" width="900" height="600"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunModuleUMAP.html">RunModuleUMAP</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  n_hubs <span class="op">=</span><span class="fl">5</span>,
  n_neighbors<span class="op">=</span><span class="fl">15</span>,
  min_dist<span class="op">=</span><span class="fl">0.2</span>,
  spread<span class="op">=</span><span class="fl">1</span>
  <span class="co">#supervised=TRUE,</span>
  <span class="co">#target_weight=0.5</span>
<span class="op">)</span>


<span class="co"># get the hub gene UMAP table from the seurat object</span>
<span class="va">umap_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModuleUMAP.html">GetModuleUMAP</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>
<span class="op">)</span>

<span class="co"># plot with ggplot</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">umap_df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">UMAP1</span>, y<span class="op">=</span><span class="va">UMAP2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>
   color<span class="op">=</span><span class="va">umap_df</span><span class="op">$</span><span class="va">color</span>,
   size<span class="op">=</span><span class="va">umap_df</span><span class="op">$</span><span class="va">kME</span><span class="op">*</span><span class="fl">2</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">fig_dir</span>, <span class="st">'variable_test_hubgene_umap_ggplot_uns.pdf'</span><span class="op">)</span>, width<span class="op">=</span><span class="fl">5</span>, height<span class="op">=</span><span class="fl">5</span><span class="op">)</span>
<span class="va">p</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">fig_dir</span>, <span class="st">'variable_coex_umap.png'</span><span class="op">)</span>, width<span class="op">=</span><span class="fl">6</span>, height<span class="op">=</span><span class="fl">6</span>, units<span class="op">=</span><span class="st">'in'</span>, res<span class="op">=</span><span class="fl">500</span><span class="op">)</span>
<span class="fu"><a href="../reference/ModuleUMAPPlot.html">ModuleUMAPPlot</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  edge.alpha<span class="op">=</span><span class="fl">0.5</span>,
  sample_edges<span class="op">=</span><span class="cn">TRUE</span>,
  keep_grey_edges<span class="op">=</span><span class="cn">FALSE</span>,
  edge_prop<span class="op">=</span><span class="fl">0.075</span>, <span class="co"># taking the top 20% strongest edges in each module</span>
<span class="co">#   label_genes = umap_df$isoform_name[umap_df$gene_name %in% plot_genes],</span>
  <span class="co">#label_genes = c(''),</span>
  label_hubs<span class="op">=</span><span class="fl">2</span> <span class="co"># how many hub genes to plot per module?</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">seurat_obj</span>, file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">'PBMC_masseq_hdWGCNA_wip.rds'</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="figures/isoseq/variable_coex_umap.png" width="900" height="600"></p>
</div>
<div class="section level3">
<h3 id="example-2-major-isoforms-of-cluster-marker-genes-in-all-pbmcs">Example 2: Major isoforms of cluster marker genes in all PBMCs<a class="anchor" aria-label="anchor" href="#example-2-major-isoforms-of-cluster-marker-genes-in-all-pbmcs"></a>
</h3>
<p>Here we use a different approach for obtaining a list of input isoforms for hdWGCNA. We first identify <strong>Major Isoforms</strong>, defined as the set of isoforms accounting for 80% of a gene’s total expression. Major isoforms are identified for each gene and for each cell population, and the 80% threshold parameter can be tweaked. hdWGCNA includes <code>FindMajorIsoforms</code>, a function to call major isoforms from a Seurat object.</p>
<p>The set of major isoforms across all of the cell populations may still contain feature set that would be challenging to perform network analysis on due to its size. Therefore, in this tutorial we will intersect the major isoform list with a cluster marker gene list, giving us a set of major isoforms for each cell population’s marker genes. In the following block of code, we run Seurat’s <code>FindAllMarkers</code> function on the RNA assay to get a marker gene table. Note that we are identifying <strong>marker genes</strong> and not marker isoforms for this step.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># switch to the RNA assay because we want marker genes</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'RNA'</span>

<span class="co"># change the idents to your desired clusters, cell annotations, etc</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">$</span><span class="va">annotation</span>

<span class="co"># Run FindAllMarkers to get cluster marker genes!!</span>
<span class="va">cluster_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  logfc.threshold <span class="op">=</span> <span class="fl">0.5</span>,
  min.pct <span class="op">=</span> <span class="fl">0.1</span>
<span class="op">)</span>

<span class="co"># save your cluster markers as a table</span>
<span class="va">cluster_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>file<span class="op">=</span><span class="st">'data/cluster_markers.csv'</span><span class="op">)</span>
<span class="va">cluster_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cluster_markers</span>, <span class="op">!</span><span class="op">(</span><span class="va">cluster</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'DC'</span>, <span class="st">'Platelet'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Next, we run <code>FindMajorIsoforms</code> to identify major isoforms of the cluster marker genes, and then we set up the Seurat object with a new hdWGCNA experiment using these isoforms as the input features.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span>

<span class="co"># get the list of major isoforms in each cell population</span>
<span class="co"># intersected with the cluster_markers table</span>
<span class="va">major_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindMajorIsoforms.html">FindMajorIsoforms</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by <span class="op">=</span> <span class="st">'annotation'</span>, 
  replicate_col <span class="op">=</span> <span class="st">'Sample'</span>,
  isoform_delim <span class="op">=</span> <span class="st">'[.]'</span>,
  cluster_markers <span class="op">=</span> <span class="va">cluster_markers</span>
<span class="op">)</span>
<span class="va">major_marker_list</span> <span class="op">&lt;-</span> <span class="va">major_list</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>
<span class="va">major_list</span> <span class="op">&lt;-</span> <span class="va">major_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>

<span class="co"># merge into a single list of features for hdWGCNA</span>
<span class="va">selected_isoforms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">major_marker_list</span><span class="op">)</span><span class="op">)</span>

<span class="co"># setup for WGCNA</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetupForWGCNA.html">SetupForWGCNA</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  gene_select <span class="op">=</span> <span class="st">"custom"</span>,
  gene_list <span class="op">=</span> <span class="va">selected_isoforms</span>,
  metacell_location <span class="op">=</span> <span class="st">'variable'</span>, <span class="co"># this will use the same Metacell seurat object that we made before</span>
  wgcna_name <span class="op">=</span> <span class="st">"major"</span>
<span class="op">)</span>

<span class="co"># how many isoforms?</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">selected_isoforms</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 13217</code></pre>
<p>This approach gives us 13217 isoforms for network analysis. Now we run the main steps of co-expression network analysis with hdWGCNA similarly to what we did in the previous section.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set up expression matrix</span>
<span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="../reference/GetMetacellObject.html">GetMetacellObject</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span><span class="op">$</span><span class="va">annotation</span><span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetDatExpr.html">SetDatExpr</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by<span class="op">=</span><span class="st">'annotation'</span>,
  group_name <span class="op">=</span> <span class="va">groups</span>,
  use_metacells<span class="op">=</span><span class="cn">TRUE</span>,
  slot <span class="op">=</span> <span class="st">'data'</span>,
  assay <span class="op">=</span> <span class="st">'iso'</span>
<span class="op">)</span>

<span class="co"># test soft powers</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TestSoftPowers.html">TestSoftPowers</a></span><span class="op">(</span><span class="va">seurat_obj</span>,<span class="op">)</span>
<span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotSoftPowers.html">PlotSoftPowers</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">plot_list</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span>

<span class="co"># construct wgcna network:</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ConstructNetwork.html">ConstructNetwork</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  soft_power <span class="op">=</span> <span class="fl">5</span>,
  detectCutHeight<span class="op">=</span><span class="fl">0.995</span>,
  mergeCutHeight<span class="op">=</span><span class="fl">0.05</span>,
  TOM_name <span class="op">=</span> <span class="st">'major'</span>,
  minModuleSize<span class="op">=</span><span class="fl">30</span>,
  overwrite_tom <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="co"># plot the dendrogram</span>
<span class="fu"><a href="../reference/PlotDendrogram.html">PlotDendrogram</a></span><span class="op">(</span><span class="va">seurat_obj</span>, main<span class="op">=</span><span class="st">'hdWGCNA Dendrogram'</span><span class="op">)</span>

<span class="co"># have to have this ScaleData line or else Harmony gets angry</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seurat_obj</span>, features<span class="op">=</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span><span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleEigengenes.html">ModuleEigengenes</a></span><span class="op">(</span><span class="va">seurat_obj</span>, group.by.vars <span class="op">=</span> <span class="st">'Sample'</span><span class="op">)</span>

<span class="co"># compute module connectivity:</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleConnectivity.html">ModuleConnectivity</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span></code></pre></div>
<p><img src="figures/isoseq/major_dendro.png" width="900" height="600"></p>
<p>Next we visualize the MEiso signatures for each module in each cell population using a DotPlot.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">MEs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMEs.html">GetMEs</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="va">modules</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>
<span class="va">mods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">modules</span><span class="op">$</span><span class="va">module</span><span class="op">)</span>
<span class="va">mods</span> <span class="op">&lt;-</span> <span class="va">mods</span><span class="op">[</span><span class="va">mods</span><span class="op">!=</span><span class="st">'grey'</span><span class="op">]</span>

<span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span>
<span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">meta</span>, <span class="va">MEs</span><span class="op">)</span>


<span class="co"># make dotplot</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by<span class="op">=</span><span class="st">'annotation'</span>,
  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">mods</span><span class="op">)</span>
<span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">RotatedAxis</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_color_gradient2</a></span><span class="op">(</span>high<span class="op">=</span><span class="st">'red'</span>, mid<span class="op">=</span><span class="st">'grey95'</span>, low<span class="op">=</span><span class="st">'blue'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,
    axis.line.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
    axis.line.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
    panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span>, fill<span class="op">=</span><span class="cn">NA</span>, size<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
  <span class="op">)</span>

<span class="va">p</span></code></pre></div>
<p><img src="figures/isoseq/major_iso_dotplot_MEs.png" width="900" height="600"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> <span class="va">meta</span>


<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunModuleUMAP.html">RunModuleUMAP</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  n_hubs <span class="op">=</span><span class="fl">5</span>,
  n_neighbors<span class="op">=</span><span class="fl">15</span>,
  min_dist<span class="op">=</span><span class="fl">0.2</span>,
  spread<span class="op">=</span><span class="fl">1</span>
  <span class="co">#supervised=TRUE,</span>
  <span class="co">#target_weight=0.5</span>
<span class="op">)</span>


<span class="co"># get the hub gene UMAP table from the seurat object</span>
<span class="va">umap_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModuleUMAP.html">GetModuleUMAP</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>
<span class="op">)</span>

<span class="co"># plot with ggplot</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">umap_df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">UMAP1</span>, y<span class="op">=</span><span class="va">UMAP2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>
   color<span class="op">=</span><span class="va">umap_df</span><span class="op">$</span><span class="va">color</span>,
   size<span class="op">=</span><span class="va">umap_df</span><span class="op">$</span><span class="va">kME</span><span class="op">*</span><span class="fl">2</span>
  <span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="../reference/umap_theme.html">umap_theme</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html" class="external-link">pdf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">fig_dir</span>, <span class="st">'major_iso_hubgene_umap_ggplot_uns.pdf'</span><span class="op">)</span>, width<span class="op">=</span><span class="fl">5</span>, height<span class="op">=</span><span class="fl">5</span><span class="op">)</span>
<span class="va">p</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">fig_dir</span>, <span class="st">'major_coex_umap.png'</span><span class="op">)</span>, width<span class="op">=</span><span class="fl">7</span>, height<span class="op">=</span><span class="fl">7</span>, units<span class="op">=</span><span class="st">'in'</span>, res<span class="op">=</span><span class="fl">500</span><span class="op">)</span>
<span class="fu"><a href="../reference/ModuleUMAPPlot.html">ModuleUMAPPlot</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  edge.alpha<span class="op">=</span><span class="fl">0.5</span>,
  sample_edges<span class="op">=</span><span class="cn">TRUE</span>,
  keep_grey_edges<span class="op">=</span><span class="cn">FALSE</span>,
  edge_prop<span class="op">=</span><span class="fl">0.075</span>, <span class="co"># taking the top 20% strongest edges in each module</span>
<span class="co">#   label_genes = umap_df$isoform_name[umap_df$gene_name %in% plot_genes],</span>
  <span class="co">#label_genes = c(''),</span>
  label_hubs<span class="op">=</span><span class="fl">2</span> <span class="co"># how many hub genes to plot per module?</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span>


<span class="co"># individual module networks</span>
<span class="fu"><a href="../reference/ModuleNetworkPlot.html">ModuleNetworkPlot</a></span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  mods <span class="op">=</span> <span class="st">"all"</span>,
  <span class="co">#label_center=TRUE,</span>
  outdir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">fig_dir</span>, <span class="st">'hubNetworks/'</span><span class="op">)</span>
<span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">seurat_obj</span>, file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">'PBMC_masseq_hdWGCNA_wip.rds'</span><span class="op">)</span><span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">'tutorial_data/PBMC_masseq_hdWGCNA_major.rds'</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-3-consensus-wgcna-between-the-two-samples">Example 3: consensus wgcna between the two samples<a class="anchor" aria-label="anchor" href="#example-3-consensus-wgcna-between-the-two-samples"></a>
</h3>
<p>Coming soon!</p>
</div>
</div>
<div class="section level2">
<h2 id="isoform-plotting-with-ggtranscript-and-swan">Isoform plotting with ggtranscript and Swan<a class="anchor" aria-label="anchor" href="#isoform-plotting-with-ggtranscript-and-swan"></a>
</h2>
<p>Coming soon!</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
