<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using SCTransform normalized data â€¢ hdWGCNA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using SCTransform normalized data">
<meta property="og:description" content="Tutorial demonstrating hdWGCNA with SCTransform normalized data.
">
<meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hdWGCNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.02</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/hdWGCNA.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Core functionality</li>
    <li>
      <a href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a>
    </li>
    <li>
      <a href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a>
    </li>
    <li>
      <a href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a>
    </li>
    <li>
      <a href="../articles/network_visualizations.html">Network visualization</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Biological context for co-expression modules</li>
    <li>
      <a href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a>
    </li>
    <li>
      <a href="../articles/module_trait_correlation.html">Module trait correlation</a>
    </li>
    <li>
      <a href="../articles/enrichment_analysis.html">Enrichment analysis</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Exploring modules in external datasets</li>
    <li>
      <a href="../articles/projecting_modules.html">Projecting modules to new datasets</a>
    </li>
    <li>
      <a href="../articles/module_preservation.html">Module preservation and reproducibility</a>
    </li>
    <li>
      <a href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Advanced topics</li>
    <li>
      <a href="../articles/consensus_wgcna.html">Consensus network analysis</a>
    </li>
    <li>
      <a href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a>
    </li>
    <li>
      <a href="../articles/motif_analysis.html">Motif analysis</a>
    </li>
    <li>
      <a href="../articles/other_metacells.html">Alternative metacell algorithms</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Other</li>
    <li>
      <a href="../articles/customization.html">Module customization</a>
    </li>
    <li>
      <a href="../articles/sctransform.html">Using SCTransform normalized data</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using SCTransform normalized data</h1>
            
      
      
      <div class="hidden name"><code>sctransform.Rmd</code></div>

    </div>

    
    
<p>In this tutorial, we briefly cover how to use data normalized with <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1874-1" class="external-link"><code>SCTransform</code></a> for hdWGCNA. Overall we do not recommend using SCTransform with hdWGCNA, however we have included this tutorial due to numerous user requests. This tutorial assumes that you are familiar with the <a href="basic_tutorial.html">basics of the hdWGCNA workflow</a>. This tutorial also assumes that you are familiar with the <a href="https://github.com/satijalab/sctransform" class="external-link">SCTransform R package</a> and the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1874-1" class="external-link">SCTransform methods paper</a>. Once again, we do not advise using SCTransform with hdWGCNA so proceed with caution. If you have made it past these warnings, at least make sure you are fully aware of how SCTransform works and the underlying assumptions that it makes about gene expression distributions etc.</p>
<p>First we load the required R libraries.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># single-cell analysis package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>

<span class="co"># sctransform</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/satijalab/sctransform" class="external-link">sctransform</a></span><span class="op">)</span>

<span class="co"># plotting and data science packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span>

<span class="co"># co-expression network analysis packages:</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/" class="external-link">WGCNA</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://smorabit.github.io/hdWGCNA/">hdWGCNA</a></span><span class="op">)</span>

<span class="co"># network analysis &amp; visualization package:</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span>

<span class="co"># using the cowplot theme for ggplot</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

<span class="co"># set random seed for reproducibility</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></code></pre></div>
<p>Download the tutorial dataset:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://swaruplab.bio.uci.edu/public_data/Zhou_2020.rds</span></code></pre></div>
<div class="section level2">
<h2 id="run-sctransform">Run SCTransform<a class="anchor" aria-label="anchor" href="#run-sctransform"></a>
</h2>
<p>Here we run <code>SCTransform</code> normalization on the tutorial dataset, regressing <code>percent.mt</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load the tutorial dataset</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'data/Zhou_control.rds'</span><span class="op">)</span>

<span class="co"># subset just one cell type for the sake of speed</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span> <span class="op"><a href="https://rdrr.io/pkg/igraph/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cell_type</span> <span class="op">==</span> <span class="st">'ASC'</span><span class="op">)</span>

<span class="co"># compute percentage mitochondrial genes</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PercentageFeatureSet.html" class="external-link">PercentageFeatureSet</a></span><span class="op">(</span><span class="va">seurat_subset</span>, pattern <span class="op">=</span> <span class="st">"^MT-"</span>, col.name <span class="op">=</span> <span class="st">"percent.mt"</span><span class="op">)</span>

<span class="co"># run SCTransform</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span><span class="va">seurat_subset</span>, vars.to.regress <span class="op">=</span> <span class="st">"percent.mt"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>It is important to note that <code>SCTransform</code> pearson residuals are typically only output for the highly variable features. We can check this by extracting the expression matrix.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract sct expression data</span>
<span class="va">sct_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">seurat_subset</span>, slot<span class="op">=</span><span class="st">'scale.data'</span>, assay<span class="op">=</span><span class="st">'SCT'</span><span class="op">)</span>

<span class="co"># print the shape of this matrix (genes by cells)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sct_data</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 3000 3162</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># print the shape of the seurat object (genes by cells)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sct_data</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 36601  3162</code></pre>
<p>We see that <code>seurat_obj</code> has 36,601 genes, but only 3,000 are in the SCTransform <code>scale.data</code> slot. Therefore, if we want to use hdWGCNA on the SCTransform pearson residuals, we must only include the highly variable genes. Alternatively, SCTransform does output a <code>counts</code> and normalized <code>data</code> slot which may also be used.</p>
</div>
<div class="section level2">
<h2 id="option-1-sctransform-on-single-cell-data">Option 1: SCTransform on single-cell data<a class="anchor" aria-label="anchor" href="#option-1-sctransform-on-single-cell-data"></a>
</h2>
<p>Here we demonstrate how to run the standard hdWGCNA workflow on SCTransform normalized single-cell data. First we set up the hdWGCNA experiment, ensuring to only include genes that were used for SCTransform.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># only supply features that were used for SCTransform!!!</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetupForWGCNA.html">SetupForWGCNA</a></span><span class="op">(</span>
  <span class="va">seurat_subset</span>,
  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>,
  wgcna_name <span class="op">=</span> <span class="st">"SCT"</span>
<span class="op">)</span></code></pre></div>
<p>Next, we construct metacells while specifying <code>slot='scale.data'</code> and <code>assay='SCT'</code> in order to use the SCTransform normalized data.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MetacellsByGroups.html">MetacellsByGroups</a></span><span class="op">(</span>
  seurat_obj <span class="op">=</span> <span class="va">seurat_subset</span>,
  group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sample"</span><span class="op">)</span>,
  k <span class="op">=</span> <span class="fl">25</span>,
  max_shared<span class="op">=</span><span class="fl">12</span>,
  min_cells <span class="op">=</span> <span class="fl">50</span>,
  reduction <span class="op">=</span> <span class="st">'harmony'</span>,
  ident.group <span class="op">=</span> <span class="st">'Sample'</span>,
  slot <span class="op">=</span> <span class="st">'scale.data'</span>,
  assay <span class="op">=</span> <span class="st">'SCT'</span>
<span class="op">)</span></code></pre></div>
<p>Next, we run the rest of the main steps of the hdWGCNA pipeline.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set expression matrix for hdWGCNA</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetDatExpr.html">SetDatExpr</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="co"># test different soft power thresholds</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TestSoftPowers.html">TestSoftPowers</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>
<span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotSoftPowers.html">PlotSoftPowers</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">plot_list</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="figures/sctransform/test_softpower_SCT_cells.png" width="800" height="900"></p>
<p>Interestingly, none of the soft power thresholds tested have a scale-free topology moddel fit of 0.8 or higher. For network construction, we will choose <code>soft_power=14</code> since that is where the model fit starts to plateau.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ConstructNetwork.html">ConstructNetwork</a></span><span class="op">(</span>
  <span class="va">seurat_subset</span>,
  soft_power <span class="op">=</span> <span class="fl">14</span>,
  tom_name <span class="op">=</span> <span class="st">"SCT_cells"</span>,
  overwrite_tom <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="co"># compute module eigengenes and connectivity</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleEigengenes.html">ModuleEigengenes</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleConnectivity.html">ModuleConnectivity</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="co"># plot the dendrogram</span>
<span class="fu"><a href="../reference/PlotDendrogram.html">PlotDendrogram</a></span><span class="op">(</span><span class="va">seurat_subset</span>, main<span class="op">=</span><span class="st">'hdWGCNA SCT Dendrogram'</span><span class="op">)</span></code></pre></div>
<p><img src="figures/sctransform/dendro_SCT_cells.png" width="800" height="900"></p>
</div>
<div class="section level2">
<h2 id="option-2-sctransform-on-metacell-data">Option 2: SCTransform on metacell data<a class="anchor" aria-label="anchor" href="#option-2-sctransform-on-metacell-data"></a>
</h2>
<p>Alternatively, we can apply SCTransform after we have constructed metacells.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetupForWGCNA.html">SetupForWGCNA</a></span><span class="op">(</span>
  <span class="va">seurat_subset</span>,
  features <span class="op">=</span> <span class="fu"><a href="../reference/GetWGCNAGenes.html">GetWGCNAGenes</a></span><span class="op">(</span><span class="va">seurat_subset</span>, <span class="st">'SCT'</span><span class="op">)</span>,
  wgcna_name <span class="op">=</span> <span class="st">"SCT_meta"</span>
<span class="op">)</span>

<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MetacellsByGroups.html">MetacellsByGroups</a></span><span class="op">(</span>
  seurat_obj <span class="op">=</span> <span class="va">seurat_subset</span>,
  group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sample"</span><span class="op">)</span>,
  k <span class="op">=</span> <span class="fl">25</span>,
  max_shared<span class="op">=</span><span class="fl">12</span>,
  min_cells <span class="op">=</span> <span class="fl">50</span>,
  reduction <span class="op">=</span> <span class="st">'harmony'</span>,
  ident.group <span class="op">=</span> <span class="st">'Sample'</span>,
  slot <span class="op">=</span> <span class="st">'counts'</span>,
  assay <span class="op">=</span> <span class="st">'RNA'</span>
<span class="op">)</span></code></pre></div>
<p>Here, we extract the metacell Seurat object, then run SCTransform on it. Importantly, this will give us a different set of genes than we used previously.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get metacell object and run SCTransform</span>
<span class="va">mobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMetacellObject.html">GetMetacellObject</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>
<span class="va">mobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PercentageFeatureSet.html" class="external-link">PercentageFeatureSet</a></span><span class="op">(</span><span class="va">mobj</span>, pattern <span class="op">=</span> <span class="st">"^MT-"</span>, col.name <span class="op">=</span> <span class="st">"percent.mt"</span><span class="op">)</span>

<span class="co"># run SCTransform</span>
<span class="va">mobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span><span class="va">mobj</span>, vars.to.regress <span class="op">=</span> <span class="st">"percent.mt"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, return.only.var.genes<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># only keep genes that were used for SCTransform</span>
<span class="va">sct_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">mobj</span>, slot<span class="op">=</span><span class="st">'scale.data'</span>, assay<span class="op">=</span><span class="st">'SCT'</span><span class="op">)</span>
<span class="va">sct_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sct_data</span><span class="op">)</span>
<span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetWGCNAGenes.html">GetWGCNAGenes</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>
<span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="va">gene_list</span><span class="op">[</span><span class="va">gene_list</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">sct_genes</span><span class="op">]</span>

<span class="co"># update the genes used for WGCNA, and reset the metacell object</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetWGCNAGenes.html">SetWGCNAGenes</a></span><span class="op">(</span><span class="va">seurat_subset</span>, <span class="va">gene_list</span><span class="op">)</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetMetacellObject.html">SetMetacellObject</a></span><span class="op">(</span><span class="va">seurat_subset</span>, <span class="va">mobj</span><span class="op">)</span></code></pre></div>
<p>Next we continue with the hdWGCNA pipeline using this metacell object.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># specify SCT assay and scale.data slot</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetDatExpr.html">SetDatExpr</a></span><span class="op">(</span><span class="va">seurat_subset</span>, assay <span class="op">=</span> <span class="st">'SCT'</span>, slot<span class="op">=</span><span class="st">'scale.data'</span><span class="op">)</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TestSoftPowers.html">TestSoftPowers</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotSoftPowers.html">PlotSoftPowers</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="co"># assemble with patchwork</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">plot_list</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="figures/sctransform/test_softpower_SCT_meta.png" width="800" height="900"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># construct wgcna network:</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ConstructNetwork.html">ConstructNetwork</a></span><span class="op">(</span>
  <span class="va">seurat_subset</span>,
  tom_name <span class="op">=</span> <span class="st">"SCT_meta"</span>,
  overwrite_tom <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleEigengenes.html">ModuleEigengenes</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleConnectivity.html">ModuleConnectivity</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="fu"><a href="../reference/PlotDendrogram.html">PlotDendrogram</a></span><span class="op">(</span><span class="va">seurat_subset</span>, main<span class="op">=</span><span class="st">'hdWGCNA SCT-meta Dendrogram'</span><span class="op">)</span></code></pre></div>
<p><img src="figures/sctransform/dendro_SCT_meta.png" width="800" height="900"></p>
</div>
<div class="section level2">
<h2 id="comparing-to-the-standard-hdwgcna-workflow">Comparing to the standard hdWGCNA workflow<a class="anchor" aria-label="anchor" href="#comparing-to-the-standard-hdwgcna-workflow"></a>
</h2>
<p>Here, we run the standard hdWGCNA workflow without SCTransform so we can make comparisons later. Note that we are using the same set of genes that we did above when using SCTransform.</p>
<details><summary>
Show standard hdWGCNA
</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># set default assay to RNA so we don't use SCTransform data</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'RNA'</span>

<span class="co"># setup hdWGCNA experiment</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetupForWGCNA.html">SetupForWGCNA</a></span><span class="op">(</span>
  <span class="va">seurat_subset</span>,
  features <span class="op">=</span> <span class="fu"><a href="../reference/GetWGCNAGenes.html">GetWGCNAGenes</a></span><span class="op">(</span><span class="va">seurat_subset</span>, <span class="st">'SCT'</span><span class="op">)</span>,
  wgcna_name <span class="op">=</span> <span class="st">"RNA"</span>
<span class="op">)</span>

<span class="co"># construct metacells</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MetacellsByGroups.html">MetacellsByGroups</a></span><span class="op">(</span>
  seurat_obj <span class="op">=</span> <span class="va">seurat_subset</span>,
  group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sample"</span><span class="op">)</span>,
  k <span class="op">=</span> <span class="fl">25</span>,
  max_shared<span class="op">=</span><span class="fl">12</span>,
  min_cells <span class="op">=</span> <span class="fl">50</span>,
  reduction <span class="op">=</span> <span class="st">'harmony'</span>,
  ident.group <span class="op">=</span> <span class="st">'Sample'</span>,
  slot <span class="op">=</span> <span class="st">'counts'</span>,
  assay <span class="op">=</span> <span class="st">'RNA'</span>
<span class="op">)</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NormalizeMetacells.html">NormalizeMetacells</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SetDatExpr.html">SetDatExpr</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TestSoftPowers.html">TestSoftPowers</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>
<span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotSoftPowers.html">PlotSoftPowers</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="co"># plot softpowers</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">plot_list</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="figures/sctransform/test_softpower_RNA.png" width="800" height="900"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># construct wgcna network:</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ConstructNetwork.html">ConstructNetwork</a></span><span class="op">(</span>
  <span class="va">seurat_subset</span>,
  tom_name <span class="op">=</span> <span class="st">"RNA"</span>,
  overwrite_tom <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleEigengenes.html">ModuleEigengenes</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>
<span class="va">seurat_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ModuleConnectivity.html">ModuleConnectivity</a></span><span class="op">(</span><span class="va">seurat_subset</span><span class="op">)</span>

<span class="fu"><a href="../reference/PlotDendrogram.html">PlotDendrogram</a></span><span class="op">(</span><span class="va">seurat_subset</span>, main<span class="op">=</span><span class="st">'hdWGCNA RNA Dendrogram'</span><span class="op">)</span></code></pre></div>
<p><img src="figures/sctransform/dendro_RNA.png" width="800" height="900"></p>
</details><p>Now we can plot the modules from the RNA under the modules from SCT to compare.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get both sets of modules</span>
<span class="va">m1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_subset</span>, <span class="st">'SCT'</span><span class="op">)</span>
<span class="va">m2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_subset</span>, <span class="st">'RNA'</span><span class="op">)</span>

<span class="co"># get consensus dendrogram</span>
<span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetNetworkData.html">GetNetworkData</a></span><span class="op">(</span><span class="va">seurat_subset</span>, wgcna_name<span class="op">=</span><span class="st">"SCT"</span><span class="op">)</span>
<span class="va">dendro</span> <span class="op">&lt;-</span> <span class="va">net</span><span class="op">$</span><span class="va">dendrograms</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>

<span class="co"># get the gene and module color for consensus</span>
<span class="va">m2_genes</span> <span class="op">&lt;-</span> <span class="va">m2</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">m2_colors</span> <span class="op">&lt;-</span> <span class="va">m2</span><span class="op">$</span><span class="va">color</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">m2_colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">m2_genes</span>

<span class="co"># get the gene and module color for standard</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">m1</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">colors</span> <span class="op">&lt;-</span> <span class="va">m1</span><span class="op">$</span><span class="va">color</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">genes</span>

<span class="co"># re-order the genes to match the consensus genes</span>
<span class="va">colors</span> <span class="op">&lt;-</span> <span class="va">colors</span><span class="op">[</span><span class="va">m2_genes</span><span class="op">]</span>

<span class="co"># set up dataframe for plotting</span>
<span class="va">color_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  SCT <span class="op">=</span> <span class="va">colors</span>,
  RNA <span class="op">=</span> <span class="va">m2_colors</span>
<span class="op">)</span>

<span class="co"># plot dendrogram using WGCNA function</span>
<span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/plotDendroAndColors.html" class="external-link">plotDendroAndColors</a></span><span class="op">(</span>
  <span class="va">net</span><span class="op">$</span><span class="va">dendrograms</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  <span class="va">color_df</span>,
  groupLabels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">color_df</span><span class="op">)</span>,
  dendroLabels <span class="op">=</span> <span class="cn">FALSE</span>, hang <span class="op">=</span> <span class="fl">0.03</span>, addGuide <span class="op">=</span> <span class="cn">TRUE</span>, guideHang <span class="op">=</span> <span class="fl">0.05</span>,
  main <span class="op">=</span> <span class="st">"SCT Dendrogram vs RNA modules"</span>,
<span class="op">)</span></code></pre></div>
<p><img src="figures/sctransform/dendro_SCT_compare.png" width="800" height="900"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get both sets of modules</span>
<span class="va">m1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_subset</span>, <span class="st">'RNA'</span><span class="op">)</span>
<span class="va">m2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetModules.html">GetModules</a></span><span class="op">(</span><span class="va">seurat_subset</span>, <span class="st">'SCT'</span><span class="op">)</span>

<span class="co"># get consensus dendrogram</span>
<span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetNetworkData.html">GetNetworkData</a></span><span class="op">(</span><span class="va">seurat_subset</span>, wgcna_name<span class="op">=</span><span class="st">"RNA"</span><span class="op">)</span>
<span class="va">dendro</span> <span class="op">&lt;-</span> <span class="va">net</span><span class="op">$</span><span class="va">dendrograms</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>

<span class="co"># get the gene and module color for consensus</span>
<span class="va">m2_genes</span> <span class="op">&lt;-</span> <span class="va">m2</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">m2_colors</span> <span class="op">&lt;-</span> <span class="va">m2</span><span class="op">$</span><span class="va">color</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">m2_colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">m2_genes</span>

<span class="co"># get the gene and module color for standard</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">m1</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">colors</span> <span class="op">&lt;-</span> <span class="va">m1</span><span class="op">$</span><span class="va">color</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">colors</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">genes</span>

<span class="co"># re-order the genes to match the consensus genes</span>
<span class="va">colors</span> <span class="op">&lt;-</span> <span class="va">colors</span><span class="op">[</span><span class="va">m2_genes</span><span class="op">]</span>

<span class="co"># set up dataframe for plotting</span>
<span class="va">color_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  RNA <span class="op">=</span> <span class="va">colors</span>,
  SCT <span class="op">=</span> <span class="va">m2_colors</span>
<span class="op">)</span>

<span class="co"># plot dendrogram using WGCNA function</span>
<span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/plotDendroAndColors.html" class="external-link">plotDendroAndColors</a></span><span class="op">(</span>
  <span class="va">net</span><span class="op">$</span><span class="va">dendrograms</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
  <span class="va">color_df</span>,
  groupLabels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">color_df</span><span class="op">)</span>,
  dendroLabels <span class="op">=</span> <span class="cn">FALSE</span>, hang <span class="op">=</span> <span class="fl">0.03</span>, addGuide <span class="op">=</span> <span class="cn">TRUE</span>, guideHang <span class="op">=</span> <span class="fl">0.05</span>,
  main <span class="op">=</span> <span class="st">"RNA Dendrogram vs SCT modules"</span>,
<span class="op">)</span></code></pre></div>
<p><img src="figures/sctransform/dendro_RNA_compare.png" width="800" height="900"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
