<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ConstructNetwork — ConstructNetwork • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ConstructNetwork — ConstructNetwork"><meta name="description" content="Constructs a co-expression network and groups genes into modules
given a Seurat object that has been prepared for network analysis."><meta property="og:description" content="Constructs a co-expression network and groups genes into modules
given a Seurat object that has been prepared for network analysis."><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ConstructNetwork</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/ConstructNetwork.R" class="external-link"><code>R/ConstructNetwork.R</code></a></small>
      <div class="d-none name"><code>ConstructNetwork.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Constructs a co-expression network and groups genes into modules
given a Seurat object that has been prepared for network analysis.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ConstructNetwork</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  soft_power <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_power <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  tom_outdir <span class="op">=</span> <span class="st">"TOM"</span>,</span>
<span>  tom_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  consensus <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  overwrite_tom <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  blocks <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxBlockSize <span class="op">=</span> <span class="fl">30000</span>,</span>
<span>  randomSeed <span class="op">=</span> <span class="fl">12345</span>,</span>
<span>  corType <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  consensusQuantile <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  networkType <span class="op">=</span> <span class="st">"signed"</span>,</span>
<span>  TOMType <span class="op">=</span> <span class="st">"signed"</span>,</span>
<span>  TOMDenom <span class="op">=</span> <span class="st">"min"</span>,</span>
<span>  scaleTOMs <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  calibrationQuantile <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  sampleForCalibration <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sampleForCalibrationFactor <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  useDiskCache <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  chunkSize <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  deepSplit <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  pamStage <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  detectCutHeight <span class="op">=</span> <span class="fl">0.995</span>,</span>
<span>  minModuleSize <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  mergeCutHeight <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  saveConsensusTOMs <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object</p></dd>


<dt id="arg-soft-power">soft_power<a class="anchor" aria-label="anchor" href="#arg-soft-power"></a></dt>
<dd><p>the soft power used for network construction. Automatically selected by default.</p></dd>


<dt id="arg-min-power">min_power<a class="anchor" aria-label="anchor" href="#arg-min-power"></a></dt>
<dd><p>the smallest soft power to be selected if soft_power=NULL</p></dd>


<dt id="arg-tom-outdir">tom_outdir<a class="anchor" aria-label="anchor" href="#arg-tom-outdir"></a></dt>
<dd><p>path to the directory where the TOM will be written</p></dd>


<dt id="arg-tom-name">tom_name<a class="anchor" aria-label="anchor" href="#arg-tom-name"></a></dt>
<dd><p>prefix name given to the TOM output file</p></dd>


<dt id="arg-consensus">consensus<a class="anchor" aria-label="anchor" href="#arg-consensus"></a></dt>
<dd><p>flag indicating whether or not to perform Consensus network analysis</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>name of the WGCNA experiment</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>additional parameters passed to blockwiseConsensusModules</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>seurat_obj with the co-expression network and gene modules computed for the selected wgcna experiment</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>ConstructNetwork builds a co-expression network and identifies clusters of
highly co-expressed genes (modules) from the metacell or metaspot
expression matrix stored in the Seurat object. Before running this function,
the following functions must be run on the input Seurat object:</p><ol><li><p>SetupForWGCNA</p></li>
<li><p>MetacellsByGroups or MetaspotsByGroups, and NormalizeMetacells</p></li>
<li><p>SetDatExpr or SetMultiExpr</p></li>
<li><p>TestSoftPowers or TestSoftPowersConsensus</p></li>
</ol><p>This function can also be used to perform consensus network analysis if consensus=TRUE
is selected. ConstructNetwork calls the WGCNA function blockwiseConsensusModules
to compute the adjacency matrix, topological overlap matrix, and to run the
Dynamic Tree Cut algorithm to identify gene modules. blockwiseConsensusModules
has numerous parameters but here we have selected default parameters that we
have found to provide reasonable results on a variety of single-cell and
spatial transcriptomic datasets.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

