<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>FindMajorIsoforms — FindMajorIsoforms • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="FindMajorIsoforms — FindMajorIsoforms"><meta name="description" content="Finds the set of &quot;major isoforms&quot; for each gene in each cell population. The set of
major isoforms consists of the isoforms accounting for a desired proportion of a gene's
expression. Pseudobulk replicates in each sample and cell population are used for this calculation."><meta property="og:description" content="Finds the set of &quot;major isoforms&quot; for each gene in each cell population. The set of
major isoforms consists of the isoforms accounting for a desired proportion of a gene's
expression. Pseudobulk replicates in each sample and cell population are used for this calculation."><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>FindMajorIsoforms</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/FindMajorIsoforms.R" class="external-link"><code>R/FindMajorIsoforms.R</code></a></small>
      <div class="d-none name"><code>FindMajorIsoforms.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Finds the set of "major isoforms" for each gene in each cell population. The set of
major isoforms consists of the isoforms accounting for a desired proportion of a gene's
expression. Pseudobulk replicates in each sample and cell population are used for this calculation.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">FindMajorIsoforms</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  <span class="va">group.by</span>,</span>
<span>  <span class="va">replicate_col</span>,</span>
<span>  isoform_delim <span class="op">=</span> <span class="st">"[.]"</span>,</span>
<span>  proportion_thresh <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  low_thresh <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"iso"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  cluster_markers <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object</p></dd>


<dt id="arg-group-by">group.by<a class="anchor" aria-label="anchor" href="#arg-group-by"></a></dt>
<dd><p>column in seurat_obj@meta.data containing grouping info, ie clusters or celltypes</p></dd>


<dt id="arg-replicate-col">replicate_col<a class="anchor" aria-label="anchor" href="#arg-replicate-col"></a></dt>
<dd><p>column in seurat_obj@meta.data denoting each replicate / sample</p></dd>


<dt id="arg-isoform-delim">isoform_delim<a class="anchor" aria-label="anchor" href="#arg-isoform-delim"></a></dt>
<dd></dd>


<dt id="arg-proportion-thresh">proportion_thresh<a class="anchor" aria-label="anchor" href="#arg-proportion-thresh"></a></dt>
<dd><p>desired proportion of expression to define the set of major isoforms. Default = 0.8.</p></dd>


<dt id="arg-low-thresh">low_thresh<a class="anchor" aria-label="anchor" href="#arg-low-thresh"></a></dt>
<dd><p>lower bound of expression level for considering an isoform as part of the major isoform set.</p></dd>


<dt id="arg-assay">assay<a class="anchor" aria-label="anchor" href="#arg-assay"></a></dt>
<dd><p>Assay in seurat_obj containing isoform expression information.</p></dd>


<dt id="arg-slot">slot<a class="anchor" aria-label="anchor" href="#arg-slot"></a></dt>
<dd><p>Slot in seurat_obj, default to counts slot.</p></dd>


<dt id="arg-cluster-markers">cluster_markers<a class="anchor" aria-label="anchor" href="#arg-cluster-markers"></a></dt>
<dd><p>Cell population marker gene table from Seurat FindAllMarkers for the same cell populations specified in group.by. Optional parameter, will exclude isoforms that are not from marker genes.</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_obj@misc slot</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a list of major isoforms for each cell population</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>FindMajorIsoforms computes the set of major isoforms in a given Seurat object that contains isoform-level
expression information. First, pseudobulk replicates are computed for the given cell populations and samples
present in the Seurat object. For each gene in each cell population, we rank the gene's isoforms by expression
level and take the top expressing isoforms that make up the desired proportion of the gene's total expression,
making sure to exclude any very lowly expressed isoforms.</p>
<p>Optionally, the user may supply a marker gene table for each cell population (formatted like the output of Seurat
FindAllMarkers), and then the algorithm will only return major isoforms of the given marker genes.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

