<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>MetacellsByGroups — MetacellsByGroups • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="MetacellsByGroups — MetacellsByGroups"><meta property="og:description" content="This function takes a Seurat object and constructs averaged 'metacells' based
on neighboring cells in provided groupings, such as cluster or cell type."><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hdWGCNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.02</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/hdWGCNA.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-header">Core functionality</li>
    <li>
      <a href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a>
    </li>
    <li>
      <a href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a>
    </li>
    <li>
      <a href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a>
    </li>
    <li>
      <a href="../articles/network_visualizations.html">Network visualization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Biological context for co-expression modules</li>
    <li>
      <a href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a>
    </li>
    <li>
      <a href="../articles/module_trait_correlation.html">Module trait correlation</a>
    </li>
    <li>
      <a href="../articles/enrichment_analysis.html">Enrichment analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Exploring modules in external datasets</li>
    <li>
      <a href="../articles/projecting_modules.html">Projecting modules to new datasets</a>
    </li>
    <li>
      <a href="../articles/module_preservation.html">Module preservation and reproducibility</a>
    </li>
    <li>
      <a href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Advanced topics</li>
    <li>
      <a href="../articles/consensus_wgcna.html">Consensus network analysis</a>
    </li>
    <li>
      <a href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a>
    </li>
    <li>
      <a href="../articles/motif_analysis.html">Motif analysis</a>
    </li>
    <li>
      <a href="../articles/other_metacells.html">Alternative metacell algorithms</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Other</li>
    <li>
      <a href="../articles/customization.html">Module customization</a>
    </li>
    <li>
      <a href="../articles/sctransform.html">Using SCTransform normalized data</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>MetacellsByGroups</h1>
    
    <div class="hidden name"><code>MetacellsByGroups.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function takes a Seurat object and constructs averaged 'metacells' based
on neighboring cells in provided groupings, such as cluster or cell type.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">MetacellsByGroups</span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"seurat_clusters"</span><span class="op">)</span>,
  ident.group <span class="op">=</span> <span class="st">"seurat_clusters"</span>,
  k <span class="op">=</span> <span class="fl">25</span>,
  reduction <span class="op">=</span> <span class="st">"pca"</span>,
  dims <span class="op">=</span> <span class="cn">NULL</span>,
  assay <span class="op">=</span> <span class="cn">NULL</span>,
  slot <span class="op">=</span> <span class="st">"counts"</span>,
  layer <span class="op">=</span> <span class="st">"counts"</span>,
  mode <span class="op">=</span> <span class="st">"average"</span>,
  cells.use <span class="op">=</span> <span class="cn">NULL</span>,
  min_cells <span class="op">=</span> <span class="fl">100</span>,
  max_shared <span class="op">=</span> <span class="fl">15</span>,
  target_metacells <span class="op">=</span> <span class="fl">1000</span>,
  max_iter <span class="op">=</span> <span class="fl">5000</span>,
  verbose <span class="op">=</span> <span class="cn">FALSE</span>,
  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seurat_obj</dt>
<dd><p>A Seurat object</p></dd>
<dt>group.by</dt>
<dd><p>A character vector of Seurat metadata column names representing groups for which metacells will be computed.</p></dd>
<dt>k</dt>
<dd><p>Number of nearest neighbors to aggregate. Default = 50</p></dd>
<dt>reduction</dt>
<dd><p>A dimensionality reduction stored in the Seurat object. Default = 'pca'</p></dd>
<dt>dims</dt>
<dd><p>A vector represnting the dimensions of the reduction to use. Either specify the names of the dimensions or the indices. Default = NULL to include all dims.</p></dd>
<dt>assay</dt>
<dd><p>Assay to extract data for aggregation. Default = 'RNA'</p></dd>
<dt>slot</dt>
<dd><p>Slot to extract data for aggregation. Default = 'counts'. Slot is used with Seurat v4 instead of layer.</p></dd>
<dt>layer</dt>
<dd><p>Layer to extract data for aggregation. Default = 'counts'. Layer is used with Seurat v5 instead of slot.</p></dd>
<dt>mode</dt>
<dd><p>determines how to make gene expression profiles for metacells from their constituent single cells. Options are "average" or "sum".</p></dd>
<dt>min_cells</dt>
<dd><p>the minimum number of cells in a particular grouping to construct metacells</p></dd>
<dt>max_shared</dt>
<dd><p>the maximum number of cells to be shared across two metacells</p></dd>
<dt>target_metacells</dt>
<dd><p>the maximum target number of metacells to construct</p></dd>
<dt>max_iter</dt>
<dd><p>the maximum number of iterations in the metacells bootstrapping loop</p></dd>
<dt>verbose</dt>
<dd><p>logical indicating whether to print additional information</p></dd>
<dt>wgcna_name</dt>
<dd><p>name of the WGCNA experiment</p></dd>
<dt>name</dt>
<dd><p>A string appended to resulting metalcells. Default = 'agg'</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>seurat_obj with a metacell seurat_obj stored in the specified WGCNA experiment</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>MetacellsByGroups merges transcriptomically similar cells into "metacells".
Given a dimensionally-reduced representation of the input dataset, this algorithm
first uses KNN to identify similar cells. A bootstrapped sampling procedure is then
used to group together similar cells until convergence is reached. Importantly,
this procedure is done in a context-specific manner based on the provided group.by parameters.
Typically this means that metacells will be constructed separately for each biological
replicate, cell type or cell state, disease condition, etc. The metacell representation is
considerably less sparse than the original single-cell dataset, which is preferable for
co-expression network analysis or othter analyses that rely on correlations.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

