<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ModuleFeaturePlot — ModuleFeaturePlot • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleFeaturePlot — ModuleFeaturePlot"><meta name="description" content="Plot module eigengenes as a FeaturePlot"><meta property="og:description" content="Plot module eigengenes as a FeaturePlot"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ModuleFeaturePlot</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/plotting.R" class="external-link"><code>R/plotting.R</code></a></small>
      <div class="d-none name"><code>ModuleFeaturePlot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plot module eigengenes as a FeaturePlot</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ModuleFeaturePlot</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  module_names <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">"umap"</span>,</span>
<span>  features <span class="op">=</span> <span class="st">"hMEs"</span>,</span>
<span>  order_points <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  restrict_range <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  label_legend <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ucell <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  raster_dpi <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  raster_scale <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  plot_ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  title <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object</p></dd>


<dt id="arg-features">features<a class="anchor" aria-label="anchor" href="#arg-features"></a></dt>
<dd><p>What to plot? Can select hMEs, MEs, scores, or average</p></dd>


<dt id="arg-order">order<a class="anchor" aria-label="anchor" href="#arg-order"></a></dt>
<dd><p>TRUE, FALSE, or "shuffle" are valid options</p></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">ModuleFeaturePlot</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, module_names = NULL, wgcna_name = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     reduction = "umap", features = "hMEs", order_points = TRUE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     restrict_range = TRUE, point_size = 0.5, alpha = 1, label_legend = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ucell = FALSE, raster = FALSE, raster_dpi = 500, raster_scale = 1, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot_ratio = 1, title = TRUE) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (features == "hMEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, TRUE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "MEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, FALSE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "scores") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetModuleScores(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "average") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetAvgModuleExpr(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         restrict_range &lt;- FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else (stop("Invalid feature selection. Valid choices: hMEs, MEs, scores, average"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (ucell) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         restrict_range &lt;- FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(module_names)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         module_names &lt;- levels(modules$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         module_names &lt;- module_names[module_names != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     umap &lt;- seurat_obj@reductions[[reduction]]@cell.embeddings</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     x_name &lt;- colnames(umap)[1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     y_name &lt;- colnames(umap)[2]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot_df &lt;- cbind(umap, MEs) %&gt;% as.data.frame()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot_list &lt;- list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     for (cur_mod in module_names) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         print(cur_mod)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_color &lt;- modules %&gt;% subset(module == cur_mod) %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             .$color %&gt;% unique</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         plot_range &lt;- plot_df[, cur_mod] %&gt;% range</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (restrict_range) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             if (abs(plot_range[1]) &gt; abs(plot_range[2])) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 plot_range[1] &lt;- -1 * plot_range[2]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 plot_range[2] &lt;- -1 * plot_range[1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             plot_df[, cur_mod] &lt;- ifelse(plot_df[, cur_mod] &gt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 plot_range[2], plot_range[2], plot_df[, cur_mod])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             plot_df[, cur_mod] &lt;- ifelse(plot_df[, cur_mod] &lt; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 plot_range[1], plot_range[1], plot_df[, cur_mod])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_plot_df &lt;- plot_df[, c(colnames(umap), cur_mod)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         colnames(cur_plot_df)[3] &lt;- "val"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (order_points == TRUE) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_plot_df &lt;- cur_plot_df %&gt;% dplyr::arrange(val)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else if (order_points == "shuffle") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_plot_df &lt;- cur_plot_df[sample(nrow(cur_plot_df)), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p &lt;- cur_plot_df %&gt;% ggplot(aes_string(x = x_name, y = y_name, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             color = "val"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (raster) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p &lt;- p + ggrastr::rasterise(geom_point(size = point_size, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 alpha = alpha), dpi = raster_dpi, scale = raster_scale)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p &lt;- p + geom_point(size = point_size, alpha = alpha)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p &lt;- p + umap_theme() + labs(color = "")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (title) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p &lt;- p + ggtitle(cur_mod)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (is.numeric(plot_ratio)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p &lt;- p + coord_fixed(ratio = plot_ratio)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (!ucell) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p &lt;- p + scale_color_gradient2(low = "grey75", mid = "grey95", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 high = cur_color, breaks = plot_range, labels = c("-", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   "+"), guide = guide_colorbar(ticks = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   barwidth = 0.5, barheight = 4))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p &lt;- p + scale_color_gradient(low = "grey95", high = cur_color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 breaks = plot_range, labels = c("0", "+"), guide = guide_colorbar(ticks = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   barwidth = 0.5, barheight = 4))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         plot_list[[cur_mod]] &lt;- p</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (length(plot_list) == 1) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p &lt;- plot_list[[1]]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p &lt;- plot_list</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7f9629c4e8e8&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

