<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ModuleCorrNetworks — ModuleCorrNetwork • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleCorrNetworks — ModuleCorrNetwork"><meta name="description" content="Plot Module Eigengene correlogram"><meta property="og:description" content="Plot Module Eigengene correlogram"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ModuleCorrNetworks</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/plotting.R" class="external-link"><code>R/plotting.R</code></a></small>
      <div class="d-none name"><code>ModuleCorrNetwork.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plot Module Eigengene correlogram</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ModuleCorrNetwork</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  cluster_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  exclude_grey <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  features <span class="op">=</span> <span class="st">"hMEs"</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">"umap"</span>,</span>
<span>  cor_cutoff <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  label_vertices <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  edge_scale <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  vertex_size <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  niter <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  vertex_frame <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object</p></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">ModuleCorrNetwork</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, cluster_col = NULL, exclude_grey = TRUE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     features = "hMEs", reduction = "umap", cor_cutoff = 0.2, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     label_vertices = FALSE, edge_scale = 5, vertex_size = 15, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     niter = 100, vertex_frame = FALSE, wgcna_name = NULL) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (features == "hMEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, TRUE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "MEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, FALSE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "scores") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetModuleScores(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "average") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetAvgModuleExpr(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         restrict_range &lt;- FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else (stop("Invalid feature selection. Valid choices: hMEs, MEs, scores, average"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (exclude_grey) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- MEs[, colnames(MEs) != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         modules &lt;- modules %&gt;% subset(color != "grey")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- colnames(MEs)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(cluster_col)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clusters &lt;- Idents(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clusters &lt;- droplevels(seurat_obj@meta.data[[cluster_col]])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     MEs$cluster &lt;- clusters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cluster_ME_av &lt;- do.call(rbind, lapply(split(MEs, MEs$cluster), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             colMeans(x[, mods])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         })) %&gt;% as.data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     top_clusters &lt;- sapply(mods, function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rownames(cluster_ME_av[cluster_ME_av[, x] == max(cluster_ME_av[, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             x]), ])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     })</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     red_df &lt;- as.data.frame(seurat_obj@reductions[[reduction]]@cell.embeddings)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     red_df$cluster &lt;- clusters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     red_av &lt;- do.call(rbind, lapply(split(red_df, red_df$cluster), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             colMeans(x[, 1:2])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         })) %&gt;% as.data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_mat &lt;- Hmisc::rcorr(as.matrix(MEs[, 1:ncol(MEs) - 1]))$r</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_mat[lower.tri(cor_mat)] &lt;- NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_df &lt;- reshape2::melt(cor_mat) %&gt;% na.omit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_df &lt;- cor_df %&gt;% subset(!(Var1 == Var2))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_df &lt;- cor_df %&gt;% subset(abs(value) &gt;= cor_cutoff)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     v_df &lt;- data.frame(name = mods, cluster = as.character(top_clusters))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     unique_mods &lt;- dplyr::distinct(modules[, c("module", "color")])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rownames(unique_mods) &lt;- unique_mods$module</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     v_df$color &lt;- unique_mods[v_df$name, "color"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     v_df$x &lt;- red_av[v_df$cluster, 1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     v_df$y &lt;- red_av[v_df$cluster, 2]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     g &lt;- igraph::graph_from_data_frame(cor_df, directed = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertices = v_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     e &lt;- get.edgelist(g, name = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     l &lt;- qgraph::qgraph.layout.fruchtermanreingold(e, vcount = vcount(g), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         weights = igraph::E(g)$value, repulse.rad = (vcount(g)), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         niter = niter, )</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot_df &lt;- rbind(cor_df, data.frame(Var1 = c("x", "y"), Var2 = c("y", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "x"), value = c(-1, 1)))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temp &lt;- ggplot(plot_df, aes(x = value, y = value, color = value)) + </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         geom_point() + scale_color_gradient2(high = "darkorchid1", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mid = "white", low = "seagreen", midpoint = 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temp &lt;- ggplot_build(temp)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     igraph::E(g)$color &lt;- temp$data[[1]]$colour[1:nrow(cor_df)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (label_vertices) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         labels &lt;- igraph::V(g)$name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         labels &lt;- NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (vertex_frame) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         frame_color &lt;- "black"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         frame_color &lt;- igraph::V(g)$color</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot(g, layout = l, edge.color = igraph::E(g)$color, edge.curved = 0, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         edge.width = abs(igraph::E(g)$value) * edge_scale, vertex.color = igraph::V(g)$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.frame.color = frame_color, vertex.label = labels, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.family = "Helvetica", vertex.label.color = "black", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.cex = 0.5, vertex.size = vertex_size)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7f9151688808&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

