<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ModuleTFNetwork — ModuleTFNetwork • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleTFNetwork — ModuleTFNetwork"><meta name="description" content="Plotting the relationships between a TF and the co-expression modules"><meta property="og:description" content="Plotting the relationships between a TF and the co-expression modules"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ModuleTFNetwork</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/plotting.R" class="external-link"><code>R/plotting.R</code></a></small>
      <div class="d-none name"><code>ModuleTFNetwork.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plotting the relationships between a TF and the co-expression modules</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ModuleTFNetwork</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  <span class="va">tf_name</span>,</span>
<span>  <span class="va">tf_gene_name</span>,</span>
<span>  edge.alpha <span class="op">=</span> <span class="fl">0.75</span>,</span>
<span>  cor_thresh <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  mid_color <span class="op">=</span> <span class="st">"grey"</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"blue"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span>
<span>  size.scale <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  tf_x <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  tf_y <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object</p></dd>


<dt id="arg-tf-name">tf_name<a class="anchor" aria-label="anchor" href="#arg-tf-name"></a></dt>
<dd><p>the Motif name for the tf</p></dd>


<dt id="arg-tf-gene-name">tf_gene_name<a class="anchor" aria-label="anchor" href="#arg-tf-gene-name"></a></dt>
<dd><p>the gene associated with this tf in the rownames(seurat_obj)</p></dd>


<dt id="arg-edge-alpha">edge.alpha<a class="anchor" aria-label="anchor" href="#arg-edge-alpha"></a></dt>
<dd><p>scaling factor for edge opacity in the network</p></dd>


<dt id="arg-cor-thresh">cor_thresh<a class="anchor" aria-label="anchor" href="#arg-cor-thresh"></a></dt>
<dd><p>threshold to plot correlation edges between modules</p></dd>


<dt id="arg-high-color">high_color<a class="anchor" aria-label="anchor" href="#arg-high-color"></a></dt>
<dd><p>color for positive correlation</p></dd>


<dt id="arg-mid-color">mid_color<a class="anchor" aria-label="anchor" href="#arg-mid-color"></a></dt>
<dd><p>color for zero correlation</p></dd>


<dt id="arg-low-color">low_color<a class="anchor" aria-label="anchor" href="#arg-low-color"></a></dt>
<dd><p>color for negative correlation</p></dd>


<dt id="arg-slot">slot<a class="anchor" aria-label="anchor" href="#arg-slot"></a></dt>
<dd><p>the slot in the seurat object to extract expression data for the tf_gene_name</p></dd>


<dt id="arg-size-scale">size.scale<a class="anchor" aria-label="anchor" href="#arg-size-scale"></a></dt>
<dd><p>scaling factor for the size of each node</p></dd>


<dt id="arg-tf-x">tf_x<a class="anchor" aria-label="anchor" href="#arg-tf-x"></a></dt>
<dd><p>x coordinate for the TF if the TF is not found in the UMAP</p></dd>


<dt id="arg-tf-y">tf_y<a class="anchor" aria-label="anchor" href="#arg-tf-y"></a></dt>
<dd><p>y coordinate for the TF if the TF is not foudn in the UMAP</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>the name of the WGCNA experiment in the seurat object</p></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">ModuleTFNetwork</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, tf_name, tf_gene_name, edge.alpha = 0.75, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_thresh = 0.25, high_color = "red", mid_color = "grey", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     low_color = "blue", slot = "data", size.scale = 30, tf_x = 0, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf_y = 0, wgcna_name = NULL) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name) %&gt;% subset(module != </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "grey") %&gt;% mutate(module = droplevels(module))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     MEs &lt;- GetMEs(seurat_obj, TRUE, wgcna_name) %&gt;% as.matrix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     MEs &lt;- MEs[, colnames(MEs) != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mod_sizes &lt;- table(modules$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor &lt;- Hmisc::rcorr(x = MEs, type = "pearson")$r</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor[lower.tri(module_cor)] &lt;- NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor &lt;- reshape2::melt(module_cor) %&gt;% na.omit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor &lt;- subset(module_cor, abs(value) &gt;= cor_thresh &amp; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Var1 != Var2)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_exp &lt;- GetAssayData(seurat_obj, slot = slot)[tf_gene_name, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     exp_cor &lt;- Hmisc::rcorr(x = MEs, y = cur_exp)$r[1:ncol(MEs), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "y"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     exp_cor &lt;- data.frame(mod = names(exp_cor), value = as.numeric(exp_cor))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot_lim &lt;- abs(max(c(abs(range(exp_cor$value)), abs(range(module_cor$value)))))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p &lt;- ggplot(module_cor, aes(x = Var1, y = Var2, color = value)) + </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         geom_point() + scale_color_gradient2(high = high_color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mid = mid_color, low = low_color, limits = c(-1 * plot_lim, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             plot_lim))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ggp &lt;- ggplot_build(p)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor$color &lt;- ggp$data[[1]]$colour</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p &lt;- ggplot(exp_cor, aes(x = mod, y = mod, color = value)) + </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         geom_point() + scale_color_gradient2(high = high_color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mid = mid_color, low = low_color, limits = c(-1 * plot_lim, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             plot_lim))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ggp &lt;- ggplot_build(p)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     exp_cor$color &lt;- ggp$data[[1]]$colour</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     umap_df &lt;- GetModuleUMAP(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- levels(umap_df$modules)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf_match &lt;- GetMotifMatrix(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf_targets &lt;- GetMotifTargets(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     motif_df &lt;- GetMotifs(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     overlap_df &lt;- GetMotifOverlap(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     centroid_df &lt;- umap_df %&gt;% dplyr::select(c(UMAP1, UMAP2, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         module)) %&gt;% dplyr::group_by(module) %&gt;% dplyr::summarise(x = mean(UMAP1), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         y = mean(UMAP2))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_overlap &lt;- subset(overlap_df, tf == tf_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     node_df &lt;- dplyr::left_join(centroid_df, cur_overlap, by = "module") %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dplyr::rename(c(UMAP1 = x, UMAP2 = y, name = module))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     node_df$size &lt;- as.numeric(node_df$size_intersection)/as.numeric(mod_sizes)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (tf_gene_name %in% umap_df$gene) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tf_df &lt;- umap_df[umap_df$gene == tf_gene_name, ] %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             dplyr::select(-c(hub, kME, module)) %&gt;% dplyr::rename(name = gene)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tf_df &lt;- data.frame(name = tf_gene_name, module = "grey", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             color = "grey", UMAP1 = tf_x, UMAP2 = tf_y)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf_df$size &lt;- 0.25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- data.frame(Var1 = tf_gene_name, Var2 = as.character(node_df$name), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         value = node_df$odds_ratio, color = exp_cor$color)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df$color &lt;- ifelse(node_df$fdr &lt;= 0.05, edge_df$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "grey")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- subset(edge_df, color != "grey")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     node_df &lt;- dplyr::bind_rows(node_df, tf_df) %&gt;% as.data.frame()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rownames(node_df) &lt;- as.character(node_df$name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     g1 &lt;- igraph::graph_from_data_frame(edge_df, directed = TRUE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertices = node_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     g2 &lt;- igraph::graph_from_data_frame(module_cor, directed = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertices = node_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot(g2, layout = as.matrix(node_df[, c("UMAP1", "UMAP2")]), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.size = 1, edge.curved = 0, edge.width = 1, vertex.color = "grey", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label = "", edge.color = adjustcolor(igraph::E(g2)$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             alpha.f = edge.alpha), )</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot(g1, layout = as.matrix(node_df[, c("UMAP1", "UMAP2")]), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         edge.color = adjustcolor(igraph::E(g1)$color), vertex.size = igraph::V(g1)$size * </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             size.scale, edge.curved = 0, edge.width = edge_df$value * </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             2, vertex.color = igraph::V(g1)$color, vertex.label = igraph::V(g1)$name, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.dist = 1.1, vertex.label.degree = -pi/4, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.family = "Helvetica", vertex.label.font = 3, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.color = "black", vertex.label.cex = 0, vertex.frame.color = "black", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         margin = 0, edge.arrow.size = edge_df$value/2, add = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7fced738ad68&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

