<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ModuleTopologyHeatmap — ModuleTopologyHeatmap • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleTopologyHeatmap — ModuleTopologyHeatmap"><meta name="description" content="Plots a heatmap of the co-expression network topology of a given module."><meta property="og:description" content="Plots a heatmap of the co-expression network topology of a given module."><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ModuleTopologyHeatmap</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/plotting.R" class="external-link"><code>R/plotting.R</code></a></small>
      <div class="d-none name"><code>ModuleTopologyHeatmap.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plots a heatmap of the co-expression network topology of a given module.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ModuleTopologyHeatmap</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  <span class="va">mod</span>,</span>
<span>  matrix <span class="op">=</span> <span class="st">"TOM"</span>,</span>
<span>  matrix_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  order_by <span class="op">=</span> <span class="st">"kME"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  raster_dpi <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  plot_max <span class="op">=</span> <span class="st">"q99"</span>,</span>
<span>  plot_min <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  return_genes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  genes_order <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  TOM_use <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object</p></dd>


<dt id="arg-mod">mod<a class="anchor" aria-label="anchor" href="#arg-mod"></a></dt>
<dd><p>the name of the co-expression module to plot</p></dd>


<dt id="arg-matrix">matrix<a class="anchor" aria-label="anchor" href="#arg-matrix"></a></dt>
<dd><p>specify which matrix to plot, use 'TOM' (topological overlap matrix) or 'Cor' (correlation matrix), or pass a custom square matrix where the rownames and colnames match the genes in this module</p></dd>


<dt id="arg-matrix-name">matrix_name<a class="anchor" aria-label="anchor" href="#arg-matrix-name"></a></dt>
<dd><p>name of the matrix plotted that will be used as the label in the plot legend</p></dd>


<dt id="arg-order-by">order_by<a class="anchor" aria-label="anchor" href="#arg-order-by"></a></dt>
<dd><p>order genes in this module by 'kME' (default) or by 'degree' (sum of all connections to this gene in the co-expression network)</p></dd>


<dt id="arg-high-color">high_color<a class="anchor" aria-label="anchor" href="#arg-high-color"></a></dt>
<dd><p>color used for high values in the heatmap, default is the module's unique color</p></dd>


<dt id="arg-low-color">low_color<a class="anchor" aria-label="anchor" href="#arg-low-color"></a></dt>
<dd><p>color used for low values in the heatmap, default is 'white'</p></dd>


<dt id="arg-raster">raster<a class="anchor" aria-label="anchor" href="#arg-raster"></a></dt>
<dd><p>logical indicating whether or not to rasterise the plot</p></dd>


<dt id="arg-raster-dpi">raster_dpi<a class="anchor" aria-label="anchor" href="#arg-raster-dpi"></a></dt>
<dd><p>dpi used for a rasterised plot</p></dd>


<dt id="arg-plot-max">plot_max<a class="anchor" aria-label="anchor" href="#arg-plot-max"></a></dt>
<dd><p>maximum value to plot on the heatmap, can pass a numeric value or a string indicating the quantile ('q99' would be the 99th percentile)</p></dd>


<dt id="arg-plot-min">plot_min<a class="anchor" aria-label="anchor" href="#arg-plot-min"></a></dt>
<dd><p>minimum value to plot on the heatmap, can pass a numeric value or a string indicating the quantile ('q1' would be the 1st percentile)</p></dd>


<dt id="arg-return-genes">return_genes<a class="anchor" aria-label="anchor" href="#arg-return-genes"></a></dt>
<dd><p>logical indicating whether or not to return</p></dd>


<dt id="arg-genes-order">genes_order<a class="anchor" aria-label="anchor" href="#arg-genes-order"></a></dt>
<dd><p>a character vector of genes to plot in this specific order, this option will override the order_by parameter</p></dd>


<dt id="arg-tom-use">TOM_use<a class="anchor" aria-label="anchor" href="#arg-tom-use"></a></dt>
<dd><p>The name of the hdWGCNA experiment containing the TOM that will be used for plotting</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_obj@misc slot</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>ggplot object containing the ModuleTopologyHeatmap</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>ModuleTopologyHeatmap generates a triangular heatmap plot showing the network "topology" of a
specific co-expression module. Each cell in the heatmap represents a gene-gene pair, and the
the heatmap is colored by the strength of the connection between these two genes. By default
the genes in this heatmap are ordered in both the rows and the columns based on their importance
in the module, ranked either by eigengene-based connectivity (kME) or by network degree.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

