<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>ModuleTraitCorrelation' — ModuleTraitCorrelation • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleTraitCorrelation' — ModuleTraitCorrelation"><meta property="og:description" content="Correlates categorical and numeric variables with Module Eigengenes or hub-gene scores."><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hdWGCNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.03</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/hdWGCNA.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-header">Core functionality</li>
    <li>
      <a href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a>
    </li>
    <li>
      <a href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a>
    </li>
    <li>
      <a href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a>
    </li>
    <li>
      <a href="../articles/network_visualizations.html">Network visualization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Biological context for co-expression modules</li>
    <li>
      <a href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a>
    </li>
    <li>
      <a href="../articles/module_trait_correlation.html">Module trait correlation</a>
    </li>
    <li>
      <a href="../articles/enrichment_analysis.html">Enrichment analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Exploring modules in external datasets</li>
    <li>
      <a href="../articles/projecting_modules.html">Projecting modules to new datasets</a>
    </li>
    <li>
      <a href="../articles/module_preservation.html">Module preservation and reproducibility</a>
    </li>
    <li>
      <a href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Advanced topics</li>
    <li>
      <a href="../articles/consensus_wgcna.html">Consensus network analysis</a>
    </li>
    <li>
      <a href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a>
    </li>
    <li>
      <a href="../articles/motif_analysis.html">Motif analysis</a>
    </li>
    <li>
      <a href="../articles/other_metacells.html">Alternative metacell algorithms</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Other</li>
    <li>
      <a href="../articles/customization.html">Module customization</a>
    </li>
    <li>
      <a href="../articles/sctransform.html">Using SCTransform normalized data</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>ModuleTraitCorrelation'</h1>
    
    <div class="hidden name"><code>ModuleTraitCorrelation.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Correlates categorical and numeric variables with Module Eigengenes or hub-gene scores.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">ModuleTraitCorrelation</span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  <span class="va">traits</span>,
  group.by <span class="op">=</span> <span class="cn">NULL</span>,
  features <span class="op">=</span> <span class="st">"hMEs"</span>,
  cor_method <span class="op">=</span> <span class="st">"pearson"</span>,
  subset_by <span class="op">=</span> <span class="cn">NULL</span>,
  subset_groups <span class="op">=</span> <span class="cn">NULL</span>,
  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seurat_obj</dt>
<dd><p>A list of column names in the Seurat object's metadata that you wish to correlate with each module.
Traits must be a categorical variable (not a character vector), or a numeric variable.</p></dd>
<dt>features</dt>
<dd><p>Which features to use to summarize each modules? Valid choices are hMEs, MEs, or scores</p></dd>
<dt>wgcna_name</dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_obj@misc slot</p></dd>
<dt>cor_meth</dt>
<dd><p>Which method to use for correlation? Valid choices are pearson, spearman, kendall.</p></dd>
</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">ModuleTraitCorrelation</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, traits, group.by = NULL, features = "hMEs", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_method = "pearson", subset_by = NULL, subset_groups = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     wgcna_name = NULL, ...) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     CheckWGCNAName(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (features == "hMEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, TRUE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "MEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, FALSE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "scores") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetModuleScores(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         stop("Invalid feature selection. Valid choices: hMEs, MEs, scores, average")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!is.null(subset_by)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         print("subsetting")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_full &lt;- seurat_obj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- MEs[seurat_obj@meta.data[[subset_by]] %in% subset_groups, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj &lt;- seurat_obj[, seurat_obj@meta.data[[subset_by]] %in% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             subset_groups]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (sum(traits %in% colnames(seurat_obj@meta.data)) != length(traits)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         stop(paste("Some of the provided traits were not found in the Seurat obj:", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             paste(traits[!(traits %in% colnames(seurat_obj@meta.data))], </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 collapse = ", ")))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(group.by)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         group.by &lt;- "temp_ident"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj$temp_ident &lt;- Idents(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     valid_types &lt;- c("numeric", "factor", "integer")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     data_types &lt;- sapply(traits, function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class(seurat_obj@meta.data[, x])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     })</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!all(data_types %in% valid_types)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         incorrect &lt;- traits[!(data_types %in% valid_types)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         stop(paste0("Invalid data types for ", paste(incorrect, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             collapse = ", "), ". Accepted data types are numeric, factor, integer."))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (any(data_types == "factor")) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         factor_traits &lt;- traits[data_types == "factor"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         for (tr in factor_traits) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             warning(paste0("Trait ", tr, " is a factor with levels ", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 paste0(levels(seurat_obj@meta.data[, tr]), collapse = ", "), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 ". Levels will be converted to numeric IN THIS ORDER for the correlation, is this the expected order?"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- levels(modules$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- mods[mods != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     trait_df &lt;- seurat_obj@meta.data[, traits]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (length(traits == 1)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         trait_df &lt;- data.frame(x = trait_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         colnames(trait_df) &lt;- traits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (any(data_types == "factor")) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         factor_traits &lt;- traits[data_types == "factor"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         for (tr in factor_traits) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             trait_df[, tr] &lt;- as.numeric(trait_df[, tr])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_list &lt;- list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     pval_list &lt;- list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     fdr_list &lt;- list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temp &lt;- Hmisc::rcorr(as.matrix(trait_df), as.matrix(MEs), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type = cor_method)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_cor &lt;- temp$r[traits, mods]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_p &lt;- temp$P[traits, mods]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p_df &lt;- cur_p %&gt;% reshape2::melt()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (length(traits) == 1) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tmp &lt;- rep(mods, length(traits))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tmp &lt;- factor(tmp, levels = mods)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tmp &lt;- tmp[order(tmp)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df$Var1 &lt;- traits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df$Var2 &lt;- tmp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rownames(p_df) &lt;- 1:nrow(p_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df &lt;- dplyr::select(p_df, c(Var1, Var2, value))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p_df &lt;- p_df %&gt;% dplyr::mutate(fdr = p.adjust(value, method = "fdr")) %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dplyr::select(c(Var1, Var2, fdr))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_fdr &lt;- reshape2::dcast(p_df, Var1 ~ Var2, value.var = "fdr")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rownames(cur_fdr) &lt;- cur_fdr$Var1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_fdr &lt;- cur_fdr[, -1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_list[["all_cells"]] &lt;- cur_cor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     pval_list[["all_cells"]] &lt;- cur_p</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     fdr_list[["all_cells"]] &lt;- cur_fdr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     trait_df &lt;- cbind(trait_df, seurat_obj@meta.data[, group.by])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     colnames(trait_df)[ncol(trait_df)] &lt;- "group"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     MEs &lt;- cbind(as.data.frame(MEs), seurat_obj@meta.data[, group.by])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     colnames(MEs)[ncol(MEs)] &lt;- "group"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (class(seurat_obj@meta.data[, group.by]) == "factor") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         group_names &lt;- levels(seurat_obj@meta.data[, group.by])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         group_names &lt;- levels(as.factor(seurat_obj@meta.data[, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             group.by]))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     trait_list &lt;- dplyr::group_split(trait_df, group, .keep = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ME_list &lt;- dplyr::group_split(MEs, group, .keep = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     names(trait_list) &lt;- group_names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     names(ME_list) &lt;- group_names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     for (i in names(trait_list)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         temp &lt;- Hmisc::rcorr(as.matrix(trait_list[[i]]), as.matrix(ME_list[[i]]))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_cor &lt;- temp$r[traits, mods]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_p &lt;- temp$P[traits, mods]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df &lt;- cur_p %&gt;% reshape2::melt()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (length(traits) == 1) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             tmp &lt;- rep(mods, length(traits))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             tmp &lt;- factor(tmp, levels = mods)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             tmp &lt;- tmp[order(tmp)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p_df$Var1 &lt;- traits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p_df$Var2 &lt;- tmp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             rownames(p_df) &lt;- 1:nrow(p_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p_df &lt;- dplyr::select(p_df, c(Var1, Var2, value))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df &lt;- p_df %&gt;% dplyr::mutate(fdr = p.adjust(value, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             method = "fdr")) %&gt;% dplyr::select(c(Var1, Var2, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             fdr))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_fdr &lt;- reshape2::dcast(p_df, Var1 ~ Var2, value.var = "fdr")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rownames(cur_fdr) &lt;- cur_fdr$Var1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_fdr &lt;- cur_fdr[, -1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cor_list[[i]] &lt;- cur_cor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         pval_list[[i]] &lt;- cur_p</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         fdr_list[[i]] &lt;- as.matrix(cur_fdr)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mt_cor &lt;- list(cor = cor_list, pval = pval_list, fdr = fdr_list)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!is.null(subset_by)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_full &lt;- SetModuleTraitCorrelation(seurat_full, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             mt_cor, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj &lt;- seurat_full</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj &lt;- SetModuleTraitCorrelation(seurat_obj, mt_cor, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seurat_obj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7fea1ca357e0&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

