<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ModuleTraitCorrelation' — ModuleTraitCorrelation • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleTraitCorrelation' — ModuleTraitCorrelation"><meta name="description" content="Correlates categorical and numeric variables with Module Eigengenes or hub-gene scores."><meta property="og:description" content="Correlates categorical and numeric variables with Module Eigengenes or hub-gene scores."><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ModuleTraitCorrelation'</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/ModuleTraitCorrelation.R" class="external-link"><code>R/ModuleTraitCorrelation.R</code></a></small>
      <div class="d-none name"><code>ModuleTraitCorrelation.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Correlates categorical and numeric variables with Module Eigengenes or hub-gene scores.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ModuleTraitCorrelation</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  <span class="va">traits</span>,</span>
<span>  group.by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  features <span class="op">=</span> <span class="st">"hMEs"</span>,</span>
<span>  cor_method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  subset_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  subset_groups <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A list of column names in the Seurat object's metadata that you wish to correlate with each module.
Traits must be a categorical variable (not a character vector), or a numeric variable.</p></dd>


<dt id="arg-features">features<a class="anchor" aria-label="anchor" href="#arg-features"></a></dt>
<dd><p>Which features to use to summarize each modules? Valid choices are hMEs, MEs, or scores</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_obj@misc slot</p></dd>


<dt id="arg-cor-meth">cor_meth<a class="anchor" aria-label="anchor" href="#arg-cor-meth"></a></dt>
<dd><p>Which method to use for correlation? Valid choices are pearson, spearman, kendall.</p></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">ModuleTraitCorrelation</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, traits, group.by = NULL, features = "hMEs", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_method = "pearson", subset_by = NULL, subset_groups = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     wgcna_name = NULL, ...) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     CheckWGCNAName(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (features == "hMEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, TRUE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "MEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, FALSE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "scores") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetModuleScores(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         stop("Invalid feature selection. Valid choices: hMEs, MEs, scores, average")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!is.null(subset_by)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         print("subsetting")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_full &lt;- seurat_obj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- MEs[seurat_obj@meta.data[[subset_by]] %in% subset_groups, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj &lt;- seurat_obj[, seurat_obj@meta.data[[subset_by]] %in% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             subset_groups]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (sum(traits %in% colnames(seurat_obj@meta.data)) != length(traits)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         stop(paste("Some of the provided traits were not found in the Seurat obj:", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             paste(traits[!(traits %in% colnames(seurat_obj@meta.data))], </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 collapse = ", ")))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(group.by)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         group.by &lt;- "temp_ident"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj$temp_ident &lt;- Idents(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     valid_types &lt;- c("numeric", "factor", "integer")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     data_types &lt;- sapply(traits, function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class(seurat_obj@meta.data[, x])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     })</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!all(data_types %in% valid_types)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         incorrect &lt;- traits[!(data_types %in% valid_types)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         stop(paste0("Invalid data types for ", paste(incorrect, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             collapse = ", "), ". Accepted data types are numeric, factor, integer."))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (any(data_types == "factor")) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         factor_traits &lt;- traits[data_types == "factor"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         for (tr in factor_traits) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             warning(paste0("Trait ", tr, " is a factor with levels ", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 paste0(levels(seurat_obj@meta.data[, tr]), collapse = ", "), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 ". Levels will be converted to numeric IN THIS ORDER for the correlation, is this the expected order?"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- levels(modules$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- mods[mods != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     trait_df &lt;- seurat_obj@meta.data[, traits]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (length(traits == 1)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         trait_df &lt;- data.frame(x = trait_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         colnames(trait_df) &lt;- traits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (any(data_types == "factor")) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         factor_traits &lt;- traits[data_types == "factor"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         for (tr in factor_traits) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             trait_df[, tr] &lt;- as.numeric(trait_df[, tr])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_list &lt;- list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     pval_list &lt;- list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     fdr_list &lt;- list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temp &lt;- Hmisc::rcorr(as.matrix(trait_df), as.matrix(MEs), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type = cor_method)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_cor &lt;- temp$r[traits, mods]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_p &lt;- temp$P[traits, mods]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p_df &lt;- cur_p %&gt;% reshape2::melt()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (length(traits) == 1) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tmp &lt;- rep(mods, length(traits))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tmp &lt;- factor(tmp, levels = mods)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tmp &lt;- tmp[order(tmp)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df$Var1 &lt;- traits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df$Var2 &lt;- tmp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rownames(p_df) &lt;- 1:nrow(p_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df &lt;- dplyr::select(p_df, c(Var1, Var2, value))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p_df &lt;- p_df %&gt;% dplyr::mutate(fdr = p.adjust(value, method = "fdr")) %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dplyr::select(c(Var1, Var2, fdr))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_fdr &lt;- reshape2::dcast(p_df, Var1 ~ Var2, value.var = "fdr")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rownames(cur_fdr) &lt;- cur_fdr$Var1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_fdr &lt;- cur_fdr[, -1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_list[["all_cells"]] &lt;- cur_cor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     pval_list[["all_cells"]] &lt;- cur_p</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     fdr_list[["all_cells"]] &lt;- cur_fdr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     trait_df &lt;- cbind(trait_df, seurat_obj@meta.data[, group.by])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     colnames(trait_df)[ncol(trait_df)] &lt;- "group"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     MEs &lt;- cbind(as.data.frame(MEs), seurat_obj@meta.data[, group.by])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     colnames(MEs)[ncol(MEs)] &lt;- "group"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (class(seurat_obj@meta.data[, group.by]) == "factor") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         group_names &lt;- levels(seurat_obj@meta.data[, group.by])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         group_names &lt;- levels(as.factor(seurat_obj@meta.data[, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             group.by]))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     trait_list &lt;- dplyr::group_split(trait_df, group, .keep = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ME_list &lt;- dplyr::group_split(MEs, group, .keep = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     names(trait_list) &lt;- group_names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     names(ME_list) &lt;- group_names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     for (i in names(trait_list)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         temp &lt;- Hmisc::rcorr(as.matrix(trait_list[[i]]), as.matrix(ME_list[[i]]))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_cor &lt;- temp$r[traits, mods]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_p &lt;- temp$P[traits, mods]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df &lt;- cur_p %&gt;% reshape2::melt()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (length(traits) == 1) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             tmp &lt;- rep(mods, length(traits))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             tmp &lt;- factor(tmp, levels = mods)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             tmp &lt;- tmp[order(tmp)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p_df$Var1 &lt;- traits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p_df$Var2 &lt;- tmp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             rownames(p_df) &lt;- 1:nrow(p_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             p_df &lt;- dplyr::select(p_df, c(Var1, Var2, value))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p_df &lt;- p_df %&gt;% dplyr::mutate(fdr = p.adjust(value, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             method = "fdr")) %&gt;% dplyr::select(c(Var1, Var2, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             fdr))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_fdr &lt;- reshape2::dcast(p_df, Var1 ~ Var2, value.var = "fdr")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rownames(cur_fdr) &lt;- cur_fdr$Var1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_fdr &lt;- cur_fdr[, -1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cor_list[[i]] &lt;- cur_cor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         pval_list[[i]] &lt;- cur_p</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         fdr_list[[i]] &lt;- as.matrix(cur_fdr)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mt_cor &lt;- list(cor = cor_list, pval = pval_list, fdr = fdr_list)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!is.null(subset_by)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_full &lt;- SetModuleTraitCorrelation(seurat_full, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             mt_cor, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj &lt;- seurat_full</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj &lt;- SetModuleTraitCorrelation(seurat_obj, mt_cor, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seurat_obj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7f96306795a8&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

