<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>ModuleCorrNetworks — ModuleCorrNetwork • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleCorrNetworks — ModuleCorrNetwork"><meta property="og:description" content="Plot Module Eigengene correlogram"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hdWGCNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.01</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/hdWGCNA.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-header">Core functionality</li>
    <li>
      <a href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a>
    </li>
    <li>
      <a href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a>
    </li>
    <li>
      <a href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a>
    </li>
    <li>
      <a href="../articles/network_visualizations.html">Network visualization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Biological context for co-expression modules</li>
    <li>
      <a href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a>
    </li>
    <li>
      <a href="../articles/module_trait_correlation.html">Module trait correlation</a>
    </li>
    <li>
      <a href="../articles/enrichment_analysis.html">Enrichment analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Exploring modules in external datasets</li>
    <li>
      <a href="../articles/projecting_modules.html">Projecting modules to new datasets</a>
    </li>
    <li>
      <a href="../articles/module_preservation.html">Module preservation and reproducibility</a>
    </li>
    <li>
      <a href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Advanced topics</li>
    <li>
      <a href="../articles/consensus_wgcna.html">Consensus network analysis</a>
    </li>
    <li>
      <a href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a>
    </li>
    <li>
      <a href="../articles/motif_analysis.html">Motif analysis</a>
    </li>
    <li>
      <a href="../articles/other_metacells.html">Alternative metacell algorithms</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Other</li>
    <li>
      <a href="../articles/customization.html">Module customization</a>
    </li>
    <li>
      <a href="../articles/sctransform.html">Using SCTransform normalized data</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>ModuleCorrNetworks</h1>
    
    <div class="hidden name"><code>ModuleCorrNetwork.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot Module Eigengene correlogram</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">ModuleCorrNetwork</span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,
  cluster_col <span class="op">=</span> <span class="cn">NULL</span>,
  exclude_grey <span class="op">=</span> <span class="cn">TRUE</span>,
  features <span class="op">=</span> <span class="st">"hMEs"</span>,
  reduction <span class="op">=</span> <span class="st">"umap"</span>,
  cor_cutoff <span class="op">=</span> <span class="fl">0.2</span>,
  label_vertices <span class="op">=</span> <span class="cn">FALSE</span>,
  edge_scale <span class="op">=</span> <span class="fl">5</span>,
  vertex_size <span class="op">=</span> <span class="fl">15</span>,
  niter <span class="op">=</span> <span class="fl">100</span>,
  vertex_frame <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seurat_obj</dt>
<dd><p>A Seurat object</p></dd>
</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">ModuleCorrNetwork</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, wgcna_name = NULL, cluster_col = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     exclude_grey = TRUE, features = "hMEs", reduction = "umap", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_cutoff = 0.2, label_vertices = FALSE, edge_scale = 5, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     vertex_size = 15, niter = 100, vertex_frame = FALSE) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (features == "hMEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, TRUE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "MEs") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetMEs(seurat_obj, FALSE, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "scores") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetModuleScores(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else if (features == "average") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- GetAvgModuleExpr(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         restrict_range &lt;- FALSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else (stop("Invalid feature selection. Valid choices: hMEs, MEs, scores, average"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (exclude_grey) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         MEs &lt;- MEs[, colnames(MEs) != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         modules &lt;- modules %&gt;% subset(color != "grey")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- colnames(MEs)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(cluster_col)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clusters &lt;- Idents(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clusters &lt;- droplevels(seurat_obj@meta.data[[cluster_col]])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     MEs$cluster &lt;- clusters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cluster_ME_av &lt;- do.call(rbind, lapply(split(MEs, MEs$cluster), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             colMeans(x[, mods])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         })) %&gt;% as.data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     top_clusters &lt;- sapply(mods, function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rownames(cluster_ME_av[cluster_ME_av[, x] == max(cluster_ME_av[, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             x]), ])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     })</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     red_df &lt;- as.data.frame(seurat_obj@reductions[[reduction]]@cell.embeddings)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     red_df$cluster &lt;- clusters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     red_av &lt;- do.call(rbind, lapply(split(red_df, red_df$cluster), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             colMeans(x[, 1:2])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         })) %&gt;% as.data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_mat &lt;- Hmisc::rcorr(as.matrix(MEs[, 1:ncol(MEs) - 1]))$r</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_mat[lower.tri(cor_mat)] &lt;- NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_df &lt;- reshape2::melt(cor_mat) %&gt;% na.omit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_df &lt;- cor_df %&gt;% subset(!(Var1 == Var2))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_df &lt;- cor_df %&gt;% subset(abs(value) &gt;= cor_cutoff)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     v_df &lt;- data.frame(name = mods, cluster = as.character(top_clusters))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     unique_mods &lt;- distinct(modules[, c("module", "color")])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rownames(unique_mods) &lt;- unique_mods$module</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     v_df$color &lt;- unique_mods[v_df$name, "color"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     v_df$x &lt;- red_av[v_df$cluster, 1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     v_df$y &lt;- red_av[v_df$cluster, 2]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     g &lt;- igraph::graph_from_data_frame(cor_df, directed = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertices = v_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     e &lt;- get.edgelist(g, name = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     l &lt;- qgraph::qgraph.layout.fruchtermanreingold(e, vcount = vcount(g), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         weights = igraph::E(g)$value, repulse.rad = (vcount(g)), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         niter = niter, )</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot_df &lt;- rbind(cor_df, data.frame(Var1 = c("x", "y"), Var2 = c("y", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "x"), value = c(-1, 1)))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temp &lt;- ggplot(plot_df, aes(x = value, y = value, color = value)) + </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         geom_point() + scale_color_gradient2(high = "darkorchid1", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mid = "white", low = "seagreen", midpoint = 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temp &lt;- ggplot_build(temp)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     igraph::E(g)$color &lt;- temp$data[[1]]$colour[1:nrow(cor_df)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (label_vertices) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         labels &lt;- igraph::V(g)$name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         labels &lt;- NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (vertex_frame) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         frame_color &lt;- "black"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         frame_color &lt;- igraph::V(g)$color</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot(g, layout = l, edge.color = igraph::E(g)$color, edge.curved = 0, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         edge.width = abs(igraph::E(g)$value) * edge_scale, vertex.color = igraph::V(g)$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.frame.color = frame_color, vertex.label = labels, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.family = "Helvetica", vertex.label.color = "black", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.cex = 0.5, vertex.size = vertex_size)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7f77b7b1f6e8&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

