<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ModuleUMAPPlot — ModuleUMAPPlot • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleUMAPPlot — ModuleUMAPPlot"><meta name="description" content="Makes a igraph network plot using the module UMAP"><meta property="og:description" content="Makes a igraph network plot using the module UMAP"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ModuleUMAPPlot</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/plotting.R" class="external-link"><code>R/plotting.R</code></a></small>
      <div class="d-none name"><code>ModuleUMAPPlot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Makes a igraph network plot using the module UMAP</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ModuleUMAPPlot</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  sample_edges <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  edge_prop <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  label_hubs <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  edge.alpha <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  vertex.label.cex <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  label_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  return_graph <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  keep_grey_edges <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object</p></dd>


<dt id="arg-sample-edges">sample_edges<a class="anchor" aria-label="anchor" href="#arg-sample-edges"></a></dt>
<dd><p>logical determining whether we downsample edges for plotting (TRUE), or take the strongst edges.</p></dd>


<dt id="arg-edge-prop">edge_prop<a class="anchor" aria-label="anchor" href="#arg-edge-prop"></a></dt>
<dd><p>proportion of edges to plot. If sample_edges=FALSE, the strongest edges are selected.</p></dd>


<dt id="arg-label-hubs">label_hubs<a class="anchor" aria-label="anchor" href="#arg-label-hubs"></a></dt>
<dd><p>the number of hub genes to label in each module</p></dd>


<dt id="arg-edge-alpha">edge.alpha<a class="anchor" aria-label="anchor" href="#arg-edge-alpha"></a></dt>
<dd><p>scaling factor for edge opacity</p></dd>


<dt id="arg-vertex-label-cex">vertex.label.cex<a class="anchor" aria-label="anchor" href="#arg-vertex-label-cex"></a></dt>
<dd><p>font size for labeled genes</p></dd>


<dt id="arg-return-graph">return_graph<a class="anchor" aria-label="anchor" href="#arg-return-graph"></a></dt>
<dd><p>logical determining whether to plot thr graph (FALSE) or return the igraph object (TRUE)</p></dd>


<dt id="arg-keep-grey-edges">keep_grey_edges<a class="anchor" aria-label="anchor" href="#arg-keep-grey-edges"></a></dt>
<dd><p>logical determining whether to show edges between genes in different modules (grey edges)</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_obj@misc slot</p></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">ModuleUMAPPlot</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, sample_edges = TRUE, edge_prop = 0.2, label_hubs = 5, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge.alpha = 0.25, vertex.label.cex = 0.5, label_genes = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     return_graph = FALSE, keep_grey_edges = TRUE, wgcna_name = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     TOM &lt;- GetTOM(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     umap_df &lt;- GetModuleUMAP(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- levels(umap_df$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- mods[mods != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     subset_TOM &lt;- TOM[umap_df$gene, umap_df$gene[umap_df$hub == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "hub"]]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     hub_list &lt;- lapply(mods, function(cur_mod) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur &lt;- subset(modules, module == cur_mod)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur[, c("gene_name", paste0("kME_", cur_mod))] %&gt;% top_n(label_hubs) %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             .$gene_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     })</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     names(hub_list) &lt;- mods</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     hub_labels &lt;- as.character(unlist(hub_list))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print("hub labels")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(hub_labels)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(label_genes)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(label_genes)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         label_genes &lt;- hub_labels</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (!any(label_genes %in% umap_df$gene)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             stop("Some genes in label_genes not found in the UMAP.")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         label_genes &lt;- unique(c(label_genes, hub_labels))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(label_genes)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules &lt;- modules[umap_df$gene, ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules &lt;- cbind(selected_modules, umap_df[, c("UMAP1", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "UMAP2", "hub", "kME")])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules$label &lt;- ifelse(selected_modules$gene_name %in% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         label_genes, selected_modules$gene_name, "")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules$fontcolor &lt;- ifelse(selected_modules$color == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "black", "gray50", "black")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules$framecolor &lt;- ifelse(selected_modules$gene_name %in% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         label_genes, "black", selected_modules$color)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- subset_TOM %&gt;% reshape2::melt()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(dim(edge_df))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df$color &lt;- future.apply::future_sapply(1:nrow(edge_df), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         function(i) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             gene1 = as.character(edge_df[i, "Var1"])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             gene2 = as.character(edge_df[i, "Var2"])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             col1 &lt;- selected_modules[selected_modules$gene_name == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 gene1, "color"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             col2 &lt;- selected_modules[selected_modules$gene_name == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 gene2, "color"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             if (col1 == col2) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 col = col1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 col = "grey90"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             col</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         })</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!keep_grey_edges) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         edge_df &lt;- edge_df %&gt;% subset(color != "grey90")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     groups &lt;- unique(edge_df$color)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (sample_edges) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         temp &lt;- do.call(rbind, lapply(groups, function(cur_group) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_df &lt;- edge_df %&gt;% subset(color == cur_group)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             n_edges &lt;- nrow(cur_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_sample &lt;- sample(1:n_edges, round(n_edges * edge_prop))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_df[cur_sample, ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         temp &lt;- do.call(rbind, lapply(groups, function(cur_group) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_df &lt;- edge_df %&gt;% subset(color == cur_group)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             n_edges &lt;- nrow(cur_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_df %&gt;% dplyr::top_n(round(n_edges * edge_prop), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 wt = value)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- temp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(dim(edge_df))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- edge_df %&gt;% group_by(color) %&gt;% mutate(value = scale01(value))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- edge_df %&gt;% arrange(value)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- rbind(subset(edge_df, color == "grey90"), subset(edge_df, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         color != "grey90"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df$color_alpha &lt;- ifelse(edge_df$color == "grey90", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alpha(edge_df$color, alpha = edge_df$value/2), alpha(edge_df$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             alpha = edge_df$value))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules &lt;- rbind(subset(selected_modules, hub == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "other"), subset(selected_modules, hub != "other"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules &lt;- rbind(subset(selected_modules, label == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ""), subset(selected_modules, label != ""))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     g &lt;- igraph::graph_from_data_frame(edge_df, directed = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertices = selected_modules)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (return_graph) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         return(g)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot(g, layout = as.matrix(selected_modules[, c("UMAP1", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "UMAP2")]), edge.color = adjustcolor(igraph::E(g)$color_alpha, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alpha.f = edge.alpha), vertex.size = igraph::V(g)$kME * </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         3, edge.curved = 0, edge.width = 0.5, vertex.color = igraph::V(g)$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label = igraph::V(g)$label, vertex.label.dist = 1.1, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.degree = -pi/4, vertex.label.family = "Helvetica", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.font = 3, vertex.label.color = igraph::V(g)$fontcolor, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.cex = 0, vertex.frame.color = igraph::V(g)$framecolor, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         margin = 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7f9177afc530&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

