<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>ModuleUMAPPlot — ModuleUMAPPlot • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleUMAPPlot — ModuleUMAPPlot"><meta property="og:description" content="Makes a igraph network plot using the module UMAP"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hdWGCNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.03</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/hdWGCNA.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-header">Core functionality</li>
    <li>
      <a href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a>
    </li>
    <li>
      <a href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a>
    </li>
    <li>
      <a href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a>
    </li>
    <li>
      <a href="../articles/network_visualizations.html">Network visualization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Biological context for co-expression modules</li>
    <li>
      <a href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a>
    </li>
    <li>
      <a href="../articles/module_trait_correlation.html">Module trait correlation</a>
    </li>
    <li>
      <a href="../articles/enrichment_analysis.html">Enrichment analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Exploring modules in external datasets</li>
    <li>
      <a href="../articles/projecting_modules.html">Projecting modules to new datasets</a>
    </li>
    <li>
      <a href="../articles/module_preservation.html">Module preservation and reproducibility</a>
    </li>
    <li>
      <a href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Advanced topics</li>
    <li>
      <a href="../articles/consensus_wgcna.html">Consensus network analysis</a>
    </li>
    <li>
      <a href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a>
    </li>
    <li>
      <a href="../articles/motif_analysis.html">Motif analysis</a>
    </li>
    <li>
      <a href="../articles/other_metacells.html">Alternative metacell algorithms</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Other</li>
    <li>
      <a href="../articles/customization.html">Module customization</a>
    </li>
    <li>
      <a href="../articles/sctransform.html">Using SCTransform normalized data</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>ModuleUMAPPlot</h1>
    
    <div class="hidden name"><code>ModuleUMAPPlot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Makes a igraph network plot using the module UMAP</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">ModuleUMAPPlot</span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  sample_edges <span class="op">=</span> <span class="cn">TRUE</span>,
  edge_prop <span class="op">=</span> <span class="fl">0.2</span>,
  label_hubs <span class="op">=</span> <span class="fl">5</span>,
  edge.alpha <span class="op">=</span> <span class="fl">0.25</span>,
  vertex.label.cex <span class="op">=</span> <span class="fl">0.5</span>,
  label_genes <span class="op">=</span> <span class="cn">NULL</span>,
  return_graph <span class="op">=</span> <span class="cn">FALSE</span>,
  keep_grey_edges <span class="op">=</span> <span class="cn">TRUE</span>,
  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seurat_obj</dt>
<dd><p>A Seurat object</p></dd>
<dt>sample_edges</dt>
<dd><p>logical determining whether we downsample edges for plotting (TRUE), or take the strongst edges.</p></dd>
<dt>edge_prop</dt>
<dd><p>proportion of edges to plot. If sample_edges=FALSE, the strongest edges are selected.</p></dd>
<dt>label_hubs</dt>
<dd><p>the number of hub genes to label in each module</p></dd>
<dt>edge.alpha</dt>
<dd><p>scaling factor for edge opacity</p></dd>
<dt>vertex.label.cex</dt>
<dd><p>font size for labeled genes</p></dd>
<dt>return_graph</dt>
<dd><p>logical determining whether to plot thr graph (FALSE) or return the igraph object (TRUE)</p></dd>
<dt>keep_grey_edges</dt>
<dd><p>logical determining whether to show edges between genes in different modules (grey edges)</p></dd>
<dt>wgcna_name</dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_obj@misc slot</p></dd>
</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">ModuleUMAPPlot</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, sample_edges = TRUE, edge_prop = 0.2, label_hubs = 5, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge.alpha = 0.25, vertex.label.cex = 0.5, label_genes = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     return_graph = FALSE, keep_grey_edges = TRUE, wgcna_name = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     TOM &lt;- GetTOM(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     umap_df &lt;- GetModuleUMAP(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- levels(umap_df$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- mods[mods != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     subset_TOM &lt;- TOM[umap_df$gene, umap_df$gene[umap_df$hub == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "hub"]]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     hub_list &lt;- lapply(mods, function(cur_mod) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur &lt;- subset(modules, module == cur_mod)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur[, c("gene_name", paste0("kME_", cur_mod))] %&gt;% top_n(label_hubs) %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             .$gene_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     })</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     names(hub_list) &lt;- mods</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     hub_labels &lt;- as.character(unlist(hub_list))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print("hub labels")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(hub_labels)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(label_genes)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(label_genes)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         label_genes &lt;- hub_labels</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (!any(label_genes %in% umap_df$gene)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             stop("Some genes in label_genes not found in the UMAP.")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         label_genes &lt;- unique(c(label_genes, hub_labels))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(label_genes)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules &lt;- modules[umap_df$gene, ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules &lt;- cbind(selected_modules, umap_df[, c("UMAP1", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "UMAP2", "hub", "kME")])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules$label &lt;- ifelse(selected_modules$gene_name %in% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         label_genes, selected_modules$gene_name, "")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules$fontcolor &lt;- ifelse(selected_modules$color == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "black", "gray50", "black")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules$framecolor &lt;- ifelse(selected_modules$gene_name %in% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         label_genes, "black", selected_modules$color)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- subset_TOM %&gt;% reshape2::melt()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(dim(edge_df))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df$color &lt;- future.apply::future_sapply(1:nrow(edge_df), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         function(i) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             gene1 = as.character(edge_df[i, "Var1"])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             gene2 = as.character(edge_df[i, "Var2"])</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             col1 &lt;- selected_modules[selected_modules$gene_name == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 gene1, "color"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             col2 &lt;- selected_modules[selected_modules$gene_name == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 gene2, "color"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             if (col1 == col2) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 col = col1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 col = "grey90"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             col</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         })</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!keep_grey_edges) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         edge_df &lt;- edge_df %&gt;% subset(color != "grey90")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     groups &lt;- unique(edge_df$color)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (sample_edges) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         temp &lt;- do.call(rbind, lapply(groups, function(cur_group) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_df &lt;- edge_df %&gt;% subset(color == cur_group)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             n_edges &lt;- nrow(cur_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_sample &lt;- sample(1:n_edges, round(n_edges * edge_prop))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_df[cur_sample, ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         temp &lt;- do.call(rbind, lapply(groups, function(cur_group) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_df &lt;- edge_df %&gt;% subset(color == cur_group)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             n_edges &lt;- nrow(cur_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_df %&gt;% dplyr::top_n(round(n_edges * edge_prop), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 wt = value)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- temp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print(dim(edge_df))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- edge_df %&gt;% group_by(color) %&gt;% mutate(value = scale01(value))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- edge_df %&gt;% arrange(value)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- rbind(subset(edge_df, color == "grey90"), subset(edge_df, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         color != "grey90"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df$color_alpha &lt;- ifelse(edge_df$color == "grey90", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alpha(edge_df$color, alpha = edge_df$value/2), alpha(edge_df$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             alpha = edge_df$value))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules &lt;- rbind(subset(selected_modules, hub == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "other"), subset(selected_modules, hub != "other"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     selected_modules &lt;- rbind(subset(selected_modules, label == </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ""), subset(selected_modules, label != ""))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     g &lt;- igraph::graph_from_data_frame(edge_df, directed = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertices = selected_modules)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (return_graph) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         return(g)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot(g, layout = as.matrix(selected_modules[, c("UMAP1", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "UMAP2")]), edge.color = adjustcolor(igraph::E(g)$color_alpha, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alpha.f = edge.alpha), vertex.size = igraph::V(g)$kME * </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         3, edge.curved = 0, edge.width = 0.5, vertex.color = igraph::V(g)$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label = igraph::V(g)$label, vertex.label.dist = 1.1, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.degree = -pi/4, vertex.label.family = "Helvetica", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.font = 3, vertex.label.color = igraph::V(g)$fontcolor, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.cex = 0, vertex.frame.color = igraph::V(g)$framecolor, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         margin = 0)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7fea45e92b00&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

