<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>ModuleTFNetwork — ModuleTFNetwork • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleTFNetwork — ModuleTFNetwork"><meta property="og:description" content="Plotting the relationships between a TF and the co-expression modules"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hdWGCNA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.01</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/hdWGCNA.html">Get started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li class="dropdown-header">Core functionality</li>
    <li>
      <a href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a>
    </li>
    <li>
      <a href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a>
    </li>
    <li>
      <a href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a>
    </li>
    <li>
      <a href="../articles/network_visualizations.html">Network visualization</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Biological context for co-expression modules</li>
    <li>
      <a href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a>
    </li>
    <li>
      <a href="../articles/module_trait_correlation.html">Module trait correlation</a>
    </li>
    <li>
      <a href="../articles/enrichment_analysis.html">Enrichment analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Exploring modules in external datasets</li>
    <li>
      <a href="../articles/projecting_modules.html">Projecting modules to new datasets</a>
    </li>
    <li>
      <a href="../articles/module_preservation.html">Module preservation and reproducibility</a>
    </li>
    <li>
      <a href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Advanced topics</li>
    <li>
      <a href="../articles/consensus_wgcna.html">Consensus network analysis</a>
    </li>
    <li>
      <a href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a>
    </li>
    <li>
      <a href="../articles/motif_analysis.html">Motif analysis</a>
    </li>
    <li>
      <a href="../articles/other_metacells.html">Alternative metacell algorithms</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Other</li>
    <li>
      <a href="../articles/customization.html">Module customization</a>
    </li>
    <li>
      <a href="../articles/sctransform.html">Using SCTransform normalized data</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/index.html">All vignettes</a>
    </li>
  </ul></li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>ModuleTFNetwork</h1>
    
    <div class="hidden name"><code>ModuleTFNetwork.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plotting the relationships between a TF and the co-expression modules</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">ModuleTFNetwork</span><span class="op">(</span>
  <span class="va">seurat_obj</span>,
  <span class="va">tf_name</span>,
  <span class="va">tf_gene_name</span>,
  edge.alpha <span class="op">=</span> <span class="fl">0.75</span>,
  cor_thresh <span class="op">=</span> <span class="fl">0.25</span>,
  high_color <span class="op">=</span> <span class="st">"red"</span>,
  mid_color <span class="op">=</span> <span class="st">"grey"</span>,
  low_color <span class="op">=</span> <span class="st">"blue"</span>,
  slot <span class="op">=</span> <span class="st">"data"</span>,
  size.scale <span class="op">=</span> <span class="fl">30</span>,
  tf_x <span class="op">=</span> <span class="fl">0</span>,
  tf_y <span class="op">=</span> <span class="fl">0</span>,
  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seurat_obj</dt>
<dd><p>A Seurat object</p></dd>
<dt>tf_name</dt>
<dd><p>the Motif name for the tf</p></dd>
<dt>tf_gene_name</dt>
<dd><p>the gene associated with this tf in the rownames(seurat_obj)</p></dd>
<dt>edge.alpha</dt>
<dd><p>scaling factor for edge opacity in the network</p></dd>
<dt>cor_thresh</dt>
<dd><p>threshold to plot correlation edges between modules</p></dd>
<dt>high_color</dt>
<dd><p>color for positive correlation</p></dd>
<dt>mid_color</dt>
<dd><p>color for zero correlation</p></dd>
<dt>low_color</dt>
<dd><p>color for negative correlation</p></dd>
<dt>slot</dt>
<dd><p>the slot in the seurat object to extract expression data for the tf_gene_name</p></dd>
<dt>size.scale</dt>
<dd><p>scaling factor for the size of each node</p></dd>
<dt>tf_x</dt>
<dd><p>x coordinate for the TF if the TF is not found in the UMAP</p></dd>
<dt>tf_y</dt>
<dd><p>y coordinate for the TF if the TF is not foudn in the UMAP</p></dd>
<dt>wgcna_name</dt>
<dd><p>the name of the WGCNA experiment in the seurat object</p></dd>
</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">ModuleTFNetwork</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, tf_name, tf_gene_name, edge.alpha = 0.75, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cor_thresh = 0.25, high_color = "red", mid_color = "grey", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     low_color = "blue", slot = "data", size.scale = 30, tf_x = 0, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf_y = 0, wgcna_name = NULL) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj, wgcna_name) %&gt;% subset(module != </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "grey") %&gt;% mutate(module = droplevels(module))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     MEs &lt;- GetMEs(seurat_obj, TRUE, wgcna_name) %&gt;% as.matrix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     MEs &lt;- MEs[, colnames(MEs) != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mod_sizes &lt;- table(modules$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor &lt;- Hmisc::rcorr(x = MEs, type = "pearson")$r</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor[lower.tri(module_cor)] &lt;- NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor &lt;- reshape2::melt(module_cor) %&gt;% na.omit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor &lt;- subset(module_cor, abs(value) &gt;= cor_thresh &amp; </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Var1 != Var2)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_exp &lt;- GetAssayData(seurat_obj, slot = slot)[tf_gene_name, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     exp_cor &lt;- Hmisc::rcorr(x = MEs, y = cur_exp)$r[1:ncol(MEs), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "y"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     exp_cor &lt;- data.frame(mod = names(exp_cor), value = as.numeric(exp_cor))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot_lim &lt;- abs(max(c(abs(range(exp_cor$value)), abs(range(module_cor$value)))))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p &lt;- ggplot(module_cor, aes(x = Var1, y = Var2, color = value)) + </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         geom_point() + scale_color_gradient2(high = high_color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mid = mid_color, low = low_color, limits = c(-1 * plot_lim, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             plot_lim))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ggp &lt;- ggplot_build(p)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     module_cor$color &lt;- ggp$data[[1]]$colour</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     p &lt;- ggplot(exp_cor, aes(x = mod, y = mod, color = value)) + </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         geom_point() + scale_color_gradient2(high = high_color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mid = mid_color, low = low_color, limits = c(-1 * plot_lim, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             plot_lim))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ggp &lt;- ggplot_build(p)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     exp_cor$color &lt;- ggp$data[[1]]$colour</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     umap_df &lt;- GetModuleUMAP(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- levels(umap_df$modules)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf_match &lt;- GetMotifMatrix(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf_targets &lt;- GetMotifTargets(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     motif_df &lt;- GetMotifs(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     overlap_df &lt;- GetMotifOverlap(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     centroid_df &lt;- umap_df %&gt;% dplyr::select(c(UMAP1, UMAP2, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         module)) %&gt;% dplyr::group_by(module) %&gt;% dplyr::summarise(x = mean(UMAP1), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         y = mean(UMAP2))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cur_overlap &lt;- subset(overlap_df, tf == tf_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     node_df &lt;- dplyr::left_join(centroid_df, cur_overlap, by = "module") %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dplyr::rename(c(UMAP1 = x, UMAP2 = y, name = module))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     node_df$size &lt;- as.numeric(node_df$size_intersection)/as.numeric(mod_sizes)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (tf_gene_name %in% umap_df$gene) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tf_df &lt;- umap_df[umap_df$gene == tf_gene_name, ] %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             dplyr::select(-c(hub, kME, module)) %&gt;% dplyr::rename(name = gene)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         tf_df &lt;- data.frame(name = tf_gene_name, module = "grey", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             color = "grey", UMAP1 = tf_x, UMAP2 = tf_y)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     tf_df$size &lt;- 0.25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- data.frame(Var1 = tf_gene_name, Var2 = as.character(node_df$name), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         value = node_df$odds_ratio, color = exp_cor$color)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df$color &lt;- ifelse(node_df$fdr &lt;= 0.05, edge_df$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         "grey")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     edge_df &lt;- subset(edge_df, color != "grey")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     node_df &lt;- dplyr::bind_rows(node_df, tf_df) %&gt;% as.data.frame()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rownames(node_df) &lt;- as.character(node_df$name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     g1 &lt;- igraph::graph_from_data_frame(edge_df, directed = TRUE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertices = node_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     g2 &lt;- igraph::graph_from_data_frame(module_cor, directed = FALSE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertices = node_df)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot(g2, layout = as.matrix(node_df[, c("UMAP1", "UMAP2")]), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.size = 1, edge.curved = 0, edge.width = 1, vertex.color = "grey", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label = "", edge.color = adjustcolor(igraph::E(g2)$color, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             alpha.f = edge.alpha), )</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     plot(g1, layout = as.matrix(node_df[, c("UMAP1", "UMAP2")]), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         edge.color = adjustcolor(igraph::E(g1)$color), vertex.size = igraph::V(g1)$size * </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             size.scale, edge.curved = 0, edge.width = edge_df$value * </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             2, vertex.color = igraph::V(g1)$color, vertex.label = igraph::V(g1)$name, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.dist = 1.1, vertex.label.degree = -pi/4, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.family = "Helvetica", vertex.label.font = 3, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         vertex.label.color = "black", vertex.label.cex = 0, vertex.frame.color = "black", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         margin = 0, edge.arrow.size = edge_df$value/2, add = TRUE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7f77cfdfcbc0&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

