<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ModuleRegulatoryNetworkPlot — ModuleRegulatoryNetworkPlot • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ModuleRegulatoryNetworkPlot — ModuleRegulatoryNetworkPlot"><meta name="description" content="This function visualizes the regulatory network between gene modules from a Seurat object.
The plot displays transcription factor (TF) regulatory interactions between modules, with edges
representing positive or negative regulatory links."><meta property="og:description" content="This function visualizes the regulatory network between gene modules from a Seurat object.
The plot displays transcription factor (TF) regulatory interactions between modules, with edges
representing positive or negative regulatory links."><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ModuleRegulatoryNetworkPlot</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/ModuleRegulatoryNetwork.R" class="external-link"><code>R/ModuleRegulatoryNetwork.R</code></a></small>
      <div class="d-none name"><code>ModuleRegulatoryNetworkPlot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function visualizes the regulatory network between gene modules from a Seurat object.
The plot displays transcription factor (TF) regulatory interactions between modules, with edges
representing positive or negative regulatory links.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ModuleRegulatoryNetworkPlot</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  feature <span class="op">=</span> <span class="st">"delta"</span>,</span>
<span>  TFs_only <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  layout <span class="op">=</span> <span class="st">"umap"</span>,</span>
<span>  umap_background <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  max_val <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  cutoff <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  focus_source <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  focus_target <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  loops <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  label_modules <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"orange2"</span>,</span>
<span>  mid_color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"dodgerblue"</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object containing single-cell data and WGCNA results.</p></dd>


<dt id="arg-feature">feature<a class="anchor" aria-label="anchor" href="#arg-feature"></a></dt>
<dd><p>A character string specifying the type of regulatory score to plot. Options are
'positive' (positive regulatory score), 'negative' (negative regulatory score), or 'delta'
(difference between positive and negative scores). Default is 'delta'.</p></dd>


<dt id="arg-tfs-only">TFs_only<a class="anchor" aria-label="anchor" href="#arg-tfs-only"></a></dt>
<dd><p>Logical; if TRUE (default), only transcription factor (TF) genes are included
in the plot. If FALSE, all genes are considered in the regulatory network.</p></dd>


<dt id="arg-layout">layout<a class="anchor" aria-label="anchor" href="#arg-layout"></a></dt>
<dd><p>A character string specifying the layout for the plot. Default is 'umap', which
arranges the modules according to UMAP coordinates. Other layout options from <code>ggraph</code>
can also be used.</p></dd>


<dt id="arg-umap-background">umap_background<a class="anchor" aria-label="anchor" href="#arg-umap-background"></a></dt>
<dd><p>Logical; if TRUE, the UMAP of module eigengenes is plotted in the
background (if <code>layout = 'umap'</code>). Default is FALSE.</p></dd>


<dt id="arg-max-val">max_val<a class="anchor" aria-label="anchor" href="#arg-max-val"></a></dt>
<dd><p>Numeric; sets the maximum absolute value for the regulatory score. Any values
exceeding this threshold are capped. Default is 1.</p></dd>


<dt id="arg-cutoff">cutoff<a class="anchor" aria-label="anchor" href="#arg-cutoff"></a></dt>
<dd><p>Numeric; edges with absolute regulatory scores below this value are excluded
from the plot. Default is 0.</p></dd>


<dt id="arg-focus-source">focus_source<a class="anchor" aria-label="anchor" href="#arg-focus-source"></a></dt>
<dd><p>Character vector; optionally restricts the plot to only show regulatory
links from specified source modules. Default is NULL (all modules included).</p></dd>


<dt id="arg-focus-target">focus_target<a class="anchor" aria-label="anchor" href="#arg-focus-target"></a></dt>
<dd><p>Character vector; optionally restricts the plot to only show regulatory
links targeting specified modules. Default is NULL (all modules included).</p></dd>


<dt id="arg-loops">loops<a class="anchor" aria-label="anchor" href="#arg-loops"></a></dt>
<dd><p>Logical; if TRUE (default), loops (self-regulatory connections) are shown in the plot.</p></dd>


<dt id="arg-label-modules">label_modules<a class="anchor" aria-label="anchor" href="#arg-label-modules"></a></dt>
<dd><p>Logical; if TRUE (default), module names are displayed as labels in the plot.</p></dd>


<dt id="arg-high-color">high_color<a class="anchor" aria-label="anchor" href="#arg-high-color"></a></dt>
<dd><p>Character string; the color representing high regulatory scores in the
color gradient. Default is 'orange2'.</p></dd>


<dt id="arg-mid-color">mid_color<a class="anchor" aria-label="anchor" href="#arg-mid-color"></a></dt>
<dd><p>Character string; the color representing intermediate regulatory scores in the
color gradient. Default is 'white'.</p></dd>


<dt id="arg-low-color">low_color<a class="anchor" aria-label="anchor" href="#arg-low-color"></a></dt>
<dd><p>Character string; the color representing low regulatory scores in the
color gradient. Default is 'dodgerblue'.</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_obj@misc slot</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments passed to the <code><a href="https://ggraph.data-imaginist.com/reference/ggraph.html" class="external-link">ggraph::create_layout</a></code> function.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A ggplot2 object visualizing the regulatory network, which can be further customized or displayed
using <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function visualizes the regulatory network between gene modules by plotting transcription factor
regulatory interactions. Positive and negative regulatory scores are displayed as edges connecting
nodes (modules), with node sizes proportional to the number of genes in each module. Users can customize
the network layout (e.g., UMAP or other graph layouts), filter edges based on score thresholds, and highlight
specific modules by using the focus parameters. Edge colors represent the strength and direction (positive
or negative) of regulatory interactions, and loops can optionally be plotted to indicate self-regulation.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

