<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Displays the top n TFs in a set of modules as a bar plot — MotifOverlapBarPlot • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Displays the top n TFs in a set of modules as a bar plot — MotifOverlapBarPlot"><meta name="description" content="Displays the top n TFs in a set of modules as a bar plot"><meta property="og:description" content="Displays the top n TFs in a set of modules as a bar plot"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Displays the top n TFs in a set of modules as a bar plot</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/plotting.R" class="external-link"><code>R/plotting.R</code></a></small>
      <div class="d-none name"><code>MotifOverlapBarPlot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Displays the top n TFs in a set of modules as a bar plot</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MotifOverlapBarPlot</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  n_tfs <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  plot_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">6</span><span class="op">)</span>,</span>
<span>  outdir <span class="op">=</span> <span class="st">"MotifOverlaps/"</span>,</span>
<span>  motif_font <span class="op">=</span> <span class="st">"helvetica_regular"</span>,</span>
<span>  module_names <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_obj@misc slot</p></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">MotifOverlapBarPlot</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, n_tfs = 10, plot_size = c(5, 6), outdir = "MotifOverlaps/", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     motif_font = "helvetica_regular", module_names = NULL, wgcna_name = NULL) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_obj@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!dir.exists(outdir)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dir.create(outdir)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- GetModules(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- levels(modules$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- mods[mods != "grey"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(module_names)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         module_names &lt;- mods</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     overlap_df &lt;- GetMotifOverlap(seurat_obj, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     motif_df &lt;- GetMotifs(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     pfm &lt;- GetPFMList(seurat_obj)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     overlap_df$motif_ID &lt;- motif_df$motif_ID[match(overlap_df$tf, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         motif_df$motif_name)]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     overlap_df &lt;- overlap_df %&gt;% subset(module %in% module_names)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     for (cur_mod in module_names) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         print(cur_mod)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         plot_df &lt;- overlap_df %&gt;% subset(module == cur_mod) %&gt;% </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             top_n(n_tfs, wt = odds_ratio) %&gt;% arrange(desc(odds_ratio))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         p1 &lt;- plot_df %&gt;% ggplot(aes(y = reorder(tf, odds_ratio), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             fill = odds_ratio, x = odds_ratio)) + geom_bar(stat = "identity", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             width = 0.7) + NoLegend() + scale_fill_gradient(high = unique(plot_df$color), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             low = "grey90") + ylab("") + theme(axis.line.y = element_blank(), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             axis.text.y = element_blank(), plot.margin = margin(t = 0, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 r = 0, b = 0, l = 0))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         plot_list &lt;- list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         for (i in 1:nrow(plot_df)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_id &lt;- plot_df[i, "motif_ID"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_name &lt;- plot_df[i, "tf"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             plot_list[[cur_id]] &lt;- ggplot() + ggseqlogo::geom_logo(as.matrix(pfm[[cur_id]]), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 font = motif_font) + ggseqlogo::theme_logo() + </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 xlab("") + ylab(cur_name) + theme(axis.text.x = element_blank(), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 axis.text.y = element_blank(), axis.title.y = element_text(angle = 0), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 plot.margin = margin(t = 0, r = 0, b = 0, l = 0))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         patch1 &lt;- wrap_plots(plot_list, ncol = 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outplot &lt;- (patch1 | p1) + plot_layout(ncol = 2, widths = c(1, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             2)) + plot_annotation(title = paste0("Motif overlaps with ", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             cur_mod), theme = theme(plot.title = element_text(hjust = 0.5)))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         pdf(paste0(outdir, "/", cur_mod, "_motif_overlaps.pdf"), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             width = plot_size[1], height = plot_size[2], useDingbats = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         print(outplot)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dev.off()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7f916bb72978&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

