<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ProjectModules — ProjectModules • hdWGCNA</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ProjectModules — ProjectModules"><meta name="description" content="Project a set of co-expression modules from a reference to a query dataset"><meta property="og:description" content="Project a set of co-expression modules from a reference to a query dataset"><meta property="og:image" content="https://smorabit.github.io/hdWGCNA/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">hdWGCNA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.00</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/basic_tutorial.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><h6 class="dropdown-header" data-toc-skip>Core functionality</h6></li>
    <li><a class="dropdown-item" href="../articles/basic_tutorial.html">hdWGCNA in single-cell data</a></li>
    <li><a class="dropdown-item" href="../articles/ST_basics.html">hdWGCNA in spatial transcriptomics data</a></li>
    <li><a class="dropdown-item" href="../articles/isoform_pbmcs.html">Isoform co-expression network analysis with PacBio MAS-Seq</a></li>
    <li><a class="dropdown-item" href="../articles/network_visualizations.html">Network visualization</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Biological context for co-expression modules</h6></li>
    <li><a class="dropdown-item" href="../articles/differential_MEs.html">Differential module eigengene (DME) analysis</a></li>
    <li><a class="dropdown-item" href="../articles/module_trait_correlation.html">Module trait correlation</a></li>
    <li><a class="dropdown-item" href="../articles/enrichment_analysis.html">Enrichment analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Exploring modules in external datasets</h6></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules.html">Projecting modules to new datasets</a></li>
    <li><a class="dropdown-item" href="../articles/module_preservation.html">Module preservation and reproducibility</a></li>
    <li><a class="dropdown-item" href="../articles/projecting_modules_cross.html">Cross-species and cross-modality analysis</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced topics</h6></li>
    <li><a class="dropdown-item" href="../articles/tf_network.html">Transcription factor regulatory network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/consensus_wgcna.html">Consensus network analysis</a></li>
    <li><a class="dropdown-item" href="../articles/pseudobulk.html">hdWGCNA with pseudobulk data</a></li>
    <li><a class="dropdown-item" href="../articles/pseudotime.html">Co-expression module dynamics with pseudotime</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Other</h6></li>
    <li><a class="dropdown-item" href="../articles/customization.html">Module customization</a></li>
    <li><a class="dropdown-item" href="../articles/other_metacells.html">Alternative metacell algorithms</a></li>
    <li><a class="dropdown-item" href="../articles/sctransform.html">Using SCTransform normalized data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/index.html">All vignettes</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/hdWGCNA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ProjectModules</h1>
      <small class="dont-index">Source: <a href="https://github.com/smorabit/hdWGCNA/blob/HEAD/R/ProjectModules.R" class="external-link"><code>R/ProjectModules.R</code></a></small>
      <div class="d-none name"><code>ProjectModules.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Project a set of co-expression modules from a reference to a query dataset</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ProjectModules</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  <span class="va">seurat_ref</span>,</span>
<span>  modules <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group.by.vars <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genome1_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genome2_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  overlap_proportion <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  vars.to.regress <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  scale.model.use <span class="op">=</span> <span class="st">"linear"</span>,</span>
<span>  wgcna_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  wgcna_name_proj <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-obj">seurat_obj<a class="anchor" aria-label="anchor" href="#arg-seurat-obj"></a></dt>
<dd><p>A Seurat object where the modules will be projected to</p></dd>


<dt id="arg-seurat-ref">seurat_ref<a class="anchor" aria-label="anchor" href="#arg-seurat-ref"></a></dt>
<dd><p>A Seurat object containing the co-expression modules to be projected</p></dd>


<dt id="arg-modules">modules<a class="anchor" aria-label="anchor" href="#arg-modules"></a></dt>
<dd><p>optionally provide a dataframe containing gene-module information to bypass seurat_ref</p></dd>


<dt id="arg-group-by-vars">group.by.vars<a class="anchor" aria-label="anchor" href="#arg-group-by-vars"></a></dt>
<dd><p>groups to harmonize by</p></dd>


<dt id="arg-gene-mapping">gene_mapping<a class="anchor" aria-label="anchor" href="#arg-gene-mapping"></a></dt>
<dd><p>dataframe to map gene names between genomes</p></dd>


<dt id="arg-genome-col">genome1_col<a class="anchor" aria-label="anchor" href="#arg-genome-col"></a></dt>
<dd><p>column in gene_mapping containing the genes for seurat_ref (reference)</p></dd>


<dt id="arg-genome-col">genome2_col<a class="anchor" aria-label="anchor" href="#arg-genome-col"></a></dt>
<dd><p>column in gene_mapping containing the genes for seurat_obj (query)</p></dd>


<dt id="arg-overlap-proportion">overlap_proportion<a class="anchor" aria-label="anchor" href="#arg-overlap-proportion"></a></dt>
<dd><p>the proportion of genes that must be present in seurat_obj for a given module to be projected. Default = 0.5 (50% of genes)</p></dd>


<dt id="arg-vars-to-regress">vars.to.regress<a class="anchor" aria-label="anchor" href="#arg-vars-to-regress"></a></dt>
<dd><p>character vector of variables in seurat_obj@meta.data to regress when running ScaleData</p></dd>


<dt id="arg-scale-model-use">scale.model.use<a class="anchor" aria-label="anchor" href="#arg-scale-model-use"></a></dt>
<dd><p>model to scale data when running ScaleData choices are "linear", "poisson", or "negbinom"</p></dd>


<dt id="arg-wgcna-name">wgcna_name<a class="anchor" aria-label="anchor" href="#arg-wgcna-name"></a></dt>
<dd><p>The name of the hdWGCNA experiment in the seurat_ref@misc slot</p></dd>


<dt id="arg-wgcna-name-proj">wgcna_name_proj<a class="anchor" aria-label="anchor" href="#arg-wgcna-name-proj"></a></dt>
<dd><p>The name of the hdWGCNA experiment to be created for the projected modules in seurat_obj</p></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">ProjectModules</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> function (seurat_obj, seurat_ref, modules = NULL, group.by.vars = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     gene_mapping = NULL, genome1_col = NULL, genome2_col = NULL, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     overlap_proportion = 0.5, vars.to.regress = NULL, scale.model.use = "linear", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     wgcna_name = NULL, wgcna_name_proj = NULL, ...) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(wgcna_name)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         wgcna_name &lt;- seurat_ref@misc$active_wgcna</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     CheckWGCNAName(seurat_ref, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (is.null(modules)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         modules &lt;- GetModules(seurat_ref, wgcna_name)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (!all(c("gene_name", "module", "color") %in% colnames(modules))) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             stop("Missing columns in modules table. Required columns are gene_name, module, color")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!is.factor(modules$module)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         modules$module &lt;- as.factor(modules$module)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!is.null(gene_mapping)) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         modules &lt;- TransferModuleGenome(modules, gene_mapping, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             genome1_col, genome2_col)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods &lt;- as.character(unique(modules$module))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mod_props &lt;- unlist(lapply(mods, function(x) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cur_mod &lt;- subset(modules, module == x)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sum(cur_mod$gene_name %in% rownames(seurat_obj))/nrow(cur_mod)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods_keep &lt;- mods[mod_props &gt;= overlap_proportion]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mods_remove &lt;- mods[mod_props &lt; overlap_proportion]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (length(mods) &gt; 0) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         warning(paste0("The following modules will not be projected because too few genes are present in seurat_obj: ", </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             paste(mods_remove, collapse = ", ")))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- subset(modules, module %in% mods_keep) %&gt;% dplyr::mutate(module = droplevels(module))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     gene_names &lt;- modules$gene_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     genes_use &lt;- intersect(gene_names, rownames(seurat_obj))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     modules &lt;- modules %&gt;% subset(gene_name %in% genes_use)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     if (!(wgcna_name_proj %in% names(seurat_obj@misc))) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seurat_obj &lt;- SetupForWGCNA(seurat_obj, wgcna_name = wgcna_name_proj, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             features = genes_use)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     print("project modules")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seurat_obj &lt;- ModuleEigengenes(seurat_obj, group.by.vars = group.by.vars, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         modules = modules, vars.to.regress = vars.to.regress, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scale.model.use = scale.model.use, wgcna_name = wgcna_name_proj, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ...)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seurat_obj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;bytecode: 0x7f962ba3abf0&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;environment: namespace:hdWGCNA&gt;</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>, <a href="https://swaruplab.bio.uci.edu/" class="external-link">Swarup Lab</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

